<div class="container">

<table style="width: 100%;"><tr>
<td>genome_createLocalMode</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generates an interactive genome browser.</h2>

<h3>Description</h3>

<p><code>genome_createLocalMode</code> creates an interactive genome browser to open directly in your browser.
</p>


<h3>Usage</h3>

<pre><code class="language-R">genome_createLocalMode(gb, dir="GenomeBrowser")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gb</code></td>
<td>
<p>a genome browser object produced by the function <code>genomebrowser</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dir</code></td>
<td>
<p>a "character" string representing the directory where the graph will be saved.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The function creates a folder in the working directory of your computer with an HTML document (index.html) which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics, University of Salamanca. See <a href="http://d3gb.usal.es/">http://d3gb.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The ‘D3GB’ Website: <a href="http://d3gb.usal.es">http://d3gb.usal.es</a>
</p>
<p><code>genomebrowser</code>, <code>createAssembly</code>, <code>getAssemblyFromFasta</code>, <code>GRCh37.bands</code>, <code>GRCh38.bands</code>, <code>genome_addGFF</code>, <code>genome_addSequence</code>, <code>genome_addTrack</code>, <code>genome_addVCF</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Genome browser generation.
gb &lt;- genomebrowser(GRCh38.bands)
# It creates a genome browser ready to be viewed in your browser.
genome_createLocalMode(gb,dir=tempfile())
</code></pre>


</div>