<div class="container">

<table style="width: 100%;"><tr>
<td>build_example</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulation of gene expressions using independant negative binomials</h2>

<h3>Description</h3>

<p>Simulation of gene expressions using independant negative binomials
</p>


<h3>Usage</h3>

<pre><code class="language-R">build_example(
  m = 500,
  m1,
  n1 = 6,
  n2 = n1,
  fold = 100,
  mu0 = 100,
  use.scales = FALSE,
  nb.size = Inf
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>m, </code></td>
<td>
<p>number of genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m1, </code></td>
<td>
<p>number of differentially expressed genes. In the expression matrix, m1 first columns contain differentially expressed genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n1, </code></td>
<td>
<p>number of samples under the first condition. The first n1 rows in the expression matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n2, </code></td>
<td>
<p>number of samples under the second condition (default n2=n1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fold, </code></td>
<td>
<p>maximal fold change added to the first m1 genes. The fold decreases proportionally to <code>1/sqrt(1:m1)</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu0, </code></td>
<td>
<p>mean relative expression</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use.scales, </code></td>
<td>
<p>if TRUE random scales are used, otherwise all scales are set to 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nb.size, </code></td>
<td>
<p>number of successful trials in the negative binomial distribution. If nb.size is set to Inf (default), the Poisson model is used.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function generates a list, of which the first element <code>X</code> is a matrix of <code>n1+n2</code> and m dimension with simulated expressions under Poisson or Negative Binomial distribution. Lines <code>1:n1</code> correspond to the first condition (or sub-group) and lines <code>(n1+1):(n1+n2)</code> to the second one. Columns <code>1:m1</code> contain counts imitating differential expressions.
</p>
<p>In the ideal situation there is no microscopical variability between samples and all scales (so-called scaling factors) would be the same.  To simulate examples corresponding to this perfect situation, use argument <code>use.scales=FALSE</code> which will set all scales to 1.  When <code>use.scales=TRUE</code>, scales are simulated under uniform distribution <em>Unif(0.25,4)</em>.
</p>
<p>The fold is maximal for the first expression and decreases proportionally to <code>1/sqrt(1:m1)</code>. The smallest fold  <code>fold/sqrt(m1)</code> is set to the <em>m1</em>-th expression.
</p>


<h3>Value</h3>

<p>A list with components
</p>

<dl>
<dt><code>X</code></dt>
<dd>
<p>a two-dimensional array containing the expression table of n individuals in rows and m gene expressions in columns. </p>
</dd>
<dt><code>m1</code></dt>
<dd>
<p>number of differentially expressed genes (as in arguments).</p>
</dd>
<dt><code>n1</code></dt>
<dd>
<p>number of samples under the first condition (as in arguments).</p>
</dd>
<dt><code>n2</code></dt>
<dd>
<p>number of samples under the second condition (as in arguments).</p>
</dd>
<dt><code>fold</code></dt>
<dd>
<p>maximal fold change between the differentally expressed genes and invariant genes (as in arguments). </p>
</dd>
<dt><code>scales</code></dt>
<dd>
<p>vector of simulated scales.</p>
</dd>
<dt><code>mu0</code></dt>
<dd>
<p>mean relative expression (as in arguments).</p>
</dd>
</dl>
<h3>Examples</h3>

<pre><code class="language-R">
L = build_example(m=500,m1=25,n1=6,fold=20,mu0=100,use.scales=FALSE,nb.size=Inf)
</code></pre>


</div>