<div class="container">

<table style="width: 100%;"><tr>
<td>gl.read.csv</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reads SNP data from a csv file into a genlight object</h2>

<h3>Description</h3>

<p>This script takes SNP genotypes from a csv file, combines them with
individual and locus metrics and creates a genlight object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.read.csv(
  filename,
  transpose = FALSE,
  ind.metafile = NULL,
  loc.metafile = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>Name of the csv file containing the SNP genotypes [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>transpose</code></td>
<td>
<p>If TRUE, rows are loci and columns are individuals
[default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind.metafile</code></td>
<td>
<p>Name of the csv file containing the metrics for
individuals [optional].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loc.metafile</code></td>
<td>
<p>Name of the csv file containing the metrics for
loci [optional].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The SNP data need to be in one of two forms. SNPs can be coded 0 for
homozygous reference, 2 for homozygous alternate, 1 for heterozygous, and NA 
for missing values; or the SNP data can be coded A/A, A/C, C/T, G/A etc,
and -/- for missing data. In this format, the reference allele is the most 
frequent allele, as used by DArT. Other formats will throw an error.
</p>
<p>The SNP data need to be individuals as rows, labeled, and loci as columns,
also labeled. If the orientation is individuals as columns and loci by rows,
then set transpose=TRUE.
</p>
<p>The individual metrics need to be in a csv file, with headings, with a
mandatory id column corresponding exactly to the individual identity labels
provided with the SNP data and in the same order.
</p>
<p>The locus metadata needs to be in a csv file with headings, with a mandatory
column headed AlleleID corresponding exactly to the locus identity labels
provided with the SNP data and in the same order.
</p>
<p>Note that the locus metadata will be complemented by calculable statistics
corresponding to those that would be provided by Diversity Arrays Technology
(e.g. CallRate).
</p>


<h3>Value</h3>

<p>A genlight object with the SNP data and associated metadata included.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos â€“ Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">csv_file &lt;- system.file('extdata','platy_test.csv', package='dartR')
ind_metadata &lt;- system.file('extdata','platy_ind.csv', package='dartR')
gl  &lt;- gl.read.csv(filename = csv_file, ind.metafile = ind_metadata)
</code></pre>


</div>