<div class="container">

<table style="width: 100%;"><tr>
<td>gl2vcf</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Converts a genlight object into vcf format</h2>

<h3>Description</h3>

<p>This function exports a genlight object into VCF format and save it into a
file.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl2vcf(
  x,
  plink.bin.path = getwd(),
  outfile = "gl_vcf",
  outpath = NULL,
  snp.pos = "0",
  snp.chr = "0",
  chr.format = "character",
  pos.cM = "0",
  ID.dad = "0",
  ID.mum = "0",
  sex.code = "unknown",
  phen.value = "0",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plink.bin.path</code></td>
<td>
<p>Path of PLINK binary file [default getwd())].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfile</code></td>
<td>
<p>File name of the output file [default 'gl_vcf'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outpath</code></td>
<td>
<p>Path where to save the output file [default global working 
directory or if not specified, tempdir()].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp.pos</code></td>
<td>
<p>Field name from the slot loc.metrics where the SNP position is
stored [default '0'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp.chr</code></td>
<td>
<p>Field name from the slot loc.metrics where the chromosome of
each is stored [default '0'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr.format</code></td>
<td>
<p>Whether chromosome information is stored as 'numeric' or as
'character', see details [default 'character'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos.cM</code></td>
<td>
<p>A vector, with as many elements as there are loci, containing
the SNP position in morgans or centimorgans [default '0'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ID.dad</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the ID of the father, '0' if father isn't in dataset [default '0'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ID.mum</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the ID of the mother, '0' if mother isn't in dataset [default '0'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sex.code</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the sex code ('male', 'female', 'unknown') [default  'unknown'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phen.value</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the phenotype value. '1' = control, '2' = case, '0' = unknown
[default '0'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function requires to download the binary file of PLINK 1.9 and provide
its path (plink.bin.path).
The binary file can be downloaded from:
<a href="https://www.cog-genomics.org/plink/">https://www.cog-genomics.org/plink/</a>
The chromosome information for unmapped SNPS is coded as 0.
Family ID is taken from  x$pop
Within-family ID (cannot be '0') is taken from indNames(x)
Variant identifier is taken from locNames(x)
</p>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Danecek, P., Auton, A., Abecasis, G., Albers, C. A., Banks, E., DePristo, M.
A., ... &amp; 1000 Genomes Project Analysis Group. (2011). The variant call
format and VCFtools. Bioinformatics, 27(15), 2156-2158.
</p>


<h3>See Also</h3>

<p>Other linker: 
<code>gl2bayesAss()</code>,
<code>gl2bayescan()</code>,
<code>gl2bpp()</code>,
<code>gl2demerelate()</code>,
<code>gl2eigenstrat()</code>,
<code>gl2faststructure()</code>,
<code>gl2gds()</code>,
<code>gl2genalex()</code>,
<code>gl2genepop()</code>,
<code>gl2geno()</code>,
<code>gl2gi()</code>,
<code>gl2hiphop()</code>,
<code>gl2phylip()</code>,
<code>gl2plink()</code>,
<code>gl2related()</code>,
<code>gl2sa()</code>,
<code>gl2structure()</code>,
<code>gl2treemix()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
#this example needs plink installed to work
require("dartR.data")
gl2vcf(platypus.gl,snp.pos='ChromPos_Platypus_Chrom_NCBIv1',
 snp.chr = 'Chrom_Platypus_Chrom_NCBIv1')

## End(Not run)

</code></pre>


</div>