<div class="container">

<table style="width: 100%;"><tr>
<td>gl.dist.ind</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculates a distance matrix for individuals defined in a genlight object</h2>

<h3>Description</h3>

<p>This script calculates various distances between individuals based on allele
frequencies or presence-absence data
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.dist.ind(
  x,
  method = NULL,
  scale = FALSE,
  swap = FALSE,
  output = "dist",
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight containing the SNP genotypes or presence-absence data [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Specify distance measure [SNP: Euclidean; P/A: Simple].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale</code></td>
<td>
<p>If TRUE, the distances are scaled to fall in the range [0,1] [default TRUE]</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>swap</code></td>
<td>
<p>If TRUE and working with presence-absence data, then presence 
(no disrupting mutation) is scored as 0 and absence (presence of a disrupting 
mutation) is scored as 1 [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output</code></td>
<td>
<p>Specify the format and class of the object to be returned, 
'dist' for a object of class dist, 'matrix' for an object of class matrix [default "dist"].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.out</code></td>
<td>
<p>If TRUE, display a histogram and a boxplot of the genetic distances [TRUE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_theme</code></td>
<td>
<p>User specified theme [default theme_dartR].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_colors</code></td>
<td>
<p>Vector with two color names for the borders and fill [default two_colors].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots to the session temporary directory [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The distance measure for SNP genotypes can be one of:
</p>

<ul>
<li>
<p> Euclidean Distance [method = "Euclidean"]
</p>
</li>
<li>
<p> Scaled Euclidean Distance [method='Euclidean", scale=TRUE]
</p>
</li>
<li>
<p> Simple Mismatch Distance [method="Simple"]
</p>
</li>
<li>
<p> Absolute Mismatch Distance [method="Absolute"]
</p>
</li>
<li>
<p> Czekanowski (Manhattan) Distance [method="Manhattan"]
</p>
</li>
</ul>
<p>The distance measure for Sequence Tag Presence/Absence data (binary) can be one of:
</p>

<ul>
<li>
<p> Euclidean Distance [method = "Euclidean"]
</p>
</li>
<li>
<p> Scaled Euclidean Distance [method='Euclidean", scale=TRUE]
</p>
</li>
<li>
<p> Simple Matching Distance [method="Simple"]
</p>
</li>
<li>
<p> Jaccard Distance [method="Jaccard"]
</p>
</li>
<li>
<p> Bray-Curtis Distance [method="Bray-Curtis"]
</p>
</li>
</ul>
<p>Refer to the dartR Technical Note on Distances in Genetics.
</p>


<h3>Value</h3>

<p>An object of class 'matrix' or dist' giving distances between individuals
</p>


<h3>Author(s)</h3>

<p>Author(s): Arthur Georges. Custodian: Arthur Georges â€“ Post to #' <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R"> 
D &lt;- gl.dist.ind(testset.gl[1:20,], method='manhattan')
D &lt;- gl.dist.ind(testset.gs[1:20,], method='Jaccard',swap=TRUE)

D &lt;- gl.dist.ind(testset.gl[1:20,], method='euclidean',scale=TRUE)
</code></pre>


</div>