<div class="container">

<table style="width: 100%;"><tr>
<td>gl.sample</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Samples individuals from populations</h2>

<h3>Description</h3>

<p>A function to subsample individuals in a genlight object
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.sample(
  x,
  nsample = min(table(pop(x))),
  replace = TRUE,
  onepop = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>genlight object containing SNP/silicodart genotypes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nsample</code></td>
<td>
<p>the number of individuals that should be sampled</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>replace</code></td>
<td>
<p>a switch to sample by replacement (default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>onepop</code></td>
<td>
<p>switch to ignore population settings of the genlight object and sample from all individuals disregarding the population definition. [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>set verbosity</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This is convenience function to facilitate a bootstrap approach
</p>
<p>This function is often used to support a bootstrap 
approach in dartR. For a bootstrap approach it is often 
desirable to sample a defined number of individuals for 
each of the populations in a genlight object and then 
calculate a certain quantity for that subset (redo a 
1000 times)
</p>


<h3>Value</h3>

<p>returns a genlight object with nsample samples from each populations.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other data manipulation: 
<code>gl.define.pop()</code>,
<code>gl.drop.ind()</code>,
<code>gl.drop.loc()</code>,
<code>gl.drop.pop()</code>,
<code>gl.edit.recode.pop()</code>,
<code>gl.impute()</code>,
<code>gl.join()</code>,
<code>gl.keep.ind()</code>,
<code>gl.keep.loc()</code>,
<code>gl.keep.pop()</code>,
<code>gl.make.recode.ind()</code>,
<code>gl.merge.pop()</code>,
<code>gl.reassign.pop()</code>,
<code>gl.recode.ind()</code>,
<code>gl.recode.pop()</code>,
<code>gl.rename.pop()</code>,
<code>gl.sim.genotypes()</code>,
<code>gl.sort()</code>,
<code>gl.subsample.ind()</code>,
<code>gl.subsample.loc()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
#bootstrap for 2 possums populations to check effect of sample size on fixed alleles
gl.set.verbosity(0)
pp &lt;- possums.gl[c(1:30,91:120),]
nrep &lt;- 1:10
nss &lt;- seq(1,10,2)
res &lt;- expand.grid(nrep=nrep, nss=nss)
for (i in 1:nrow(res)) {
dummy &lt;- gl.sample(pp, nsample=res$nss[i], replace=TRUE)
dummy &lt;- gl.compliance.check(dummy)
pas &lt;- gl.report.pa(dummy, plot.display= FALSE)
res$fixed[i] &lt;- pas$fixed[1]
}
boxplot(fixed ~ nss, data=res)

</code></pre>


</div>