<div class="container">

<table style="width: 100%;"><tr>
<td>gl.LDNe</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimates effective population size using the Linkage Disequilibrium
method based on NeEstimator (V2)</h2>

<h3>Description</h3>

<p>This function is basically a convenience function that runs the LD Ne
estimator using Neestimator2
(<a href="http://www.molecularfisherieslaboratory.com.au/neestimator-software/">http://www.molecularfisherieslaboratory.com.au/neestimator-software/</a>)
within R using the provided genlight object. To be able to do so, the
software has to be downloaded from their website and the appropriate
executable Ne2-1 has to be copied into the path as specified in the function
(see example below).
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.LDNe(
  x,
  outfile = "genepopLD.txt",
  outpath = tempdir(),
  neest.path = getwd(),
  critical = 0,
  singleton.rm = TRUE,
  mating = "random",
  pairing = "all",
  Waples.correction = NULL,
  Waples.correction.value = NULL,
  naive = FALSE,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors_pop = gl.select.colors(x, verbose = 0),
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfile</code></td>
<td>
<p>File name of the output file  with
all results from Neestimator 2 [default 'genepopLD.txt'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outpath</code></td>
<td>
<p>Path where to save the output file. Use outpath=getwd() or
outpath='.' when calling this function to direct output files to your working
directory [default tempdir(), mandated by CRAN].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>neest.path</code></td>
<td>
<p>Path to the folder of the NE2-1 file.
Please note there are 3 different executables depending on your OS:
Ne2-1.exe (=Windows), Ne2-1M (=Mac), Ne2-1L (=Linux). You only need to point
to the folder (the function will recognise which OS you are running)
[default getwd()].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>critical</code></td>
<td>
<p>(vector of) Critical values that are used to remove alleles
based on their minor allele frequency. This can be done before using the
gl.filter.maf function, therefore the default is set to 0 (no loci are
removed). To run for MAF 0 and MAF 0.05 at the same time specify: critical =
c(0,0.05) [default 0].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>singleton.rm</code></td>
<td>
<p>Whether to remove singleton alleles [default TRUE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mating</code></td>
<td>
<p>Formula for Random mating='random' or monogamy= 'monogamy'
[default 'random'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pairing</code></td>
<td>
<p>'all' [default] if all possible loci should be paired, or 'separate'
if only loci on different chromosomes should be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Waples.correction</code></td>
<td>
<p>The type of Waples et al 2016 correction to apply. 
This is ignored if <code>pairing</code> is set to 'separate'.
Options are 'nChromosomes', for eq 1a, or 'genomeLength' for eq 1b. 
NULL if none should be applied [default NULL].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Waples.correction.value</code></td>
<td>
<p>The number of chromosomes or the genome length 
in cM. See Waples et al 2016 for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>naive</code></td>
<td>
<p>Whether the naive (uncorrected for samples size - see 
eq 7 and eq 8 in Waples 2006) should also be reported. This is mostly 
to diagnose the source of Inf estimate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_colors_pop</code></td>
<td>
<p>population colors with as many colors as there are populations in the dataset
[default discrete_palette].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]
temporary directory (tempdir) [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Dataframe with the results as table
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li>
<p> Waples, R. S. (2006). "A bias correction for estimates of effective 
population size based on linkage disequilibrium at unlinked gene loci*." 
Conservation Genetics 7(2): 167-184.
</p>
</li>
<li>
<p> Waples, R. K., et al. (2016). "Estimating contemporary effective 
population size in non-model species using linkage disequilibrium across 
thousands of loci." Heredity 117(4): 233-240.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# SNP data (use two populations and only the first 100 SNPs)
pops &lt;- possums.gl[1:60, 1:100]
nes &lt;- gl.LDNe(pops,
  outfile = "popsLD.txt", outpath = tempdir(),
  neest.path = "./path_to Ne-21",
  critical = c(0, 0.05), singleton.rm = TRUE, mating = "random"
)
nes

# Using only pairs of loci on different chromosomes
# make up some chromosome location
pops@chromosome &lt;- as.factor(sample(1:10, size = nLoc(pops), replace = TRUE))
nessep &lt;- gl.LDNe(pops,
              outfile = "popsLD.txt", outpath = "./TestNe", pairing="separate",
              neest.path = "./path_to Ne-21",
              critical = c(0, 0.05), singleton.rm = TRUE, mating = "random"
nessep

## End(Not run)
</code></pre>


</div>