<div class="container">

<table style="width: 100%;"><tr>
<td>gl.report.secondaries</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reports loci containing secondary SNPs in sequence tags and calculates
number of invariant sites</h2>

<h3>Description</h3>

<p>SNP datasets generated by DArT include fragments with more than
one SNP (that is, with secondaries). They are recorded separately with the
same CloneID (=AlleleID). These multiple SNP loci within a fragment are
likely to be linked, and so you may wish to remove secondaries.
</p>
<p>This function reports statistics associated with secondaries, and the
consequences of filtering them out, and provides three plots. The first is
a boxplot, the second is a barplot of the frequency of secondaries per
sequence tag, and the third is the Poisson expectation for those
frequencies including an estimate of the zero class (no. of sequence tags
with no SNP scored).
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.report.secondaries(
  x,
  nsim = 1000,
  taglength = 69,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nsim</code></td>
<td>
<p>The number of simulations to estimate the mean of the Poisson
distribution [default 1000].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taglength</code></td>
<td>
<p>Typical length of the sequence tags [default 69].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options [default
theme_dartR()].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default two_colors].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function <code>gl.filter.secondaries</code> will filter out the
loci with secondaries retaining only one sequence tag.
</p>
<p>Heterozygosity as estimated by the function
<code>gl.report.heterozygosity</code> is in a sense relative, because it
is calculated against a background of only those loci that are polymorphic
somewhere in the dataset. To allow intercompatibility across studies and
species, any measure of heterozygosity needs to accommodate loci that are
invariant (autosomal heterozygosity. See Schmidt et al 2021). However, the
number of invariant loci are unknown given the SNPs are detected as single
point mutational variants and invariant sequences are discarded, and
because of the particular additional filtering pre-analysis. Modelling the
counts of SNPs per sequence tag as a Poisson distribution in this script
allows estimate of the zero class, that is, the number of invariant loci.
This is reported, and the veracity of the estimate can be assessed by the
correspondence of the observed frequencies against those under Poisson
expectation in the associated graphs. The number of invariant loci can then
be optionally provided to the function
<code>gl.report.heterozygosity</code> via the parameter n.invariants.
</p>
<p>In case the calculations for the Poisson expectation of the number of
invariant sequence tags fail to converge, try to rerun the analysis with a
larger <code>nsim</code> values.
</p>
<p>This function now also calculates the number of invariant sites (i.e.
nucleotides) of the sequence tags (if <code>TrimmedSequence</code> is present in
<code>x$other$loc.metrics</code>) or estimate these by assuming that the average
length of the sequence tags is 69 nucleotides. Based on the Poisson
expectation of the number of invariant sequence tags, it also estimates the
number of invariant sites for these to eventually provide an estimate of
the total number of invariant sites.
</p>
<p><strong>Note</strong>, previous version of
<code>dartR</code> would only return an estimate of the number of invariant
sequence tags (not sites).
</p>
<p>Plots are saved to the session temporary directory (tempdir).
</p>
<p>Examples of other themes that can be used can be consulted in:
</p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li>
</ul>
<h3>Value</h3>

<p>A data.frame with the list of parameter values
</p>

<ul>
<li>
<p> n.total.tags Number of sequence tags in total
</p>
</li>
<li>
<p> n.SNPs.secondaries Number of secondary SNP loci that would be removed
on filtering
</p>
</li>
<li>
<p> n.invariant.tags Estimated number of invariant sequence tags
</p>
</li>
<li>
<p> n.tags.secondaries Number of sequence tags with secondaries
</p>
</li>
<li>
<p> n.inv.gen Number of invariant sites in sequenced tags
</p>
</li>
<li>
<p> mean.len.tag Mean length of sequence tags
</p>
</li>
<li>
<p> n.invariant Total Number of invariant sites (including invariant
sequence tags)
</p>
</li>
<li>
<p> k Lambda: mean of the Poisson distribution of number of SNPs in the
sequence tags
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Schmidt, T.L., Jasper, M.-E., Weeks, A.R., Hoffmann, A.A., 2021.
Unbiased population heterozygosity estimates from genome-wide sequence
data. Methods in Ecology and Evolution n/a.
</p>


<h3>See Also</h3>

<p><code>gl.filter.secondaries</code>,<code>gl.report.heterozygosity</code>,
<code>utils.n.var.invariant</code>
</p>
<p>Other report functions: 
<code>gl.report.bases()</code>,
<code>gl.report.callrate()</code>,
<code>gl.report.diversity()</code>,
<code>gl.report.hamming()</code>,
<code>gl.report.heterozygosity()</code>,
<code>gl.report.hwe()</code>,
<code>gl.report.ld.map()</code>,
<code>gl.report.locmetric()</code>,
<code>gl.report.maf()</code>,
<code>gl.report.monomorphs()</code>,
<code>gl.report.overshoot()</code>,
<code>gl.report.parent.offspring()</code>,
<code>gl.report.pa()</code>,
<code>gl.report.rdepth()</code>,
<code>gl.report.reproducibility()</code>,
<code>gl.report.sexlinked()</code>,
<code>gl.report.taglength()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">require("dartR.data")
test &lt;- gl.filter.callrate(platypus.gl,threshold = 1)
n.inv &lt;- gl.report.secondaries(test)
gl.report.heterozygosity(test, n.invariant = n.inv[7, 2])
</code></pre>


</div>