<div class="container">

<table style="width: 100%;"><tr>
<td>utils.is.fixed</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>An internal function to tests if two populations are fixed at a given locus</h2>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class="language-R">utils.is.fixed(s1, s2, tloc = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>s1</code></td>
<td>
<p>Percentage SNP allele or sequence tag frequency for the first 
population [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>s2</code></td>
<td>
<p>Percentage SNP allele or sequence tag frequency for the second 
population [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tloc</code></td>
<td>
<p>Threshold value for tolerance in when a difference is regarded as
fixed [default 0].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This script compares two percent allele frequencies
and reports TRUE if they represent a fixed difference, FALSE otherwise.
</p>
<p>A fixed difference at a locus occurs when two populations share no alleles,
noting that SNPs are biallelic (ploidy=2).
Tolerance in the definition of a fixed difference is provided by the t
parameter. For example, t=0.05 means that SNP allele frequencies of 95,5 and
5,95 percent will be reported as fixed (TRUE).
</p>


<h3>Value</h3>

<p>TRUE (fixed difference) or FALSE (alleles shared) or NA (one or both s1 or s2 missing)
</p>


<h3>Author(s)</h3>

<p>Maintainer: Arthur Georges (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code>gl.fixed.diff</code>
</p>
<p>Other utilities: 
<code>gl.alf()</code>,
<code>utils.check.datatype()</code>,
<code>utils.dart2genlight()</code>,
<code>utils.dist.binary()</code>,
<code>utils.flag.start()</code>,
<code>utils.hamming()</code>,
<code>utils.het.pop()</code>,
<code>utils.impute</code>,
<code>utils.jackknife()</code>,
<code>utils.n.var.invariant()</code>,
<code>utils.plot.save()</code>,
<code>utils.read.fasta()</code>,
<code>utils.read.ped()</code>,
<code>utils.recalc.avgpic()</code>,
<code>utils.recalc.callrate()</code>,
<code>utils.recalc.freqhets()</code>,
<code>utils.recalc.freqhomref()</code>,
<code>utils.recalc.freqhomsnp()</code>,
<code>utils.recalc.maf()</code>,
<code>utils.reset.flags()</code>,
<code>utils.transpose()</code>,
<code>utils.vcfr2genlight.polyploid()</code>
</p>


</div>