<div class="container">

<table style="width: 100%;"><tr>
<td>gl.report.overshoot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reports loci for which the SNP has been trimmed from the sequence tag
along with the adaptor</h2>

<h3>Description</h3>

<p>This function checks the position of the SNP within the trimmed sequence tag
and identifies those for which the SNP position is outside the trimmed
sequence tag. This can happen, rarely, when the sequence containing the SNP
resembles the adaptor.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.report.overshoot(x, save2tmp = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The SNP genotype can still be used in most analyses, but functions like
gl2fasta() will present challenges if the SNP has been trimmed from the 
sequence tag.
</p>
<p>Resultant ggplot(s) and the tabulation(s) are saved to the session's
temporary directory.
</p>


<h3>Value</h3>

<p>An unaltered genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges â€“ Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code>gl.filter.overshoot</code>
</p>
<p>Other report functions: 
<code>gl.report.bases()</code>,
<code>gl.report.callrate()</code>,
<code>gl.report.diversity()</code>,
<code>gl.report.hamming()</code>,
<code>gl.report.heterozygosity()</code>,
<code>gl.report.hwe()</code>,
<code>gl.report.ld.map()</code>,
<code>gl.report.locmetric()</code>,
<code>gl.report.maf()</code>,
<code>gl.report.monomorphs()</code>,
<code>gl.report.parent.offspring()</code>,
<code>gl.report.pa()</code>,
<code>gl.report.rdepth()</code>,
<code>gl.report.reproducibility()</code>,
<code>gl.report.secondaries()</code>,
<code>gl.report.sexlinked()</code>,
<code>gl.report.taglength()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">gl.report.overshoot(testset.gl)
</code></pre>


</div>