<div class="container">

<table style="width: 100%;"><tr>
<td>utils.dist.ind.snp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculates a distance matrix for individuals defined in a 
genlight object using SNP data (DArTseq)</h2>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class="language-R">utils.dist.ind.snp(
  x,
  method = "Euclidean",
  scale = FALSE,
  type = "dist",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight containing the genotypes [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Specify distance measure [default Euclidean].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale</code></td>
<td>
<p>If TRUE and method='Euclidean', the distance will be scaled to 
fall in the range [0,1] [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Specify the format and class of the object to be returned, 
dist for a object of class dist, matrix for an object of class matrix [default "dist"].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report [default 2].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This script calculates various distances between individuals based on 
SNP genotypes.
The distance measure can be one of:
</p>

<ul>
<li>
<p> Euclidean – Euclidean Distance applied to Cartesian coordinates defined
by the loci, scored as 0, 1 or 2. 
</p>
</li>
<li>
<p> Simple – simple mismatch, 0 where no alleles are shared, 1 where one
allele is shared, 2 where both alleles are shared. 
</p>
</li>
<li>
<p> Absolute – absolute mismatch, 0 where no alleles are shared, 1 where
one or both alleles are shared.
</p>
</li>
<li>
<p> Czekanowski (or Manhattan) calculates the city block metric distance
by summing the scores on each axis (locus).
</p>
</li>
</ul>
<h3>Value</h3>

<p>An object of class 'dist' or 'matrix' giving distances between individuals
</p>


<h3>Author(s)</h3>

<p>Author(s): Arthur Georges. Custodian: Arthur Georges – Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other distance: 
<code>gl.dist.ind()</code>,
<code>gl.dist.pop()</code>,
<code>gl.fdsim()</code>
</p>


</div>