<div class="container">

<table style="width: 100%;"><tr>
<td>gl2vcf</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Converts a genlight object into vcf format</h2>

<h3>Description</h3>

<p>This function exports a genlight object into VCF format and save it into a
file.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl2vcf(
  x,
  plink_path = getwd(),
  outfile = "gl_vcf",
  outpath = tempdir(),
  snp_pos = "0",
  snp_chr = "0",
  chr_format = "character",
  pos_cM = "0",
  ID_dad = "0",
  ID_mom = "0",
  sex_code = "unknown",
  phen_value = "0",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plink_path</code></td>
<td>
<p>Path of PLINK binary file [default getwd())].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfile</code></td>
<td>
<p>File name of the output file [default 'gl_vcf'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp_pos</code></td>
<td>
<p>Field name from the slot loc.metrics where the SNP position is
stored [default '0'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp_chr</code></td>
<td>
<p>Field name from the slot loc.metrics where the chromosome of
each is stored [default '0'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr_format</code></td>
<td>
<p>Whether chromosome information is stored as 'numeric' or as
'character', see details [default 'character'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos_cM</code></td>
<td>
<p>A vector, with as many elements as there are loci, containing
the SNP position in morgans or centimorgans [default '0'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ID_dad</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the ID of the father, '0' if father isn't in dataset [default '0'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ID_mom</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the ID of the mother, '0' if mother isn't in dataset [default '0'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sex_code</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the sex code ('male', 'female', 'unknown') [default  'unknown'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phen_value</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the phenotype value. '1' = control, '2' = case, '0' = unknown
[default '0'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function requires to download the binary file of PLINK 1.9 and provide
its path (plink_path).
The binary file can be downloaded from:
<a href="https://www.cog-genomics.org/plink/">https://www.cog-genomics.org/plink/</a>
</p>
<p>The chromosome information for unmapped SNPS is coded as 0.
Family ID is taken from  x$pop
Within-family ID (cannot be '0') is taken from indNames(x)
Variant identifier is taken from locNames(x)
</p>
<p>#' Note that if names of populations or individuals contain spaces, they are 
replaced by an underscore "_".
</p>
<p>If you like to use chromosome information when converting to plink format and
your chromosome names are not from human, you need to change the chromosome 
names as 'contig1', 'contig2', etc. as described in the section "Nonstandard
chromosome IDs" in the following link:
https://www.cog-genomics.org/plink/1.9/input
</p>
<p>Note that the function might not work if there are spaces in the path to the
plink executable.
</p>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Danecek, P., Auton, A., Abecasis, G., Albers, C. A., Banks, E., DePristo, M.
A., ... &amp; 1000 Genomes Project Analysis Group. (2011). The variant call
format and VCFtools. Bioinformatics, 27(15), 2156-2158.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
require("dartR.data")
gl2vcf(platypus.gl,snp_pos='ChromPos_Platypus_Chrom_NCBIv1',
 snp_chr = 'Chrom_Platypus_Chrom_NCBIv1')

## End(Not run)
</code></pre>


</div>