<div class="container">

<table style="width: 100%;"><tr>
<td>gl.ld.distance</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plots linkage disequilibrium against distance by population 
disequilibrium patterns</h2>

<h3>Description</h3>

<p>The function creates a plot showing
the pairwise LD measure against distance in number of base pairs pooled over
all the chromosomes and a red line representing the threshold (R.squared = 
0.2) that is commonly used to imply that two loci are unlinked (Delourme et
al., 2013; Li et al., 2014).
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.ld.distance(
  ld_report,
  ld_resolution = 1e+05,
  pop_colors = NULL,
  plot_theme = NULL,
  plot.out = TRUE,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ld_report</code></td>
<td>
<p>Output from function <code>gl.report.ld.map</code> 
[required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ld_resolution</code></td>
<td>
<p>Resolution at which LD should be reported in number of 
base pairs [default NULL].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop_colors</code></td>
<td>
<p>A color palette for box plots by population or a list
with as many colors as there are populations in the dataset
[default NULL].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_theme</code></td>
<td>
<p>User specified theme [default NULL].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A dataframe with information of LD against distance by population.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos – Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li>
<p> Delourme, R., Falentin, C., Fomeju, B. F., Boillot, M., Lassalle, G., 
André, I., . . . Marty, A. (2013). High-density SNP-based genetic map 
development and linkage disequilibrium assessment in Brassica napusL. BMC 
genomics, 14(1), 120.
</p>
</li>
<li>
<p> Li, X., Han, Y., Wei, Y., Acharya, A., Farmer, A. D., Ho, J., . . . 
Brummer, E. C. (2014). Development of an alfalfa SNP array and its use to 
evaluate patterns of population structure and linkage disequilibrium. PLoS 
One, 9(1), e84329.
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Other ld functions: 
<code>gl.ld.haplotype()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">if ((requireNamespace("snpStats", quietly = TRUE)) &amp; (requireNamespace("fields", quietly = TRUE))) {
require("dartR.data")
x &lt;- platypus.gl
x &lt;- gl.filter.callrate(x,threshold = 1)
x &lt;- gl.filter.monomorphs(x)
x$position &lt;- x$other$loc.metrics$ChromPos_Platypus_Chrom_NCBIv1
x$chromosome &lt;- as.factor(x$other$loc.metrics$Chrom_Platypus_Chrom_NCBIv1)
ld_res &lt;- gl.report.ld.map(x,ld_max_pairwise = 10000000)
ld_res_2 &lt;- gl.ld.distance(ld_res,ld_resolution= 1000000)
}
</code></pre>


</div>