<div class="container">

<table style="width: 100%;"><tr>
<td>utils.recalc.avgpic</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>A utility function to recalculate intermediate locus metrics</h2>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class="language-R">utils.recalc.avgpic(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Recalculates OneRatioRef, OneRatioSnp, PICRef, PICSnp,
and AvgPIC by locus after some individuals or populations have been deleted.
</p>
<p>The locus metadata supplied by DArT has OneRatioRef, OneRatioSnp, PICRef,
PICSnp, and AvgPIC included, but the allelic composition will change when
some individuals,or populations, are removed from the dataset and so the
initial statistics will no longer apply. This script recalculates these
statistics and places the recalculated values in the appropriate place in
the genlight object.
If the locus metadata OneRatioRef|Snp, PICRef|Snp and/or AvgPIC do not exist,
the script creates and populates them.
</p>


<h3>Value</h3>

<p>The modified genlight object.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code>utils.recalc.metrics</code> for recalculating all metrics,
<code>utils.recalc.callrate</code> for recalculating CallRate,
<code>utils.recalc.freqhomref</code> for recalculating frequency of homozygous
reference, <code>utils.recalc.freqhomsnp</code> for recalculating frequency of
homozygous alternate, <code>utils.recalc.freqhet</code> for recalculating frequency
of heterozygotes, <code>gl.recalc.maf</code> for recalculating minor allele
frequency, <code>gl.recalc.rdepth</code> for recalculating average read depth
</p>
<p>Other utilities: 
<code>gl.alf()</code>,
<code>utils.check.datatype()</code>,
<code>utils.dart2genlight()</code>,
<code>utils.dist.binary()</code>,
<code>utils.flag.start()</code>,
<code>utils.hamming()</code>,
<code>utils.het.pop()</code>,
<code>utils.impute</code>,
<code>utils.is.fixed()</code>,
<code>utils.jackknife()</code>,
<code>utils.n.var.invariant()</code>,
<code>utils.plot.save()</code>,
<code>utils.read.fasta()</code>,
<code>utils.read.ped()</code>,
<code>utils.recalc.callrate()</code>,
<code>utils.recalc.freqhets()</code>,
<code>utils.recalc.freqhomref()</code>,
<code>utils.recalc.freqhomsnp()</code>,
<code>utils.recalc.maf()</code>,
<code>utils.reset.flags()</code>,
<code>utils.transpose()</code>,
<code>utils.vcfr2genlight.polyploid()</code>
</p>


</div>