<div class="container">

<table style="width: 100%;"><tr>
<td>gl2snapp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Converts a genlight object to nexus format suitable for phylogenetic analysis
by SNAPP (via BEAUti)</h2>

<h3>Description</h3>

<p>The output nexus file contains the SNP data and relevant PAUP command lines
suitable for BEAUti.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl2snapp(x, outfile = "snapp.nex", outpath = tempdir(), verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfile</code></td>
<td>
<p>File name of the output file (including extension)
[default "snapp.nex"].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Bryant, D., Bouckaert, R., Felsenstein, J., Rosenberg, N.A. and
RoyChoudhury, A. (2012). Inferring species trees directly from biallelic
genetic markers: bypassing gene trees in a full coalescent analysis.
Molecular Biology and Evolution 29:1917-1932.
</p>


<h3>Examples</h3>

<pre><code class="language-R">gl2snapp(testset.gl)
</code></pre>


</div>