<div class="container">

<table style="width: 100%;"><tr>
<td>gl.filter.locmetric</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Filters loci on the basis of numeric information stored in
other$loc.metrics in a genlight {adegenet} object</h2>

<h3>Description</h3>

<p>This script uses any field with numeric values stored in $other$loc.metrics
to filter loci. The loci to keep can be within the upper and lower thresholds
('within') or outside of the upper and lower thresholds ('outside').
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.filter.locmetric(x, metric, upper, lower, keep = "within", verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>metric</code></td>
<td>
<p>Name of the metric to be used for filtering [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>upper</code></td>
<td>
<p>Filter upper threshold [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lower</code></td>
<td>
<p>Filter lower threshold  [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep</code></td>
<td>
<p>Whether keep loci within of upper and lower thresholds or keep
loci outside of upper and lower thresholds [within].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The fields that are included in dartR, and a short description, are found
below. Optionally, the user can also set his/her own filter by adding a
vector into $other$loc.metrics as shown in the example.
</p>

<ol>
<li>
<p> SnpPosition - position (zero is position 1) in the sequence tag of the
defined SNP variant base.
</p>
</li>
<li>
<p> CallRate - proportion of samples for which the genotype call is
non-missing (that is, not '-' ).
</p>
</li>
<li>
<p> OneRatioRef - proportion of samples for which the genotype score is 0.
</p>
</li>
<li>
<p> OneRatioSnp - proportion of samples for which the genotype score is 2.
</p>
</li>
<li>
<p> FreqHomRef - proportion of samples homozygous for the Reference allele.
</p>
</li>
<li>
<p> FreqHomSnp - proportion of samples homozygous for the Alternate (SNP)
allele.
</p>
</li>
<li>
<p> FreqHets - proportion of samples which score as heterozygous, that is,
scored as 1.
</p>
</li>
<li>
<p> PICRef - polymorphism information content (PIC) for the Reference
allele.
</p>
</li>
<li>
<p> PICSnp - polymorphism information content (PIC) for the SNP.
</p>
</li>
<li>
<p> AvgPIC - average of the polymorphism information content (PIC) of the
Reference and SNP alleles.
</p>
</li>
<li>
<p> AvgCountRef - sum of the tag read counts for all samples, divided by
the number of samples with non-zero tag read counts, for the Reference allele
row.
</p>
</li>
<li>
<p> AvgCountSnp - sum of the tag read counts for all samples, divided by
the number of samples with non-zero tag read counts, for the Alternate (SNP)
allele row.
</p>
</li>
<li>
<p> RepAvg - proportion of technical replicate assay pairs for which the
marker score is consistent.
</p>
</li>
</ol>
<h3>Value</h3>

<p>The reduced genlight dataset.
</p>


<h3>Author(s)</h3>

<p>Luis Mijangos â€“ Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other matched filter: 
<code>gl.filter.callrate()</code>,
<code>gl.filter.hamming()</code>,
<code>gl.filter.ld()</code>,
<code>gl.filter.maf()</code>,
<code>gl.filter.monomorphs()</code>,
<code>gl.filter.overshoot()</code>,
<code>gl.filter.pa()</code>,
<code>gl.filter.secondaries()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># adding dummy data
test &lt;- testset.gl
test$other$loc.metrics$test &lt;- 1:nLoc(test)
result &lt;- gl.filter.locmetric(x=test, metric= 'test', upper=255,
lower=200, keep= 'within', verbose=3)

</code></pre>


</div>