<div class="container">

<table style="width: 100%;"><tr>
<td>gl.report.ld.map</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculates pairwise linkage disequilibrium by population</h2>

<h3>Description</h3>

<p>This function calculates pairwise linkage disequilibrium (LD) by population 
using the function <code>ld</code> (package snpStats).
</p>
<p>If SNPs are not mapped to a reference genome, the parameter
<code>ld_max_pairwise</code>
should be set as NULL (the default). In this case, the 
function will assign the same chromosome ("1") to all the SNPs in the dataset
and assign a sequence from 1 to n loci as the position of each SNP. The 
function will then calculate LD for all possible SNP pair combinations. 
</p>
<p>If SNPs are mapped to a reference genome, the parameter 
<code>ld_max_pairwise</code>
should be filled out (i.e. not NULL). In this case, the
information for SNP's position should be stored in the genlight accessor
"@position" and the SNP's chromosome name in the accessor "@chromosome"
(see examples). The function will then calculate LD within each chromosome
and for all possible SNP pair combinations within a distance of
<code>ld_max_pairwise</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.report.ld.map(
  x,
  ld_max_pairwise = NULL,
  maf = 0.05,
  ld_stat = "R.squared",
  ind.limit = 10,
  stat_keep = "AvgPIC",
  ld_threshold_pops = 0.2,
  plot.out = TRUE,
  plot_theme = NULL,
  histogram_colors = NULL,
  boxplot_colors = NULL,
  bins = 50,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ld_max_pairwise</code></td>
<td>
<p>Maximum distance in number of base pairs at which LD 
should be calculated [default NULL].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf</code></td>
<td>
<p>Minor allele frequency (by population) threshold to filter out 
loci. If a value &gt; 1 is provided it will be interpreted as MAC (i.e. the
minimum number of times an allele needs to be observed) [default 0.05].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ld_stat</code></td>
<td>
<p>The LD measure to be calculated: "LLR", "OR", "Q", "Covar",
"D.prime", "R.squared", and "R". See <code>ld</code>
(package snpStats) for details [default "R.squared"].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind.limit</code></td>
<td>
<p>Minimum number of individuals that a population should
contain to take it in account to report loci in LD [default 10].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stat_keep</code></td>
<td>
<p>Name of the column from the slot <code>loc.metrics</code> to be
used to choose SNP to be kept [default "AvgPIC"].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ld_threshold_pops</code></td>
<td>
<p>LD threshold to report in the plot of "Number of 
populations in which the same SNP pair are in LD" [default 0.2].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_theme</code></td>
<td>
<p>User specified theme [default NULL].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>histogram_colors</code></td>
<td>
<p>Vector with two color names for the borders and fill
[default NULL].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>boxplot_colors</code></td>
<td>
<p>A color palette for box plots by population or a list
with as many colors as there are populations in the dataset
[default NULL].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bins</code></td>
<td>
<p>Number of bins to display in histograms [default 50].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function reports LD between SNP pairs by population. 
The function <code>gl.filter.ld</code> filters out the SNPs in LD using as
input the results of <code>gl.report.ld.map</code>. The actual number of 
SNPs to be filtered out depends on the parameters set in the function 
<code>gl.filter.ld</code>.
</p>
<p>Boxplots of LD by population and
a histogram showing LD frequency are presented.
</p>


<h3>Value</h3>

<p>A dataframe with information for each SNP pair in LD.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos â€“ Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code>gl.filter.ld</code>
</p>
<p>Other report functions: 
<code>gl.report.bases()</code>,
<code>gl.report.callrate()</code>,
<code>gl.report.diversity()</code>,
<code>gl.report.hamming()</code>,
<code>gl.report.heterozygosity()</code>,
<code>gl.report.hwe()</code>,
<code>gl.report.locmetric()</code>,
<code>gl.report.maf()</code>,
<code>gl.report.monomorphs()</code>,
<code>gl.report.overshoot()</code>,
<code>gl.report.parent.offspring()</code>,
<code>gl.report.pa()</code>,
<code>gl.report.rdepth()</code>,
<code>gl.report.reproducibility()</code>,
<code>gl.report.secondaries()</code>,
<code>gl.report.sexlinked()</code>,
<code>gl.report.taglength()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">require("dartR.data")
x &lt;- platypus.gl
x &lt;- gl.filter.callrate(x,threshold = 1)
x &lt;- gl.filter.monomorphs(x)
x$position &lt;- x$other$loc.metrics$ChromPos_Platypus_Chrom_NCBIv1
x$chromosome &lt;- as.factor(x$other$loc.metrics$Chrom_Platypus_Chrom_NCBIv1)
ld_res &lt;- gl.report.ld.map(x,ld_max_pairwise = 10000000)
</code></pre>


</div>