<div class="container">

<table style="width: 100%;"><tr>
<td>utils.vcfr2genlight.polyploid</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Utility function to convert polyploid vcfR object as genlight</h2>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class="language-R">utils.vcfr2genlight.polyploid(x, n.cores = 1, mode2 = mode)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the vcfR object [defined in function <code>gl.read.vcf</code>].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.cores</code></td>
<td>
<p>Number of cores [default 1]</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mode2</code></td>
<td>
<p>genotype: all heterozygous sites will be coded as 1 regardless ploidy level, 
dosage: sites will be codes as copy number of alternate allele [defined in function <code>gl.read.vcf</code>].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function uses parameters from <code>gl.read.vcf</code> for conversion
Note also that this function checks to see if there are input of mode, missing input of mode 
will issued the user with a error. "Dosage" mode of this function assign ploidy levels as maximum copy number of alternate alleles. 
Please carefully check the data if "dosage" mode is used. (codes were modified from
'vcfR2genlight' in vcfR packge to convert polyploid data)
</p>


<h3>Value</h3>

<p>genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Ching Ching Lau â€“ Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li>
<p> Knaus, B. J., &amp; Grunwald, N. J. (2017). 
vcfr: a package to manipulate and visualize variant call format data in R. 
Molecular ecology resources, 17(1), 44-53.
</p>
</li>
<li>
<p> Knaus, B. J., Grunwald, N. J., Anderson, E. C., Winter, D. J., 
Kamvar, Z. N., &amp; Tabima, J. F. (2023). Package 'vcfR'.
<a href="https://github.com/knausb/vcfR/blob/master/R/vcfR_conversion.R">vcfR</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Other utilities: 
<code>gl.alf()</code>,
<code>utils.check.datatype()</code>,
<code>utils.dart2genlight()</code>,
<code>utils.dist.binary()</code>,
<code>utils.flag.start()</code>,
<code>utils.hamming()</code>,
<code>utils.het.pop()</code>,
<code>utils.impute</code>,
<code>utils.is.fixed()</code>,
<code>utils.jackknife()</code>,
<code>utils.n.var.invariant()</code>,
<code>utils.plot.save()</code>,
<code>utils.read.fasta()</code>,
<code>utils.read.ped()</code>,
<code>utils.recalc.avgpic()</code>,
<code>utils.recalc.callrate()</code>,
<code>utils.recalc.freqhets()</code>,
<code>utils.recalc.freqhomref()</code>,
<code>utils.recalc.freqhomsnp()</code>,
<code>utils.recalc.maf()</code>,
<code>utils.reset.flags()</code>,
<code>utils.transpose()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
datatype &lt;- utils.vcfr2genlight.polyploid(x=vcfr, mode2="genotype")

## End(Not run)
</code></pre>


</div>