<div class="container">

<table style="width: 100%;"><tr>
<td>gl.plot.faststructure</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plots fastStructure analysis results (Q-matrix)</h2>

<h3>Description</h3>

<p>This function takes a fastStructure run object (output from
<code>gl.run.faststructure</code>) and plots the typical structure bar
plot that visualize the q matrix of a fastStructure run.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.plot.faststructure(
  sr,
  k.range,
  met_clumpp = "greedyLargeK",
  iter_clumpp = 100,
  clumpak = TRUE,
  plot_theme = NULL,
  colors_clusters = NULL,
  ind_name = TRUE,
  border_ind = 0.15,
  den = FALSE,
  x = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sr</code></td>
<td>
<p>fastStructure run object from <code>gl.run.faststructure</code>
[required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k.range</code></td>
<td>
<p>The number for K of the q matrix that should be plotted. Needs
to be within you simulated range of K's in your sr structure run object. If
NULL, all the K's are plotted [default NULL].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>met_clumpp</code></td>
<td>
<p>The algorithm to use to infer the correct permutations.
One of 'greedy' or 'greedyLargeK' or 'stephens' [default "greedyLargeK"].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iter_clumpp</code></td>
<td>
<p>The number of iterations to use if running either 'greedy'
'greedyLargeK' [default 100].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clumpak</code></td>
<td>
<p>Whether use the Clumpak method (see details) [default TRUE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default NULL].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colors_clusters</code></td>
<td>
<p>A color palette for clusters (K) or a list with
as many colors as there are clusters (K) [default NULL].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind_name</code></td>
<td>
<p>Whether to plot individual names [default TRUE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>border_ind</code></td>
<td>
<p>The width of the border line between individuals
[default 0.25].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>den</code></td>
<td>
<p>Whether to include a dendrogram. It is necessary to include the 
original genlight object used in gl.run.structure in the parameter x 
[default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>The original genlight object used in gl.run.structure description
[default NULL].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function outputs a barplot which is the typical output of
fastStructure.
</p>
<p>This function is based on the methods of CLUMPP and Clumpak as implemented
in the R package starmie (https://github.com/sa-lee/starmie).
</p>
<p>The Clumpak method identifies sets of highly similar runs among
all the replicates of the same K. The method then separates the distinct
groups of runs representing distinct modes in the space of possible solutions.
</p>
<p>The CLUMPP method permutes the clusters output by independent runs of
clustering programs such as structure, so that they match up as closely as
possible.
</p>
<p>This function averages the replicates within each mode identified by the
Clumpak method.
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li>
</ul>
<h3>Value</h3>

<p>List of Q-matrices
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber &amp; Luis Mijangos (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li>
<p> Raj, A., Stephens, M., &amp; Pritchard, J. K. (2014). fastSTRUCTURE:
variational inference of population structure in large SNP data sets.
Genetics, 197(2), 573-589.
</p>
</li>
<li>
<p> Pritchard, J.K., Stephens, M., Donnelly, P. (2000) Inference of
population structure using multilocus genotype data. Genetics 155, 945-959.
</p>
</li>
<li>
<p> Kopelman, Naama M., et al. "Clumpak: a program for identifying
clustering modes and packaging population structure inferences across K."
Molecular ecology resources 15.5 (2015): 1179-1191.
</p>
</li>
<li>
<p> Mattias Jakobsson and Noah A. Rosenberg. 2007. CLUMPP: a cluster
matching and permutation program for dealing with label switching and
multimodality in analysis of population structure. Bioinformatics
23(14):1801-1806. Available at
<a href="http://web.stanford.edu/group/rosenberglab/clumppDownload.html">clumpp</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p><code>gl.run.faststructure</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
t1 &lt;- gl.filter.callrate(platypus.gl, threshold = 1)
res &lt;- gl.run.faststructure(t1,
  exec = "./fastStructure", k.range = 2:3,
  num.k.rep = 2, output = paste0(getwd(), "/res_str")
)
qmat &lt;- gl.plot.faststructure(res, k.range = 2:3)
gl.map.structure(qmat, K = 2, t1, scalex = 1, scaley = 0.5)

## End(Not run)
</code></pre>


</div>