<div class="container">

<table style="width: 100%;"><tr>
<td>gl.read.fasta</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reads FASTA files and converts them to genlight object</h2>

<h3>Description</h3>

<p>The following IUPAC Ambiguity Codes are taken as heterozygotes:
</p>

<ul>
<li>
<p> M is heterozygote for	AC and CA	
</p>
</li>
<li>
<p> R is heterozygote	for AG and GA	
</p>
</li>
<li>
<p> W is heterozygote	for AT and TA	
</p>
</li>
<li>
<p> S is heterozygote	for CG and GC
</p>
</li>
<li>
<p> Y is heterozygote	for CT and TC
</p>
</li>
<li>
<p> K is heterozygote	for GT and TG
</p>
</li>
</ul>
<p>The following IUPAC Ambiguity Codes are taken as missing data:
</p>

<ul>
<li>
<p> V
</p>
</li>
<li>
<p> H 
</p>
</li>
<li>
<p> D 
</p>
</li>
<li>
<p> B
</p>
</li>
<li>
<p> N 
</p>
</li>
</ul>
<p>The function can deal with missing data in individuals, e.g. when FASTA 
files have different number of individuals due to missing data.
The allele with the highest frequency is taken as the reference allele.
SNPs with more than two alleles are skipped.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.read.fasta(fasta.files, parallel = FALSE, n.cores = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>fasta.files</code></td>
<td>
<p>Fasta files to read [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel</code></td>
<td>
<p>A logical indicating whether multiple cores -if available-
should be used for the computations (TRUE), or not (FALSE); requires the
package parallel to be installed [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.cores</code></td>
<td>
<p>If parallel is TRUE, the number of cores to be used in the
computations; if NULL, then the maximum number of cores available on the
computer is used [default NULL].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Ambiguity characters are often used to code heterozygotes. However, using
heterozygotes as ambiguity characters may bias many estimates. See more
information in the link below:
<a href="https://evodify.com/heterozygotes-ambiguity-characters/">https://evodify.com/heterozygotes-ambiguity-characters/</a>
</p>


<h3>Value</h3>

<p>A genlight object.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos â€“ Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other io: 
<code>gl.load()</code>,
<code>gl.read.csv()</code>,
<code>gl.read.dart()</code>,
<code>gl.read.silicodart()</code>,
<code>gl.save()</code>,
<code>gl.write.csv()</code>,
<code>utils.read.dart()</code>
</p>


</div>