<div class="container">

<table style="width: 100%;"><tr>
<td>gl.impute</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Imputates missing data</h2>

<h3>Description</h3>

<p>This function imputes genotypes on a population-by-population basis, where
populations can be considered panmictic, or imputes the state for
presence-absence data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.impute(
  x,
  method = "neighbour",
  fill.residual = TRUE,
  parallel = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence-absence
data [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Imputation method, either "frequency" or "HW" or "neighbour" 
or "random" [default "neighbour"].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fill.residual</code></td>
<td>
<p>Should any residual missing values remaining after 
imputation be set to 0, 1, 2 at random, taking into account global allele 
frequencies at the particular locus [default TRUE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel</code></td>
<td>
<p>A logical indicating whether multiple cores -if available-
should be used for the computations (TRUE), or not (FALSE); requires the
package parallel to be installed [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>We recommend that imputation be performed on sampling locations, before
any aggregation. Imputation is achieved by replacing missing values using
either of two methods:
</p>

<ul>
<li>
<p> If "frequency", genotypes scored as missing at a locus in an individual
are imputed using the average allele frequencies at that locus in the 
population from which the individual was drawn.
</p>
</li>
<li>
<p> If "HW", genotypes scored as missing at a locus in an individual are 
imputed by sampling at random assuming Hardy-Weinberg equilibrium. Applies 
only to genotype data.
</p>
</li>
<li>
<p> If "neighbour", substitute the missing values for the focal individual
with the values taken from the nearest neighbour. Repeat with next nearest
and so on until all missing values are replaced.
</p>
</li>
<li>
<p> if "random", missing data are substituted by random values (0, 1 or 2). 
</p>
</li>
</ul>
<p>The nearest neighbour is the one with the smallest Euclidean distance in 
all the dataset.
</p>
<p>The advantage of this approach is that it works regardless of how many
individuals are in the population to which the focal individual belongs,
and the displacement of the individual is haphazard as opposed to:
</p>
<p>(a) Drawing the individual toward the population centroid (HW and Frequency).
</p>
<p>(b) Drawing the individual toward the global centroid (glPCA).
</p>
<p>Note that loci that are missing for all individuals in a population are not 
imputed with method 'frequency' or 'HW'. Consider using the function 
<code>gl.filter.allna</code> with by.pop=TRUE to remove them first.
</p>


<h3>Value</h3>

<p>A genlight object with the missing data imputed.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos 
(Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class="language-R"> 
require("dartR.data")
# SNP genotype data
gl &lt;- gl.filter.callrate(platypus.gl,threshold=0.95)
gl &lt;- gl.filter.allna(gl)
gl &lt;- gl.impute(gl,method="neighbour")
# Sequence Tag presence-absence data
gs &lt;- gl.filter.callrate(testset.gs,threshold=0.95)
gl &lt;- gl.filter.allna(gl)
gs &lt;- gl.impute(gs, method="neighbour")

gs &lt;- gl.impute(platypus.gl,method ="random")
</code></pre>


</div>