<div class="container">

<table style="width: 100%;"><tr>
<td>gl.report.locmetric</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reports summary of the slot $other$loc.metrics</h2>

<h3>Description</h3>

<p>This script uses any field with numeric values stored in $other$loc.metrics
to produce summary statistics (mean, minimum, average, quantiles), histograms
and boxplots to assist the decision of choosing thresholds for the filter
function <code>gl.filter.locmetric</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.report.locmetric(
  x,
  metric,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>metric</code></td>
<td>
<p>Name of the metric to be used for filtering [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default two_colors].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function <code>gl.filter.locmetric</code> will filter out the
loci with a locmetric value below a specified threshold.
</p>
<p>The fields that are included in dartR, and a short description, are found
below. Optionally, the user can also set his/her own field by adding a vector
into $other$loc.metrics as shown in the example. You can check the names of
all available loc.metrics via: names(gl$other$loc.metrics).
</p>

<ul>
<li>
<p> SnpPosition - position (zero is position 1) in the sequence tag of the
defined SNP variant base.
</p>
</li>
<li>
<p> CallRate - proportion of samples for which the genotype call is
non-missing (that is, not '-' ).
</p>
</li>
<li>
<p> OneRatioRef - proportion of samples for which the genotype score is 0.
</p>
</li>
<li>
<p> OneRatioSnp - proportion of samples for which the genotype score is 2.
</p>
</li>
<li>
<p> FreqHomRef - proportion of samples homozygous for the Reference allele.
</p>
</li>
<li>
<p> FreqHomSnp - proportion of samples homozygous for the Alternate (SNP)
allele.
</p>
</li>
<li>
<p> FreqHets - proportion of samples which score as heterozygous, that is,
scored as 1.
</p>
</li>
<li>
<p> PICRef - polymorphism information content (PIC) for the Reference allele.
</p>
</li>
<li>
<p> PICSnp - polymorphism information content (PIC) for the SNP.
</p>
</li>
<li>
<p> AvgPIC - average of the polymorphism information content (PIC) of the
reference and SNP alleles.
</p>
</li>
<li>
<p> AvgCountRef - sum of the tag read counts for all samples, divided by the
number of samples with non-zero tag read counts, for the Reference allele row.
</p>
</li>
<li>
<p> AvgCountSnp - sum of the tag read counts for all samples, divided by the
number of samples with non-zero tag read counts, for the Alternate (SNP) allele
row.
</p>
</li>
<li>
<p> RepAvg - proportion of technical replicate assay pairs for which the
marker score is consistent.
</p>
</li>
<li>
<p> rdepth - read depth.
</p>
</li>
</ul>
<p><strong> Function's output </strong>
</p>
<p>The minimum, maximum, mean and a tabulation of quantiles of the locmetric
values against thresholds rate are provided. Output also includes a boxplot
and a histogram.
</p>
<p>Quantiles are partitions of a finite set of values into q subsets of (nearly)
equal sizes. In this function q = 20. Quantiles are useful measures because
they are less susceptible to long-tailed distributions and outliers.
</p>
<p>Plots and table were saved to the temporal directory (tempdir) and can be
accessed with the function <code>gl.print.reports</code> and listed with
the function <code>gl.list.reports</code>. Note that they can be accessed
only in the current R session because tempdir is cleared each time that the
R session is closed.
</p>
<p>Examples of other themes that can be used can be consulted in:
</p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li>
</ul>
<h3>Value</h3>

<p>An unaltered genlight object.
</p>


<h3>Author(s)</h3>

<p>Luis Mijangos (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code>gl.filter.locmetric</code>, <code>gl.list.reports</code>,
<code>gl.print.reports</code>
</p>
<p>Other report functions: 
<code>gl.report.bases()</code>,
<code>gl.report.callrate()</code>,
<code>gl.report.diversity()</code>,
<code>gl.report.hamming()</code>,
<code>gl.report.heterozygosity()</code>,
<code>gl.report.hwe()</code>,
<code>gl.report.ld.map()</code>,
<code>gl.report.maf()</code>,
<code>gl.report.monomorphs()</code>,
<code>gl.report.overshoot()</code>,
<code>gl.report.parent.offspring()</code>,
<code>gl.report.pa()</code>,
<code>gl.report.rdepth()</code>,
<code>gl.report.reproducibility()</code>,
<code>gl.report.secondaries()</code>,
<code>gl.report.sexlinked()</code>,
<code>gl.report.taglength()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># adding dummy data
test &lt;- testset.gl
test$other$loc.metrics$test &lt;- 1:nLoc(test)
# SNP data
out &lt;- gl.report.locmetric(test,metric='test')

# adding dummy data
test.gs &lt;- testset.gs
test.gs$other$loc.metrics$test &lt;- 1:nLoc(test.gs)
# Tag P/A data
out &lt;- gl.report.locmetric(test.gs,metric='test')

</code></pre>


</div>