<div class="container">

<table style="width: 100%;"><tr>
<td>gl.edit.recode.ind</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Creates or edits individual (=specimen) names, creates a recode_ind
file and applies the changes to a genlight object
@family data manipulation</h2>

<h3>Description</h3>

<p>A function to edit names of individual in a dartR genlight object, or to create a
reassignment table taking the individual labels from a genlight object, or to
edit existing individual labels in an existing recode_ind file. The amended 
recode table is then applied to the genlight object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.edit.recode.ind(
  x,
  out.recode.file = NULL,
  outpath = NULL,
  recalc = FALSE,
  mono.rm = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out.recode.file</code></td>
<td>
<p>Name of the file to output the new individual labels
[optional].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outpath</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>recalc</code></td>
<td>
<p>If TRUE, recalculate the locus metadata statistics [default TRUE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mono.rm</code></td>
<td>
<p>If TRUE, remove monomorphic loci [default TRUE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress but not results; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Renaming individuals may be required when there have been errors in labeling
arising in the passage of samples to sequencing. There may be occasions
where renaming individuals is required for preparation of figures. 
This function will input an existing recode table for editing and optionally
save it as a new table, or if the name of an input table is not supplied,
will generate a table using the individual labels in the parent genlight
object.
When caution needs to be exercised because of the potential for breaking the
'chain of evidence' associated with the samples, recoding individuals using
a recode table (csv) can provide a durable record of the changes.
For SNP genotype data, the function, having deleted individuals, optionally 
identifies resultant monomorphic loci or loci with all values missing 
and deletes them. The script also optionally recalculates the
locus metadata as appropriate. The optional deletion of monomorphic loci
and the optional recalculation of locus statistics is not available for
Tag P/A data (SilicoDArT).
Use outpath=getwd() when calling this function to direct
output files to your working directory.
The function returns a dartR genlight object with the new population assignments  
and the recalculated locus metadata.
</p>


<h3>Value</h3>

<p>An object of class ('genlight') with the revised individual labels.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges â€“ Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code>gl.recode.ind</code>, <code>gl.drop.ind</code>,
<code>gl.keep.ind</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#this is an interactive example
if(interactive()){
gl &lt;- gl.edit.recode.ind(testset.gl)
gl &lt;- gl.edit.recode.ind(testset.gl, out.recode.file='ind.recode.table.csv')
}

</code></pre>


</div>