<div class="container">

<table style="width: 100%;"><tr>
<td>gl2eigenstrat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Converts a genlight object into eigenstrat format</h2>

<h3>Description</h3>

<p>The output of this function are three files:
</p>

<ul>
<li>
<p> genotype file: contains genotype data for each individual at each SNP
with an extension 'eigenstratgeno.'
</p>
</li>
<li>
<p> snp file: contains information about each SNP with an extension 'snp.'
</p>
</li>
<li>
<p> indiv file: contains information about each individual with an
extension 'ind.'
</p>
</li>
</ul>
<h3>Usage</h3>

<pre><code class="language-R">gl2eigenstrat(
  x,
  outfile = "gl_eigenstrat",
  outpath = tempdir(),
  snp_pos = 1,
  snp_chr = 1,
  pos_cM = 0,
  sex_code = "unknown",
  phen_value = "Case",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfile</code></td>
<td>
<p>File name of the output file [default 'gl_eigenstrat'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp_pos</code></td>
<td>
<p>Field name from the slot loc.metrics where the SNP position is
stored [default 1].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp_chr</code></td>
<td>
<p>Field name from the slot loc.metrics where the chromosome of
each is stored [default 1].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos_cM</code></td>
<td>
<p>A vector, with as many elements as there are loci, containing
the SNP position in morgans or centimorgans [default 1].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sex_code</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the sex code ('male', 'female', 'unknown') [default 'unknown'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phen_value</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the phenotype value ('Case', 'Control') [default 'Case'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Eigenstrat only accepts chromosomes coded as numeric values, as follows:
X chromosome is encoded as 23, Y is encoded as 24, mtDNA is encoded as
90, and XY is encoded as 91. SNPs with illegal chromosome values, such
as 0, will be removed.
</p>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li>
<p> Patterson, N., Price, A. L., &amp; Reich, D. (2006). Population structure
and eigenanalysis. PLoS genetics, 2(12), e190.
</p>
</li>
<li>
<p> Price, A. L., Patterson, N. J., Plenge, R. M., Weinblatt, M. E.,
Shadick, N. A., &amp; Reich, D. (2006). Principal components analysis corrects
for stratification in genome-wide association studies. Nature genetics,
38(8), 904-909.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">
require("dartR.data")
gl2eigenstrat(platypus.gl,snp_pos='ChromPos_Platypus_Chrom_NCBIv1',
snp_chr = 'Chrom_Platypus_Chrom_NCBIv1')

</code></pre>


</div>