<div class="container">

<table style="width: 100%;"><tr>
<td>utils.dist.binary</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculates a distance matrix for individuals defined in a dartR
genlight object using binary P/A data (SilicoDArT)</h2>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class="language-R">utils.dist.binary(
  x,
  method = "simple",
  scale = FALSE,
  swap = FALSE,
  type = "dist",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight containing the genotypes [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Specify distance measure [default simple].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale</code></td>
<td>
<p>If TRUE and method='euclidean', the distance will be scaled to 
fall in the range [0,1] [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>swap</code></td>
<td>
<p>If TRUE and working with presence-absence data, then presence 
(no disrupting mutation) is scored as 0 and absence (presence of a disrupting 
mutation) is scored as 1 [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Specify the format and class of the object to be returned, 
dist for N11 object of class dist, matrix for an object of class matrix [default "dist"].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report [default 2].
</p>
<p>@details
This script calculates various distances between individuals based on sequence tag
Presence/Absence data.
</p>
<p>The distance measure can be one of:
</p>

<ul>
<li>
<p> Euclidean – Euclidean Distance applied to cartesian coordinates defined
by the loci, scored as 0 or 1. Presence and absence equally weighted.
</p>
</li>
<li>
<p> simple – simple matching, both 1 or both 0 = 0; one 1 and the other
0 = 1. Presence and absence equally weighted.
</p>
</li>
<li>
<p> Jaccard – ignores matching 0, both 1 = 0; one 1 and the other 0 = 1.
Absences could be for different reasons.
</p>
</li>
<li>
<p> Bray-Curtis – both 0 = 0; both 1 = 2; one 1 and the other 0 = 1. Absences
could be for different reasons. Sometimes called the Dice or Sorensen
distance.
</p>
</li>
</ul>
<p>One might choose to disregard or downweight absences in comparison with
presences because the homology of absences is less clear (mutation at one or
the other, or both restriction sites). Your call.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class 'dist' or 'matrix' giving distances between individuals
</p>


<h3>Author(s)</h3>

<p>Author: Arthur Georges. Custodian: Arthur Georges – Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other utilities: 
<code>gl.alf()</code>,
<code>utils.check.datatype()</code>,
<code>utils.dart2genlight()</code>,
<code>utils.flag.start()</code>,
<code>utils.hamming()</code>,
<code>utils.het.pop()</code>,
<code>utils.impute</code>,
<code>utils.is.fixed()</code>,
<code>utils.jackknife()</code>,
<code>utils.n.var.invariant()</code>,
<code>utils.plot.save()</code>,
<code>utils.read.fasta()</code>,
<code>utils.read.ped()</code>,
<code>utils.recalc.avgpic()</code>,
<code>utils.recalc.callrate()</code>,
<code>utils.recalc.freqhets()</code>,
<code>utils.recalc.freqhomref()</code>,
<code>utils.recalc.freqhomsnp()</code>,
<code>utils.recalc.maf()</code>,
<code>utils.reset.flags()</code>,
<code>utils.transpose()</code>,
<code>utils.vcfr2genlight.polyploid()</code>
</p>


</div>