<div class="container">

<table style="width: 100%;"><tr>
<td>gl.grm.network</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Represents a genomic relationship matrix (GRM) as a network</h2>

<h3>Description</h3>

<p>This script takes a G matrix generated by <code>gl.grm</code> and represents
the relationship among the specimens as a network diagram. In order to use
this script, a decision is required on a threshold for relatedness to be
represented as link in the network, and on the layout used to create the
diagram.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.grm.network(
  G,
  x,
  method = "fr",
  node.size = 8,
  node.label = TRUE,
  node.label.size = 2,
  node.label.color = "black",
  link.color = NULL,
  link.size = 2,
  relatedness_factor = 0.125,
  title = "Network based on a genomic relationship matrix",
  palette_discrete = gl.select.colors(x, library = "brewer", palette = "PuOr", ncolors =
    nPop(x), verbose = 0),
  plot.dir = NULL,
  plot.file = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>G</code></td>
<td>
<p>A genomic relationship matrix (GRM) generated by
<code>gl.grm</code> [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A genlight object from which the G matrix was generated [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>One of 'fr', 'kk', 'gh' or 'mds' [default 'fr'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>node.size</code></td>
<td>
<p>Size of the symbols for the network nodes [default 8].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>node.label</code></td>
<td>
<p>TRUE to display node labels [default TRUE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>node.label.size</code></td>
<td>
<p>Size of the node labels [default 3].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>node.label.color</code></td>
<td>
<p>Color of the text of the node labels
[default 'black'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>link.color</code></td>
<td>
<p>Colors for links [default gl.select.colors].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>link.size</code></td>
<td>
<p>Size of the links [default 2].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>relatedness_factor</code></td>
<td>
<p>Factor of relatedness [default 0.125].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>Title for the plot
[default 'Network based on genomic relationship matrix'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>palette_discrete</code></td>
<td>
<p>A discrete set of colors
with as many colors as there are populations in the dataset
[default NULL].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified
by the global working directory or tempdir()]</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The gl.grm.network function takes a genomic relationship matrix (GRM)
generated by the gl.grm function to represent the relationship among
individuals in the dataset as a network diagram. To generate the GRM, the
function gl.grm uses the function A.mat from package rrBLUP, which implements
the approach developed by Endelman and Jannink (2012).
</p>
<p>The GRM is an estimate of the proportion of alleles that two individuals have
in common. It is generated by estimating the covariance of the genotypes
between two individuals, i.e. how much genotypes in the two individuals
correspond with each other. This covariance depends on the probability that
alleles at a random locus are identical by state (IBS). Two alleles are
IBS if they represent the same allele. Two alleles are identical by
descent (IBD) if one is a physical copy of the other or if they are both
physical copies of the same ancestral allele. Note that IBD is complicated
to determine. IBD implies IBS, but not conversely. However, as the number
of SNPs in a dataset increases, the mean probability of IBS approaches
the mean probability of IBD.
</p>
<p>It follows that the off-diagonal elements of the GRM are two times the
kinship coefficient, i.e. the probability that two alleles at a random locus
drawn from two individuals are IBD. Additionally, the diagonal elements of
the GRM are 1+f, where f is the inbreeding coefficient of each individual,
i.e. the probability that the two alleles at a random locus are IBD.
</p>
<p>Choosing a meaningful threshold to represent the relationship between
individuals is tricky because IBD is not an absolute state but is relative to
a reference population for which there is generally little information so
that we can estimate the kinship of a pair of individuals only relative to
some other quantity. To deal with this, we can use the average inbreeding
coefficient of the diagonal elements as the reference value. For this, the
function subtracts 1 from the mean of the diagonal elements of the GRM. In a
second step, the off-diagonal elements are divided by 2, and finally, the
mean of the diagonal elements is subtracted from each off-diagonal element
after dividing them by 2. This approach is similar to the one used by
Goudet et al. (2018).
</p>
<p>Below is a table modified from Speed &amp; Balding (2015) showing kinship values,
and their confidence intervals (CI), for different relationships that could
be used to guide the choosing of the relatedness threshold in the function.
</p>
<p>|Relationship|Kinship|95
|Identical twins/clones/same individual     | 0.5     |        -         |
</p>
<p>|Sibling/Parent-Offspring                   | 0.25    |    (0.204, 0.296)|
</p>
<p>|Half-sibling                               | 0.125   |    (0.092, 0.158)|
</p>
<p>|First cousin                               | 0.062   |    (0.038, 0.089)|
</p>
<p>|Half-cousin                                | 0.031   |    (0.012, 0.055)|
</p>
<p>|Second cousin                              | 0.016   |    (0.004, 0.031)|
</p>
<p>|Half-second cousin                         | 0.008   |    (0.001, 0.020)|
</p>
<p>|Third cousin                               | 0.004   |    (0.000, 0.012)|
</p>
<p>|Unrelated                                  | 0       |        -         |
</p>
<p>Four layout options are implemented in this function:
</p>

<ul>
<li>
<p> 'fr' Fruchterman-Reingold layout  layout_with_fr
(package igraph)
</p>
</li>
<li>
<p> 'kk' Kamada-Kawai layout layout_with_kk (package igraph)
</p>
</li>
<li>
<p> 'gh' Graphopt layout layout_with_graphopt
(package igraph)
</p>
</li>
<li>
<p> 'mds' Multidimensional scaling layout layout_with_mds
(package igraph)
</p>
</li>
</ul>
<h3>Value</h3>

<p>A network plot showing relatedness between individuals
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges â€“ Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li>
<p> Endelman, J. B. , Jannink, J.-L. (2012). Shrinkage estimation of the
realized relationship matrix. G3: Genes, Genomics, Genetics 2, 1405.
</p>
</li>
<li>
<p> Goudet, J., Kay, T., &amp; Weir, B. S. (2018). How to estimate kinship.
Molecular Ecology, 27(20), 4121-4135.
</p>
</li>
<li>
<p> Speed, D., &amp; Balding, D. J. (2015). Relatedness in the post-genomic era:
is it still useful?. Nature Reviews Genetics, 16(1), 33-44.
</p>
</li>
</ul>
<h3>See Also</h3>

<p><code>gl.grm</code>
</p>
<p>Other inbreeding functions: 
<code>gl.grm()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">if (requireNamespace("igraph", quietly = TRUE) &amp; requireNamespace("rrBLUP",
  quietly = TRUE
) &amp; requireNamespace("fields", quietly = TRUE)) {
  t1 &lt;- possums.gl
  # filtering on call rate
  t1 &lt;- gl.filter.callrate(t1)
  t1 &lt;- gl.subsample.loc(t1, n = 100)
  # relatedness matrix
  res &lt;- gl.grm(t1, plotheatmap = FALSE)
  # relatedness network
  res2 &lt;- gl.grm.network(res, t1, relatedness_factor = 0.125)
}
</code></pre>


</div>