<div class="container">

<table style="width: 100%;"><tr>
<td>gl.ld.haplotype</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Visualize patterns of linkage disequilibrium and identification of 
haplotypes</h2>

<h3>Description</h3>

<p>This function plots a Linkage disequilibrium (LD) heatmap, where the colour 
shading indicates the strength of LD. Chromosome positions (Mbp) are shown on
the horizontal axis, and haplotypes appear as triangles and delimited by 
dark yellow vertical lines. Numbers identifying each haplotype are shown in 
the upper part of the plot. 
</p>
<p>The heatmap also shows heterozygosity for each SNP. 
</p>
<p>The function identifies haplotypes based on contiguous SNPs that are in 
linkage disequilibrium using as threshold <code>ld_threshold_haplo</code> and
containing more than <code>min_snps</code> SNPs.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.ld.haplotype(
  x,
  pop_name = NULL,
  chrom_name = NULL,
  ld_max_pairwise = 1e+07,
  maf = 0.05,
  ld_stat = "R.squared",
  ind.limit = 10,
  min_snps = 10,
  ld_threshold_haplo = 0.5,
  coordinates = NULL,
  color_haplo = "viridis",
  color_het = "deeppink",
  plot.out = TRUE,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop_name</code></td>
<td>
<p>Name of the population to analyse. If NULL all the 
populations are analised [default NULL].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrom_name</code></td>
<td>
<p>Nme of the chromosome to analyse. If NULL all the 
chromosomes are analised [default NULL].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ld_max_pairwise</code></td>
<td>
<p>Maximum distance in number of base pairs at which LD
should be calculated [default 10000000].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf</code></td>
<td>
<p>Minor allele frequency (by population) threshold to filter out 
loci. If a value &gt; 1 is provided it will be interpreted as MAC (i.e. the
minimum number of times an allele needs to be observed) [default 0.05].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ld_stat</code></td>
<td>
<p>The LD measure to be calculated: "LLR", "OR", "Q", "Covar",
"D.prime", "R.squared", and "R". See <code>ld</code>
(package snpStats) for details [default "R.squared"].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind.limit</code></td>
<td>
<p>Minimum number of individuals that a population should
contain to take it in account to report loci in LD [default 10].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_snps</code></td>
<td>
<p>Minimum number of SNPs that should have a haplotype to call 
it [default 10].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ld_threshold_haplo</code></td>
<td>
<p>Minimum LD between adjacent SNPs to call a 
haplotype [default 0.5].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coordinates</code></td>
<td>
<p>A vector of two elements with the start and end 
coordinates in base pairs to which restrict the 
analysis e.g. c(1,1000000) [default NULL].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color_haplo</code></td>
<td>
<p>Color palette for haplotype plot. See details
[default "viridis"].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color_het</code></td>
<td>
<p>Color for heterozygosity [default "deeppink"].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.out</code></td>
<td>
<p>Specify if heatmap plot is to be produced [default TRUE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The information for SNP's position should be stored in the genlight accessor
"@position" and the SNP's chromosome name in the accessor "@chromosome"
(see examples). The function will then calculate LD within each chromosome.
</p>
<p>The output of the function includes a table with the haplotypes
that were identified and their location.
</p>
<p>Colors of the heatmap (<code>color_haplo</code>) are based on the function
<code>scale_fill_viridis</code> from  package <code>viridis</code>. 
Other color palettes options are "magma", "inferno", "plasma", "viridis",
"cividis", "rocket", "mako" and "turbo".
</p>


<h3>Value</h3>

<p>A table with the haplotypes that were identified.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos â€“ Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other ld functions: 
<code>gl.ld.distance()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">require("dartR.data")
x &lt;- platypus.gl
x &lt;- gl.filter.callrate(x,threshold = 1)
x &lt;- gl.keep.pop(x, pop.list = "TENTERFIELD")
x$chromosome &lt;- as.factor(x$other$loc.metrics$Chrom_Platypus_Chrom_NCBIv1)
x$position &lt;- x$other$loc.metrics$ChromPos_Platypus_Chrom_NCBIv1
ld_res &lt;- gl.ld.haplotype(x,chrom_name = "NC_041728.1_chromosome_1",
                          ld_max_pairwise = 10000000 )
     
</code></pre>


</div>