<div class="container">

<table style="width: 100%;"><tr>
<td>gl.dist.pop</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculates a distance matrix for populations with SNP genotypes in a
genlight object</h2>

<h3>Description</h3>

<p>This script calculates various distances between populations based on allele
frequencies (SNP genotypes) or frequency of presences in presence-absence data 
(Euclidean and Fixed-diff distances only).
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.dist.pop(
  x,
  method = "euclidean",
  plot.out = TRUE,
  scale = FALSE,
  output = "dist",
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight containing the SNP genotypes [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Specify distance measure [default euclidean].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.out</code></td>
<td>
<p>If TRUE, display a histogram of the genetic distances, and a
whisker plot [default TRUE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale</code></td>
<td>
<p>If TRUE and method='Euclidean', the distance will be scaled to 
fall in the range [0,1] [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output</code></td>
<td>
<p>Specify the format and class of the object to be returned, 
dist for a object of class dist, matrix for an object of class matrix [default "dist"].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_colors</code></td>
<td>
<p>Vector with two color names for the borders and fill
[default two_colors].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The distance measure can be one of 'euclidean', 'fixed-diff', 'reynolds',
'nei' and 'chord'. Refer to the documentation of functions
described in the the dartR Distance Analysis tutorial for algorithms
and definitions.
</p>


<h3>Value</h3>

<p>An object of class 'dist' giving distances between populations
</p>


<h3>Author(s)</h3>

<p>author(s): Arthur Georges. Custodian: Arthur Georges â€“ Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R"> ## Not run: 
# SNP genotypes
D &lt;- gl.dist.pop(possums.gl[1:90,1:100], method='euclidean')
D &lt;- gl.dist.pop(possums.gl[1:90,1:100], method='euclidean',scale=TRUE)
#D &lt;- gl.dist.pop(possums.gl, method='nei')
#D &lt;- gl.dist.pop(possums.gl, method='reynolds')
#D &lt;- gl.dist.pop(possums.gl, method='chord')
#D &lt;- gl.dist.pop(possums.gl, method='fixed-diff')
#Presence-Absence data [only 10 individuals due to speed]
D &lt;- gl.dist.pop(testset.gs[1:10,], method='euclidean')

## End(Not run)
res &lt;- gl.dist.pop(platypus.gl)
</code></pre>


</div>