<div class="container">

<table style="width: 100%;"><tr>
<td>gl.grm</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculates an identity by descent matrix</h2>

<h3>Description</h3>

<p>This function calculates the mean probability of identity by state (IBS)
across loci that would result from all the possible crosses of the
individuals analyzed. IBD is calculated by an additive relationship matrix
approach developed by Endelman and Jannink (2012) as implemented in the
function A.mat (package rrBLUP).
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.grm(
  x,
  plotheatmap = TRUE,
  palette_discrete = NULL,
  palette_convergent = NULL,
  legendx = 0,
  legendy = 0.5,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plotheatmap</code></td>
<td>
<p>A switch if a heatmap should be shown [default TRUE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>palette_discrete</code></td>
<td>
<p>the color of populations [gl.select.colors].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>palette_convergent</code></td>
<td>
<p>A convergent palette for the IBD values
[default convergent_palette].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legendx</code></td>
<td>
<p>x coordinates for the legend[default 0].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legendy</code></td>
<td>
<p>y coordinates for the legend[default 1].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Parameters passed to function A.mat from package rrBLUP.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Two or more alleles are identical by descent (IBD) if they are identical
copies of the same ancestral allele in a base population. The additive
relationship matrix is a theoretical framework for estimating a relationship
matrix that is consistent with an approach to estimate the probability that
the alleles at a random locus are identical in state (IBS).
</p>
<p>This function also plots a heatmap, and a dendrogram, of IBD values where
each diagonal element has a mean that equals 1+f, where f is the inbreeding
coefficient (i.e. the probability that the two alleles at a randomly chosen
locus are IBD from the base population). As this probability lies between 0
and 1, the diagonal elements range from 1 to 2. Because the inbreeding
coefficients are expressed relative to the current population, the mean of
the off-diagonal elements is -(1+f)/n, where n is the number of loci.
Individual names are shown in the margins of the heatmap and colors
represent different populations.
</p>


<h3>Value</h3>

<p>An identity by descent matrix
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges â€“ Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li>
<p> Endelman, J. B. (2011). Ridge regression and other kernels for genomic
selection with r package rrblup. The Plant Genome 4, 250.
</p>
</li>
<li>
<p> Endelman, J. B. , Jannink, J.-L. (2012). Shrinkage estimation of the
realized relationship matrix. G3: Genes, Genomics, Genetics 2, 1405.
</p>
</li>
</ul>
<h3>See Also</h3>

<p><code>gl.grm.network</code>
</p>
<p>Other inbreeding functions: 
<code>gl.grm.network()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">gl.grm(platypus.gl[1:10, 1:100])

</code></pre>


</div>