<div class="container">

<table style="width: 100%;"><tr>
<td>gl.dist.pop</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculates a distance matrix for populations with SNP genotypes in a
genlight object</h2>

<h3>Description</h3>

<p>This script calculates various distances between populations based on allele
frequencies (SNP genotypes) or frequency of presences in PA (SilicoDArT) data
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.dist.pop(
  x,
  as.pop = NULL,
  method = "euclidean",
  scale = FALSE,
  type = "dist",
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight containing [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>as.pop</code></td>
<td>
<p>Temporarily assign another locus metric as the population for
the purposes of deletions [default NULL].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Specify distance measure [default euclidean].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale</code></td>
<td>
<p>If TRUE and method='Euclidean', the distance will be scaled to 
fall in the range [0,1] [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Specify the type of output, dist or matrix [default dist]</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.display</code></td>
<td>
<p>If TRUE, resultant plots are displayed in the plot window
[default TRUE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default c("#2171B5","#6BAED6")].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The distance measure can be one of 'euclidean', 'fixed-diff', 'reynolds',
'nei' and 'chord'. Refer to the documentation of functions in
https://doi.org/10.1101/2023.03.22.533737 for algorithms
and definitions.
</p>


<h3>Value</h3>

<p>An object of class 'dist' giving distances between populations
</p>


<h3>Author(s)</h3>

<p>author(s): Arthur Georges. Custodian: Arthur Georges â€“ Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other distance: 
<code>gl.dist.ind()</code>,
<code>gl.fdsim()</code>,
<code>utils.dist.ind.snp()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"> 
# SNP genotypes
D &lt;- gl.dist.pop(possums.gl, method='euclidean')
D &lt;- gl.dist.pop(possums.gl, method='euclidean',scale=TRUE)
D &lt;- gl.dist.pop(possums.gl, method='nei')
D &lt;- gl.dist.pop(possums.gl, method='reynolds')
D &lt;- gl.dist.pop(possums.gl, method='chord')
D &lt;- gl.dist.pop(possums.gl, method='fixed-diff')
#Presence-Absence data [only 10 individuals due to speed]
D &lt;- gl.dist.pop(testset.gs[1:10,], method='euclidean')


</code></pre>


</div>