<div class="container">

<table style="width: 100%;"><tr>
<td>gl2svdquartets</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Converts a genlight object to nexus format PAUP SVDquartets</h2>

<h3>Description</h3>

<p>The output nexus file contains the SNP data in one of two forms, depending
upon what you regard as most appropriate. One form, that used by Chifman and
Kubatko, has two lines per individual, one providing the reference SNP the
second providing the alternate SNP (method=1).
A second form, recommended by Dave Swofford, has a single line per
individual, resolving heterozygous SNPs by replacing them with standard
ambiguity codes (method=2).
If the data are tag presence/absence, then method=2 is assumed.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl2svdquartets(
  x,
  outfile = "svd.nex",
  outpath = NULL,
  method = 2,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight object containing the SNP data or tag P/A data
[required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfile</code></td>
<td>
<p>File name of the output file (including extension)
[default 'svd.nex'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outpath</code></td>
<td>
<p>Path where to save the output file [default global working 
directory or if not specified, tempdir()].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Method = 1, nexus file with two lines per individual; method =
2, nexus file with one line per individual, ambiguity codes for SNP
genotypes, 0 or 1 for presence/absence data [default 2].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Chifman, J. and L. Kubatko. 2014. Quartet inference from SNP data
under the coalescent. Bioinformatics 30: 3317-3324
</p>


<h3>Examples</h3>

<pre><code class="language-R">gg &lt;- testset.gl[1:20,1:100]
gg@other$loc.metrics &lt;- gg@other$loc.metrics[1:100,]
gl2svdquartets(gg, outpath=tempdir())

</code></pre>


</div>