<div class="container">

<table style="width: 100%;"><tr>
<td>gl.make.recode.ind</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Creates a proforma recode_ind file for reassigning individual
(=specimen) names</h2>

<h3>Description</h3>

<p>Renaming individuals may be required when there have been errors in labeling
arising in the process from sample to sequencing files. There may be occasions
where renaming individuals is required for preparation of figures.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.make.recode.ind(
  x,
  out.recode.file = "default_recode_ind.csv",
  outpath = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out.recode.file</code></td>
<td>
<p>File name of the output file (including extension)
[default default_recode_ind.csv].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outpath</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2, 
progress log; 3, progress and results summary; 5, full report 
[default 2 or as specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function facilitates the construction of a recode table by producing a
proforma file with current individual (=specimen) names in two identical
columns. Edit the second column to reassign individual names. Use keyword
'Delete' to delete an individual.
When caution needs to be exercised because of the potential for breaking the
'chain of evidence' associated with the samples, recoding individuals using
a recode table (csv) can provide a clear record of the changes.
Use outpath=getwd() or when calling this function to direct output files 
to your working directory.
The function works with both genlight objects
containing SNP genotypes and Tag P/A data (SilicoDArT).
Apply the recoding using gl.recode.ind().
</p>


<h3>Value</h3>

<p>A vector containing the new individual names.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges â€“ Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other data manipulation: 
<code>gl.define.pop()</code>,
<code>gl.drop.ind()</code>,
<code>gl.drop.loc()</code>,
<code>gl.drop.pop()</code>,
<code>gl.edit.recode.pop()</code>,
<code>gl.impute()</code>,
<code>gl.join()</code>,
<code>gl.keep.ind()</code>,
<code>gl.keep.loc()</code>,
<code>gl.keep.pop()</code>,
<code>gl.merge.pop()</code>,
<code>gl.reassign.pop()</code>,
<code>gl.recode.ind()</code>,
<code>gl.recode.pop()</code>,
<code>gl.rename.pop()</code>,
<code>gl.sample()</code>,
<code>gl.sim.genotypes()</code>,
<code>gl.sort()</code>,
<code>gl.subsample.ind()</code>,
<code>gl.subsample.loc()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">result &lt;- gl.make.recode.ind(testset.gl, out.recode.file ='Emmac_recode_ind.csv',outpath=tempdir())
</code></pre>


</div>