<div class="container">

<table style="width: 100%;"><tr>
<td>gl.report.heterozygosity</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reports observed, expected and unbiased heterozygosities and FIS
(inbreeding coefficient) by population or by individual from SNP data</h2>

<h3>Description</h3>

<p>Calculates the observed, expected and unbiased expected (i.e.
corrected for sample size) heterozygosities and FIS (inbreeding coefficient)
for each population or the observed heterozygosity for each individual in a
genlight object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gl.report.heterozygosity(
  x,
  method = "pop",
  n.invariant = 0,
  subsample.pop = FALSE,
  n.limit = 10,
  nboots = 0,
  boot.method = "ind",
  conf = 0.95,
  CI.type = "bca",
  ncpus = 1,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors.pop = gl.colors("dis"),
  plot.colors.ind = gl.colors(2),
  plot.file = NULL,
  plot.dir = NULL,
  error.bar = "SD",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Calculate heterozygosity by population (method='pop') or by
individual (method='ind') [default 'pop'].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.invariant</code></td>
<td>
<p>An estimate of the number of invariant sequence tags used
to adjust the heterozygosity rate [default 0].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subsample.pop</code></td>
<td>
<p>Whether subsample populations to estimate observed 
heterozygosity (see Details) [default FALSE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.limit</code></td>
<td>
<p>Minimum number of individuals that should have a population to 
perform subsampling to estimate heterozygosity [default 10].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nboots</code></td>
<td>
<p>Number of bootstrap replicates to obtain confidence intervals
[default 0].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>boot.method</code></td>
<td>
<p>boostraping across individuals ("ind") or across loci
("loc") [default "ind"].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>conf</code></td>
<td>
<p>The confidence level of the required interval [default 0.95].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CI.type</code></td>
<td>
<p>Method to estimate confidence intervals. One of
"norm", "basic", "perc" or "bca" [default "bca"].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncpus</code></td>
<td>
<p>Number of processes to be used in parallel operation. If ncpus
&gt; 1 parallel operation is activated, see "Details" section [default 1].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.display</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.colors.pop</code></td>
<td>
<p>A color palette for population plots or a list with
as many colors as there are populations in the dataset
[default gl.colors("dis")].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.colors.ind</code></td>
<td>
<p>List of two color names for the borders and fill of
the plot by individual [default gl.colors(2)].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, 
exclude extension) [default NULL].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>error.bar</code></td>
<td>
<p>Statistic to be plotted as error bar either "SD" (standard 
deviation) or "SE" (standard error) or "CI" (confidence intervals)
[default "SD"].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Observed heterozygosity for a population takes the proportion of
heterozygous loci for each individual and averages it over all individuals in
that population. The calculations take into account missing values.
</p>
<p>Expected heterozygosity for a population takes the expected proportion of
heterozygotes, that is, expected under Hardy-Weinberg equilibrium, for each
locus, then averages this across the loci for an average estimate for the
population.
</p>
<p>The unbiased expected heterozygosity is calculated using the correction for 
sample size following equation 2 from Nei 1978.
</p>
<p>Accuracy of all heterozygosity estimates is affected by small sample sizes,
and so is their comparison between populations or repeated analysis. Expected
heterozygosities are less affected because their calculations are based on 
allele frequencies while observed heterozygosities are strongly susceptible 
to sampling effects when the sample size is small.  
</p>
<p>Observed heterozygosity for individuals is calculated as the proportion of
loci that are heterozygous for that individual.
</p>
<p>Finally, the loci that are invariant across all individuals in the dataset
(that is, across populations), is typically unknown. This can render
estimates of heterozygosity analysis specific, and so it is not valid to
compare such estimates across species or even across different analyses 
(see Schimdt et al 2021). This is a similar problem faced by microsatellites. 
If you have an estimate of the
number of invariant sequence tags (loci) in your data, such as provided by
<code>gl.report.secondaries</code>, you can specify it with the n.invariant
parameter to standardize your estimates of heterozygosity. This is called
autosomal heterozygosities by Schimddt et al (2021).
</p>
<p><strong>NOTE</strong>: It is important to realise that estimation of adjusted (autosomal)
heterozygosity requires that secondaries not to be removed.
</p>
<p>Heterozygosities and FIS (inbreeding coefficient) are calculated by locus
within each population using the following equations, and then averaged across 
all loci:
</p>

<ul>
<li>
<p> Observed heterozygosity (Ho) = number of heterozygotes / n_Ind,
where n_Ind is the number of individuals without missing data for that locus.
</p>
</li>
<li>
<p> Observed heterozygosity adjusted (Ho.adj) &lt;- Ho * n_Loc /
(n_Loc + n.invariant),
where n_Loc is the number of loci that do not have all missing data  and
n.invariant is an estimate of the number of invariant loci to adjust
heterozygosity.
</p>
</li>
<li>
<p> Expected heterozygosity (He) = 1 - (p^2 + q^2),
where p is the frequency of the reference allele and q is the frequency of
the alternative allele.
</p>
</li>
<li>
<p> Expected heterozygosity adjusted (He.adj) = He * n_Loc /
(n_Loc + n.invariant)
</p>
</li>
<li>
<p> Unbiased expected heterozygosity (uHe) = He * (2 * n_Ind /
(2 * n_Ind - 1))
</p>
</li>
<li>
<p> Inbreeding coefficient (FIS) = 1 - Ho / uHe
</p>
</li>
</ul>
<p><strong> Function's output </strong>
</p>
<p>Output for method='pop' is an ordered barchart of observed heterozygosity,
unbiased expected heterozygosity and FIS (Inbreeding coefficient) across 
populations together with a table of mean observed and expected 
heterozygosities and FIS by population and their respective standard 
deviations (SD).
In the output, it is also reported by population: the number of loci used to
estimate heterozygosity (n.Loc), the number of polymorphic loci (polyLoc),
the number of monomorphic loci (monoLoc) and loci with all missing data
(all_NALoc).
</p>
<p>Output for method='ind' is a histogram and a boxplot of heterozygosity across
individuals.
</p>
<p>If a plot.file is given, the ggplot arising from this function is saved as an 
"RDS" binary file using saveRDS(); can be reloaded with readRDS(). A file 
name must be specified for the plot to be saved.
</p>
<p>If a plot directory (plot.dir) is specified, the ggplot binary is saved to 
that directory; otherwise to the tempdir(). 
</p>
<p>Examples of other themes that can be used can be consulted in: 
</p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li>
</ul>
<p><strong>Subsampling populations</strong>
</p>
<p>To test the effect of five population sample sizes (n = 10, 5, 4, 3, 2) on 
observed heterozygosity estimates, the function subsamples individuals,
without replacement. The subsampling is repeated 10 times for each sample
size n. This approach is an implementation of Schmidt et al (2021). 
</p>
<p><strong>Error bars</strong>
</p>
<p>The best method for presenting or assessing genetic statistics depends on 
the type of data you have and the specific questions you're trying to 
answer. Here's a brief overview of when you might use each method:
</p>
<p><strong>1. Confidence Intervals ("CI"):</strong>
</p>
<p>- Usage: Often used to convey the precision of an estimate.
</p>
<p>- Advantage: Confidence intervals give a range in which the true parameter 
(like a population mean) is likely to fall, given the data and a specified 
probability (like 95%).
</p>
<p>- In Context: For genetic statistics, if you're estimating a parameter,
a 95% CI gives you a range in which you're 95% confident the true parameter
lies.
</p>
<p><strong>2. Standard Deviation ("SD"):</strong>
</p>
<p>- Usage: Describes the amount of variation from the average in a set of data.
</p>
<p>- Advantage: Allows for an understanding of the spread of individual data
points around the mean.
</p>
<p>- In Context: If you're looking at the distribution of a quantitative trait 
(like height) in a population with a particular genotype, the SD can 
describe how much individual heights vary around the average height.
</p>
<p><strong>3. Standard Error ("SE"):</strong>
</p>
<p>- Usage: Describes the precision of the sample mean as an estimate of the 
population mean.
</p>
<p>- Advantage: Smaller than the SD in large samples; it takes into account 
both the SD and the sample size. 
</p>
<p>- In Context: If you want to know how accurately your sample mean represents
the population mean, you'd look at the SE.
</p>
<p><strong>Recommendation:</strong>
</p>
<p>- If you're trying to convey the precision of an estimate, confidence 
intervals are very useful.
</p>
<p>- For understanding variability within a sample, standard deviation is key.
</p>
<p>- To see how well a sample mean might estimate a population mean, consider 
the standard error.
</p>
<p>In practice, geneticists often use a combination of these methods to 
analyze and present their data, depending on their research questions and 
the nature of the data.
</p>
<p><strong>Confidence Intervals</strong>
</p>
<p>The uncertainty of a parameter, in this case the mean of the statistic, can
be summarised by a confidence interval (CI) which includes the true parameter
value with a specified probability (i.e. confidence level; the parameter
"conf" in this function).
</p>
<p>In this function, CI are obtained using Bootstrap which is an inference
method that samples with replacement the data (i.e. loci) and calculates the
statistics every time.
</p>
<p>This function uses the function boot (package boot) to perform
the bootstrap replicates and the function boot.ci
(package boot) to perform the calculations for the CI.
</p>
<p>Four different types of nonparametric CI can be calculated
(parameter "CI.type" in this function):
</p>

<ul>
<li>
<p> First order normal approximation interval ("norm").
</p>
</li>
<li>
<p> Basic bootstrap interval ("basic").
</p>
</li>
<li>
<p> Bootstrap percentile interval ("perc").
</p>
</li>
<li>
<p> Adjusted bootstrap percentile interval ("bca").
</p>
</li>
</ul>
<p>The studentized bootstrap interval ("stud") was not included in the CI types
because it is computationally intensive, it may produce estimates outside
the range of plausible values and it has been found to be erratic in
practice, see for example the "Studentized (t) Intervals" section in:
</p>
<p><a href="https://www.r-bloggers.com/2019/09/understanding-bootstrap-confidence-interval-output-from-the-r-boot-package/">https://www.r-bloggers.com/2019/09/understanding-bootstrap-confidence-interval-output-from-the-r-boot-package/</a>
</p>
<p>Nice tutorials about the different types of CI can be found in:
</p>
<p><a href="https://www.datacamp.com/tutorial/bootstrap-r">https://www.datacamp.com/tutorial/bootstrap-r</a>
</p>
<p>and
</p>
<p><a href="https://www.r-bloggers.com/2019/09/understanding-bootstrap-confidence-interval-output-from-the-r-boot-package/">https://www.r-bloggers.com/2019/09/understanding-bootstrap-confidence-interval-output-from-the-r-boot-package/</a>
</p>
<p>Efron and Tibshirani (1993, p. 162) and Davison and Hinkley
(1997, p. 194) suggest that the number of bootstrap replicates should
be between 1000 and 2000.
</p>
<p><strong>It is important</strong> to note that unreliable confidence intervals will be
obtained if too few number of bootstrap replicates are used.
Therefore, the function boot.ci will throw warnings and errors
if bootstrap replicates are too few. Consider increasing the number of
bootstrap replicates to at least 200.
</p>
<p>The "bca" interval is often cited as the best for theoretical reasons,
however it may produce unstable results if the bootstrap distribution
is skewed or has extreme values. For example, you might get the warning
"extreme order statistics used as endpoints" or the error "estimated
adjustment 'a' is NA". In this case, you may want to use more bootstrap
replicates or a different method or check your data for outliers.
</p>
<p>The error "estimated adjustment 'w' is infinite" means that the estimated
adjustment ‘w’ for the "bca" interval is infinite, which can happen when
the empirical influence values are zero or very close to zero. This can
be caused by various reasons, such as:
</p>
<p>The number of bootstrap replicates is too small, the statistic of interest
is constant or nearly constant across the bootstrap samples, the data
contains outliers or extreme values.
</p>
<p>You can try some possible solutions, such as:
</p>
<p>Increasing the number of bootstrap replicates, using a different type of
bootstrap confidence interval or removing or transforming the outliers or
extreme values.
</p>
<p><strong>Parallelisation</strong>
</p>
<p>If the parameter ncpus &gt; 1, parallelisation is enabled. In Windows, parallel
computing employs a "socket" approach that starts new copies of R on each
core. POSIX systems, on the other hand (Mac, Linux, Unix, and BSD),
utilise a "forking" approach that replicates the whole current version of
R and transfers it to a new core.
</p>
<p>Opening and terminating R sessions in each core involves a significant
amount of processing time, therefore parallelisation in Windows machines
is only quicker than not using parallelisation when nboots &gt; 1000-2000.
</p>


<h3>Value</h3>

<p>A dataframe containing population labels, heterozygosities, FIS,
their standard deviations and sample sizes.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li>
<p> Nei, M. (1978). Estimation of average heterozygosity and genetic 
distance from a small number of individuals. Genetics, 89(3), 583-590.
</p>
</li>
<li>
<p> Schmidt, T. L., Jasper, M. E., Weeks, A. R., &amp; Hoffmann, A. A. (2021).
Unbiased population heterozygosity estimates from genome‐wide sequence data.
Methods in Ecology and Evolution, 12(10), 1888-1898.
</p>
</li>
</ul>
<h3>See Also</h3>

<p><code>gl.filter.heterozygosity</code>
</p>
<p>Other unmatched report: 
<code>gl.allele.freq()</code>,
<code>gl.report.basics()</code>,
<code>gl.report.diversity()</code>,
<code>gl.report.polyploid_heterozygosity()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"> 
require("dartR.data")
df &lt;- gl.report.heterozygosity(platypus.gl)
df &lt;- gl.report.heterozygosity(platypus.gl,method='ind')
n.inv &lt;- gl.report.secondaries(platypus.gl)
gl.report.heterozygosity(platypus.gl, n.invariant = n.inv[7, 2])
gl.report.heterozygosity(platypus.gl, subsample.pop = TRUE)

df &lt;- gl.report.heterozygosity(platypus.gl)
</code></pre>


</div>