<div class="container">

<table style="width: 100%;"><tr>
<td>utils.outflank.MakeDiploidFSTMat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Creates OutFLANK input file from individual genotype info.</h2>

<h3>Description</h3>

<p>Creates OutFLANK input file from individual genotype info.
</p>


<h3>Usage</h3>

<pre><code class="language-R">utils.outflank.MakeDiploidFSTMat(SNPmat, locusNames, popNames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>SNPmat</code></td>
<td>
<p>This is an array of genotypes with a row for each individual.
There should be a column for each SNP, with the number of copies of the focal
allele (0, 1, or 2) for that individual. If that individual is missing data
for that SNP, there should be a 9, instead.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locusNames</code></td>
<td>
<p>A list of names for each SNP locus. There should be the
same number of locus names as there are columns in SNPmat.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>popNames</code></td>
<td>
<p>A list of population names to give location for each
individual. Typically multiple individuals will have the same popName. The
list popNames should have the same length as the number of rows in SNPmat.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a data frame in the form needed for the main OutFLANK
function.
</p>


</div>