<div class="container">

<table style="width: 100%;"><tr>
<td>get_bold_data</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get genetic data from the Barcode of Life Database (BOLD) for a set of taxon names.</h2>

<h3>Description</h3>

<p><code>get_bold_data</code> uses taxon names from a tree topology, a character
vector of names or a <code>datelifeQuery</code> object, to search for genetic markers
in the Barcode of Life Database (BOLD).
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_bold_data(
  input = c("Rhea americana", "Struthio camelus", "Gallus gallus"),
  marker = "COI",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>input</code></td>
<td>
<p>One of the following:
</p>

<dl>
<dt>A character vector</dt>
<dd>
<p>With taxon names as a single comma separated
starting or concatenated with <code>c()</code>.</p>
</dd>
<dt>A phylogenetic tree with taxon names as tip labels</dt>
<dd>
<p>As a <code>phylo</code> or
<code>multiPhylo</code> object, OR as a newick character string.</p>
</dd>
<dt>A <code>datelifeQuery</code> object</dt>
<dd>
<p>An output from <code>make_datelife_query()</code>.</p>
</dd>
</dl>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker</code></td>
<td>
<p>A character vector indicating the gene from BOLD system to be
used for branch length estimation. It searches "COI" marker by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Arguments passed on to <code>get_otol_synthetic_tree</code>
</p>

<dl>
<dt><code>otol_version</code></dt>
<dd>
<p>Version of Open Tree of Life to use</p>
</dd>
<dt><code>resolve</code></dt>
<dd>
<p>Defaults to <code>TRUE</code>. Whether to resolve the tree at random or not.</p>
</dd>
<dt><code>ott_ids</code></dt>
<dd>
<p>If not NULL, it takes this argument and ignores input. A
numeric vector of ott ids obtained with <code>rotl::taxonomy_taxon_info()</code> or
<code>rotl::tnrs_match_names()</code> or <code>tnrs_match()</code>.</p>
</dd>
</dl>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If <code>input</code> is a <code>phylo</code> object or a newick string, it is used as backbone topology.
If <code>input</code> is a character vector of taxon names, an induced synthetic OpenTree
subtree is used as backbone.
</p>


<h3>Value</h3>

<p>A <code>phylo</code> object. If there are enough BOLD sequences available for the
<code>input</code> taxon names, the function returns a tree with branch lengths proportional
to relative substitution rate. If not enough BOLD sequences are available
for the <code>input</code> taxon names, the function returns the topology given as
<code>input</code>, or a synthetic Open Tree of Life for the taxon names given in
<code>input</code>, obtained with <code>get_otol_synthetic_tree()</code>.
</p>


</div>