<div class="container">

<table style="width: 100%;"><tr>
<td>make_bold_otol_tree</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Use genetic data from the Barcode of Life Database (BOLD) to reconstruct branch lengths on a tree.</h2>

<h3>Description</h3>

<p><code>make_bold_otol_tree</code> takes taxon names from a tree topology or
a vector of names to search for genetic markers in the Barcode of Life Database
(BOLD), create an alignment, and reconstruct branch lengths on a tree topology
with Maximum Likelihood.
</p>


<h3>Usage</h3>

<pre><code class="language-R">make_bold_otol_tree(
  input = c("Rhea americana", "Struthio camelus", "Gallus gallus"),
  marker = "COI",
  otol_version = "v3",
  chronogram = TRUE,
  doML = FALSE,
  aligner = "muscle",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>input</code></td>
<td>
<p>One of the following:
</p>

<dl>
<dt>A character vector</dt>
<dd>
<p>With taxon names as a single comma separated
starting or concatenated with <code>c()</code>.</p>
</dd>
<dt>A phylogenetic tree with taxon names as tip labels</dt>
<dd>
<p>As a <code>phylo</code> or
<code>multiPhylo</code> object, OR as a newick character string.</p>
</dd>
<dt>A <code>datelifeQuery</code> object</dt>
<dd>
<p>An output from <code>make_datelife_query()</code>.</p>
</dd>
</dl>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker</code></td>
<td>
<p>A character vector indicating the gene from BOLD system to be
used for branch length estimation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>otol_version</code></td>
<td>
<p>Version of Open Tree of Life to use</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chronogram</code></td>
<td>
<p>Default to <code>TRUE</code>, branch lengths returned are estimated with
<code>ape::chronoMPL()</code>. If <code>FALSE</code>, branch lengths returned are estimated with
<code>phangorn::acctran()</code> and represent relative substitution rates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>doML</code></td>
<td>
<p>Default to <code>FALSE</code>. If <code>TRUE</code>, it does a ML branch length optimization
with <code>phangorn::optim.pml()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aligner</code></td>
<td>
<p>A character vector indicating whether to use MAFFT or MUSCLE
to align BOLD sequences. It is not case sensitive. Default to MUSCLE,
supported using the <a href="https://bioconductor.org/packages/release/bioc/html/msa.html">msa</a>
package from Bioconductor, which needs to be installed using <code>BiocManager::install()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Arguments passed on to <code>get_otol_synthetic_tree</code>
</p>

<dl>
<dt><code>resolve</code></dt>
<dd>
<p>Defaults to <code>TRUE</code>. Whether to resolve the tree at random or not.</p>
</dd>
<dt><code>ott_ids</code></dt>
<dd>
<p>If not NULL, it takes this argument and ignores input. A
numeric vector of ott ids obtained with <code>rotl::taxonomy_taxon_info()</code> or
<code>rotl::tnrs_match_names()</code> or <code>tnrs_match()</code>.</p>
</dd>
</dl>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If <code>input</code> is a <code>phylo</code> object or a newick string, it is used as backbone topology.
If <code>input</code> is a character vector of taxon names, an induced synthetic OpenTree
subtree is used as backbone.
</p>


<h3>Value</h3>

<p>A <code>phylo</code> object. If there are enough BOLD sequences available for the
<code>input</code> taxon names, the function returns a tree with branch lengths proportional
to relative substitution rate. If not enough BOLD sequences are available
for the <code>input</code> taxon names, the function returns the topology given as
<code>input</code>, or a synthetic Open Tree of Life for the taxon names given in
<code>input</code>, obtained with <code>get_otol_synthetic_tree()</code>.
</p>


</div>