<div class="container">

<table style="width: 100%;"><tr>
<td>datelife_use</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate one or multiple chronograms for a set of given taxon names.</h2>

<h3>Description</h3>

<p><code>datelife_use</code> gets secondary calibrations available for any
pair of given taxon names, mined from the opentree_chronograms object,
congruifies them, and uses them to date a given tree topology with the
algorithm defined in <code>dating_method</code>. If no tree topology is provided,
it will attempt to get one for the given taxon names from Open Tree of Life
synthetic tree, using <code>make_bold_otol_tree()</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">datelife_use(input = NULL, each = FALSE, dating_method = "bladj", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>input</code></td>
<td>
<p>One of the following:
</p>

<dl>
<dt>A character vector</dt>
<dd>
<p>With taxon names as a single comma separated
starting or concatenated with <code>c()</code>.</p>
</dd>
<dt>A phylogenetic tree with taxon names as tip labels</dt>
<dd>
<p>As a <code>phylo</code> or
<code>multiPhylo</code> object, OR as a newick character string.</p>
</dd>
<dt>A <code>datelifeQuery</code> object</dt>
<dd>
<p>An output from <code>make_datelife_query()</code>.</p>
</dd>
</dl>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>each</code></td>
<td>
<p>Boolean, default to <code>FALSE</code>: all calibrations are returned in
the same <code>data.frame</code>. If <code>TRUE</code>, calibrations from each chronogram are returned
in separate data frames.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dating_method</code></td>
<td>
<p>Tree dating algorithm to use. Options are "bladj" or "pathd8"
(Webb et al., 2008, <a href="https://doi.org/10.1093/bioinformatics/btn358">doi:10.1093/bioinformatics/btn358</a>; Britton et al., 2007,
<a href="https://doi.org/10.1080/10635150701613783">doi:10.1080/10635150701613783</a>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Arguments passed on to <code>make_datelife_query</code>
</p>

<dl>
<dt><code>use_tnrs</code></dt>
<dd>
<p>Whether to use Open Tree of Life's Taxonomic Name Resolution Service (TNRS)
to process input taxon names. Default to <code>TRUE</code>, it corrects misspellings and
taxonomic name variations with <code>tnrs_match()</code>, a wrapper of <code>rotl::tnrs_match_names()</code>.</p>
</dd>
<dt><code>get_spp_from_taxon</code></dt>
<dd>
<p>Whether to search ages for all species belonging to a
given taxon or not. Default to <code>FALSE</code>. If <code>TRUE</code>, it must have same length as input.
If input is a newick string with some clades it will be converted to a <code>phylo</code>
object, and the order of <code>get_spp_from_taxon</code> will match <code>phy$tip.label</code>.</p>
</dd>
<dt><code>reference_taxonomy</code></dt>
<dd>
<p>A character vector specifying the reference taxonomy to use for TNRS.
Options are "ott", "ncbi", "gbif" or "irmng". The function defaults to "ott".</p>
</dd>
</dl>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If <code>input</code> is a vector of taxon names, the function will attempt to reconstruct a BOLD
tree with <code>make_bold_otol_tree()</code> to get a tree with branch lengths. If it fails,
it will get an Open Tree of Life synthetic tree topology.
The function then calls <code>use_calibrations()</code>.
</p>


<h3>Value</h3>

<p>A <code>phylo</code> or <code>multiPhylo</code> object with branch lengths proportional to time.
</p>


<h3>More</h3>

<p>The output object stores the used <code>calibrations</code> and <code>dating_method</code> as
<code>attributes(output)$datelife_calibrations</code> and <code>attributes(output)$dating_method</code>.
</p>


</div>