<div class="container">

<table style="width: 100%;"><tr>
<td>dada_implement</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Dada Implement</h2>

<h3>Description</h3>

<p>This function requires a main directory containing a folder(s) representing
sequencing runs which in-turn contain fastq files (the location of one of the
fastq files in one of the sequencing run folders is used as an input
argument). All sequencing folders in the main directory need to represent
data from sequencing runs that have used the same primers and protocols.
Output from this function includes all processing files and final main output
files in the form of fasta files and amplicon sequencing variant (ASV)
tables.
</p>


<h3>Usage</h3>

<pre><code class="language-R">dada_implement(
  runFolderLoc = NULL,
  primerFile = NULL,
  fwdIdent = "_R1_001",
  revIdent = "_R2_001",
  unidirectional = FALSE,
  bidirectional = TRUE,
  printQualityPdf = TRUE,
  maxPrimeMis = 2,
  fwdTrimLen = 0,
  revTrimLen = 0,
  maxEEVal = 2,
  truncQValue = 2,
  truncLenValueF = 0,
  truncLenValueR = 0,
  error = 0.1,
  nbases = 1e+80,
  maxMismatchValue = 0,
  minOverlapValue = 12,
  trimOverhang = FALSE,
  minFinalSeqLen = 100,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>runFolderLoc</code></td>
<td>
<p>Select a file in the one of the run folders with the
fastq files of interest (Default NULL).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>primerFile</code></td>
<td>
<p>Select a file with the primers for this analysis (Default NULL).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fwdIdent</code></td>
<td>
<p>Forward identifier naming string (Default '_R1_001').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>revIdent</code></td>
<td>
<p>Reverse identifier naming string (Default '_R2_001').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>unidirectional</code></td>
<td>
<p>Selection to process files independently (Default FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bidirectional</code></td>
<td>
<p>Selection to process paired forward and reverse sequence
for analysis (Default TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>printQualityPdf</code></td>
<td>
<p>Selection to process save image files showing quality metrics (Default TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxPrimeMis</code></td>
<td>
<p>Maximum number of mismatches allowed when pattern matching
trimming the primers from the ends of the reads for the ShortRead trimLRPatterns()
function (Default 2).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fwdTrimLen</code></td>
<td>
<p>Select a forward trim length for the Dada filterAndTrim()
function (Default 0).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>revTrimLen</code></td>
<td>
<p>Select a reverse trim length for the Dada filterAndTrim()
function (Default 0).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxEEVal</code></td>
<td>
<p>Maximum number of expected errors allowed in a read for the
Dada filterAndTrim() function (Default 2).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>truncQValue</code></td>
<td>
<p>Truncation value use to trim ends of reads, nucleotides
with quality values less than this value will be used to trim the remainder
of the reads for the Dada filterAndTrim() function (Default 2).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>truncLenValueF</code></td>
<td>
<p>Dada forward length trim value for the Dada filterAndTrim()
function. This function is set to 0 when the pattern matching trim function is
enabled (Default 0).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>truncLenValueR</code></td>
<td>
<p>Dada reverse length trim value for the Dada filterAndTrim()
function. This function is set to 0 when the pattern matching trim function is
enabled (Default 0).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>error</code></td>
<td>
<p>Percent of fastq files used to assess error rates for the Dada
learnErrors() function (Default 0.1).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nbases</code></td>
<td>
<p>The total number of bases used to assess errors for the Dada
learnErrors() function (Default 1e80) NOTE: this value is set very
high to get all nucleotides in the error present file subset. If the error
is to be assessed using total reads and not specific fastq files then set
the error to 1 and set this value to the desired number of reads.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxMismatchValue</code></td>
<td>
<p>Maximum number of mismatches allowed when merging
two reads for the Dada mergePairs() function (Default 2).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minOverlapValue</code></td>
<td>
<p>Minimum number of overlapping nucleotides for the
forward and reverse reads for the Dada mergePairs() function (Default 12).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trimOverhang</code></td>
<td>
<p>Trim merged reads past the start of the complimentary
primer regions for the Dada mergePairs() function (Default FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minFinalSeqLen</code></td>
<td>
<p>The minimum final desired length of the read (Default 100).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>If set to TRUE then there will be output to the R console, if
FALSE then this reporting data is suppressed (Default TRUE).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Two file types are required as input for the dada_implement() function.
The first are the fastq files in the appropriate folder structure (see below)
and the second is a file containing the primers used for the amplification of
the sequence reads.
</p>
<p>Fastq File Folder Structure
</p>
<p>Parent Directory
|
|
—————–
|               |
|               |
Run1 Directory     Run2 Directory
-Fastq             -Fastq
-Fastq             -Fastq
...                ...
</p>
<p>Format of the primer file
</p>
<p>| Forward              | Reverse                          |
| AGTGTGTAGTGATTG      | CGCATCGCTCAGACTGACTGC            |
| GAGCCCTCGATCGCT      | GGTCGATAGCTACGCGCGCATACGACT      |
|                      | GGTTCACATCGCATTCAT               |
</p>
<p>The examples are present to display the syntax for the function.
These examples are not run because there are files required to run the functions,
in some cases multiple files are necessary and some of these are quite large. To
get specific examples please see https://github.com/rgyoung6/DBTCShinyTutorial/blob/main/README.md
</p>


<h3>Value</h3>

<p>The output from this function includes four folders.
A_Qual - Contains quality pdf files for the input fastq files (if printQualityPdf set to TRUE).
B_Filt - Contains dada filtered fastq files and a folder with the end trimmed
fastq files before quality filtering.
C_FiltQual - Contains quality pdf files for the filtered fastq files (if printQualityPdf set to TRUE).
D_Output - This folder contains output files including and analysis summary,
an analysis summary table of processing values, forward and reverse error assessments,
and finally the output ASV and fasta files of obtained sequences.                                                                                                   -TotalTable.tsv
</p>


<h3>Note</h3>

<p>WARNING - NO WHITESPACE!
</p>
<p>When running DBTC functions the paths for the files selected cannot have white
space! File folder locations should be as short as possible (close to the root
as some functions do not process long naming conventions.
</p>
<p>Also, special characters should be avoided (including question mark, number
sign, exclamation mark). It is recommended that dashes be used for separations
in naming conventions while retaining underscores for use as information
delimiters (this is how DBTC functions use underscore).
</p>
<p>There are several key character strings used in the DBTC pipeline, the presence
of these strings in file or folder names will cause errors when running DBTC functions.
</p>
<p>The following strings are those used in DBTC and should not be used in file or folder naming:
- _BLAST
- _combinedDada
- _taxaAssign
- _taxaAssignCombined
- _taxaReduced
- _CombineTaxaReduced
</p>


<h3>Author(s)</h3>

<p>Robert G. Young
</p>


<h3>References</h3>

<p>&lt;https://github.com/rgyoung6/DBTC&gt;
Young, R. G., Hanner, R. H. (Submitted October 2023). Dada-BLAST-Taxon Assign-Condense
Shiny Application (DBTCShiny). Biodiversity Data Journal.
</p>


<h3>See Also</h3>

<p>combine_dada_output()
make_BLAST_DB()
seq_BLAST()
taxon_assign()
combine_assign_output()
reduce_taxa()
combine_reduced_output()
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
dada_implement()
dada_implement(runFolderLoc = NULL, primerFile = NULL,fwdIdent = "_R1_001",
revIdent = "_R2_001",unidirectional = FALSE, bidirectional = TRUE, printQualityPdf = TRUE,
maxPrimeMis = 2, fwdTrimLen = 0, revTrimLen = 0,maxEEVal=2, truncQValue = 2,
truncLenValueF = 0, truncLenValueR = 0,error = 0.1, nbases = 1e80,
maxMismatchValue = 0, minOverlapValue = 12,trimOverhang = FALSE,
minFinalSeqLen = 100)

## End(Not run)

</code></pre>


</div>