<div class="container">

<table style="width: 100%;"><tr>
<td>MaxQdataconvert</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
one-step to extract 'Maxquant' quantification data and convert
</h2>

<h3>Description</h3>

<p>'Maxquant' quantification data extract and homolog protein Uniprot ID match.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MaxQdataconvert(pgfilename, IDname = "Majority.protein.IDs",
                IDtype = c("MaxQ","none"), CONremove = TRUE,
                justID = TRUE, status1 = TRUE, ENTRY1 = TRUE,
                db1.path = NULL, db2.path = NULL,
                out.folder = NULL, blast.path = NULL,
                savecsvpath = NULL, csvfilename = NULL,
                verbose = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pgfilename</code></td>
<td>

<p>'Maxquant' quantification file "protein groups.txt"
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>IDname</code></td>
<td>

<p>The column name of uniprot ID. The default value is "<code>Majority.protein.IDs</code>"" which is the column name in MaxQuant data.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>IDtype</code></td>
<td>

<p>"<code>MaxQ</code>" means <code>proteinGroups</code> is 'Maxquant' quantification data, "<code>none</code>" means other type data. This must be (an abbreviation of) one of the strings: "<code>MaxQ</code>","<code>none</code>".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CONremove</code></td>
<td>

<p>a logical value indicated whether remove contaminant IDs. When IDtype is "none", it will remove unmatch ID compared with database2.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>justID</code></td>
<td>

<p>a logical value indicated whether only extract ID when IDtype is "MaxQ".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>status1</code></td>
<td>

<p>a logical value indicated whether extract the first ID status when IDtype is "MaxQ".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ENTRY1</code></td>
<td>

<p>a logical value indicated whether extract the first ID ENTRY NAME when IDtype is "MaxQ".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>db1.path</code></td>
<td>

<p>fasta file, database of transfered species
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>db2.path</code></td>
<td>

<p>fasta file, database of original species
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out.folder</code></td>
<td>

<p>blast result output folder, the folder path should be the same with db1.path
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blast.path</code></td>
<td>

<p>blast+ software install path
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>savecsvpath</code></td>
<td>

<p>the information of csv file name output path. The default value means don't save csv file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>csvfilename</code></td>
<td>

<p>the name of csv file which the data are to be output. The default value means don't save csv file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>integer level of verbosity. Zero means silent, higher values make the output progressively more and more verbose.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Other arguments.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>one-step to extract MaxQuant or other quantification data and convert.
The function contain ID_match function.
</p>


<h3>Value</h3>

<p>a list of proteomic information.
</p>
<table>
<tr style="vertical-align: top;">
<td><code>protein_IDs</code></td>
<td>
<p>Portein IDs which is <code>IDname</code> column information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>intensity</code></td>
<td>
<p>Quantification intensity informaton. When  <code>IDtype</code> is "<code>none</code>", it is the  <code>QuanCol</code> columns information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iBAQ</code></td>
<td>
<p>Quantification iBAQ intensity informaton.(only for <code>IDtype</code> is "<code>MaxQ</code>")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LFQ</code></td>
<td>
<p>Quantification LFQ intensity informaton.(only for <code>IDtype</code> is "<code>MaxQ</code>")</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>The function should install 'blast+' software, Version 2.7.1.
'blast+' download website:https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/
db1.path, db2.path, out.folder are both need the complete path. Out.folder and db1.path should be in the same folder. Path should have no special character.
</p>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>See Also</h3>

<p>ID_match
</p>


<h3>Examples</h3>

<pre><code class="language-R"># suggested to install blast+ software

# it will take a long time without blast+ software
data(Sample_ID_data)
if(requireNamespace("Biostrings", quietly = TRUE)){
  out.folder = tempdir();
  write.table(Sample_ID_data$db1,file.path(out.folder,"db1.fasta"),
              quote = FALSE,row.names = FALSE, col.names = FALSE);
  write.table(Sample_ID_data$db2,file.path(out.folder,"db2.fasta"),
              quote = FALSE,row.names = FALSE, col.names = FALSE);
  write.table(Sample_ID_data$pginf,
              file = file.path(out.folder,"proteingroups.txt"),
              quote = FALSE,
              sep = "\t",dec = ".", row.names = FALSE, col.names = TRUE )
  Maxdata &lt;- MaxQdataconvert(file.path(out.folder,"proteingroups.txt"),
                             IDtype = "MaxQ",
                             db1.path = file.path(out.folder,"db1.fasta"),
                             db2.path = file.path(out.folder,"db2.fasta"),
                             out.folder = out.folder,
                             blast.path = NULL)
  file.remove( file.path(out.folder,"db1.fasta"),
               file.path(out.folder,"db2.fasta"),
               file.path(out.folder,"proteingroups.txt") )
  }

</code></pre>


</div>