<div class="container">

<table style="width: 100%;"><tr>
<td>debar</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2><span class="pkg">debar</span></h2>

<h3>Description</h3>

<p><span class="pkg">debar</span> is an R package designed for the identification and removal 
of insertion and deletion errors from COI-5P DNA barcode data.
</p>


<h3>Details</h3>

<p><span class="pkg">debar</span> is built around the DNAseq object, which takes a COI-5P DNA barcode sequence
and optionally its associated name and PHRED quality information as input. The package utilizes
a nucleotide profile hidden Markov model (PHMM) for the identification of the COI-5P region of 
an input sequence and the identification and correction of indel errors from within the COI-5P 
region of the sequence. Indel corrections are by default applied in a conservative fashion, with subsequent
censorship of 7 base pairs in either direction of an indel correction to mask most instances where the exact
bp corresponding to the indel was not found exactly. Numerous filtering and double check steps are applied, and
the package includes functions for input/output for either fasta or fastq formats.
</p>
<p>The denoise pipeline is heavily paramaterized so that a user can tailor the denoising execution for their own 
data structure and goal.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>denoise_file</code> Run the denoise pipeline for each sequence in a specified input file.
</p>
</li>
<li> <p><code>denoise</code> Run the denoise pipeline for a specified sequence
</p>
</li>
<li> <p><code>read_fasta</code> Read data from a fasta file to a data frame.
</p>
</li>
<li> <p><code>read_fastq</code> Read data from a fastq file to a data frame. 
</p>
</li>
<li> <p><code>write_fasta</code> Write a denoised sequence to the specified fasta file.
</p>
</li>
<li> <p><code>write_fastq</code> Write a denoised sequence and associated quality information to the specified fastq file.
</p>
</li>
<li> <p><code>DNAseq</code> Builds a DNAseq class object
</p>
</li>
<li> <p><code>frame</code> Match a sequence against the COI-5P PHMM using the 
Viterbi algorithm to establish the reading frame,
optional rejection of sequence based on the quality of the match to the PHMM.
</p>
</li>
<li> <p><code>adjust</code> Use the PHMM path output corresponding to the sequence to 
adjust the DNA sequence and remove indels.
Optional censorship of sequence around the corrections.
</p>
</li>
<li> <p><code>aa_check</code> Translate the adjusted sequence to amino acids and check it for stop codons.
</p>
</li>
<li> <p><code>outseq</code> Construct the output data for the given sequence. 
Optionally can include or exculde sequence data from
outside of the COI-5P region (part of sequence that was not denoised).
</p>
</li>
</ul>
<h3>Data</h3>


<ul>
<li> <p><code>example_nt_string</code> An example COI-5P sequence with no errors.
</p>
</li>
<li> <p><code>example_nt_string_errors</code> An example COI-5P sequence with two indel errors.
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Cameron M. Nugent
</p>


</div>