<div class="container">

<table style="width: 100%;"><tr>
<td>mut.to.sigs.input</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Converts mutation list to correct input format</h2>

<h3>Description</h3>

<p>Given a mutation list, outputs a data frame with counts of how frequently a 
mutation is found within each trinucleotide context per sample ID.  Output 
can be used as input into getTriContextFraction.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mut.to.sigs.input(mut.ref, sample.id = "Sample", chr = "chr", pos = "pos",
  ref = "ref", alt = "alt", bsg = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>mut.ref</code></td>
<td>
<p>Location of the mutation file that is to be converted or name 
of data frame in environment</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample.id</code></td>
<td>
<p>Column name in the mutation file corresponding to the Sample
ID</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Column name in the mutation file corresponding to the chromosome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>
<p>Column name in the mutation file corresponding to the mutation 
position</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref</code></td>
<td>
<p>Column name in the mutation file corresponding to the reference 
base</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alt</code></td>
<td>
<p>Column name in the mutation file corresponding to the alternate 
base</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bsg</code></td>
<td>
<p>Only set if another genome build is required. Must be a BSgenome
object.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The context sequence is taken from the BSgenome.Hsapiens.UCSC.hgX::Hsapiens 
object. Therefore the coordinates must correspond to the human hgX assembly. 
Default is set to the UCSC hg19 assembly, which corresponds to the GRCh37 
assembly. If another assembly is required, it must already be present in the 
R workspace and fed as a parameter. This method will translate chromosome 
names from other versions of the assembly like NCBI or Ensembl. For instance,
the following transformation will be done: "1" -&gt; "chr1"; "MT" -&gt; "chrM"; 
"GL000245.1" -&gt; "chrUn_gl000245"; etc.
</p>


<h3>Value</h3>

<p>A data frame that contains sample IDs for the rows and trinucleotide 
contexts for the columns. Each entry is the count of how many times a 
mutation with that trinucleotide context is seen in the sample.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
sigs.input = mut.to.sigs.input(mut.ref = sample.mut.ref, sample.id = "Sample",
chr = "chr", pos = "pos", ref = "ref", alt = "alt", bsg = 
BSgenome.Hsapiens.UCSC.hg19)

## End(Not run)
</code></pre>


</div>