<div class="container">

<table style="width: 100%;"><tr>
<td>predict_effector</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Predict effector</h2>

<h3>Description</h3>

<p><code>predict_effector</code> is used to predict effector protein given amino acid protein sequences.
</p>


<h3>Usage</h3>

<pre><code class="language-R">predict_effector(input, taxon)

## S3 method for class 'character'
predict_effector(input, taxon)

## S3 method for class 'data.frame'
predict_effector(input, taxon)

## S3 method for class 'AAStringSet'
predict_effector(input, taxon)

## S3 method for class 'AAString'
predict_effector(input, taxon)

## Default S3 method:
predict_effector(input, taxon)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>input</code></td>
<td>
<p>Input data that contains amino acid sequence(s). It can be in fasta format, strings, AAString, AAStringset, and dataframe.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxon</code></td>
<td>
<p>Taxon group of input data. Available taxons are bacteria, fungi, and oomycete.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p><code>predict_effector</code> returns an object of class "tbl_deepredeff" or for multiple responses of class c("tbl_deepredeff", "data.frame").
</p>
<p>An object of class "tbl_deepredeff" is a data frame containing at least the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>sequence</code></td>
<td>
<p>the sequence(s) from the input data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>s_score</code></td>
<td>
<p>score obtained from sigmoid function showing how likely the sequences to be an effector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prediction</code></td>
<td>
<p>class prediction for each sequence, obtained from s_score. If the value of <code>s_score</code> &gt;= 0.5, it will be classified as an effector. Otherwise, it will be classified as a non-effector.</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">
# FASTA input
input_fasta &lt;- system.file("extdata/example/fungi_sample.fasta", package = "deepredeff")

pred_result &lt;- deepredeff::predict_effector(
  input = input_fasta,
  taxon = "fungi"
)


</code></pre>


</div>