<div class="container">

<table style="width: 100%;"><tr>
<td>dehogt_func</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Differentially Expressed Heterogeneous Overdispersion Genes Testing for Count Data
This script implements the main function of the proposed method in the above paper</h2>

<h3>Description</h3>

<p>Differentially Expressed Heterogeneous Overdispersion Genes Testing for Count Data
This script implements the main function of the proposed method in the above paper
</p>


<h3>Usage</h3>

<pre><code class="language-R">dehogt_func(
  data,
  treatment,
  norm_factors = NULL,
  covariates = NULL,
  dist = "qpois",
  padj = TRUE,
  pval_thre = 0.05,
  l2fc = FALSE,
  l2fc_thre = 1,
  num_cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>A matrix of gene expression data where rows represent genes and columns represent samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>treatment</code></td>
<td>
<p>A vector specifying the treatment conditions for each sample.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>norm_factors</code></td>
<td>
<p>An optional vector of normalization factors for each sample. Default is NULL, which assumes equal normalization factors.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>covariates</code></td>
<td>
<p>An optional matrix of gene-wise covariates. Default is NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dist</code></td>
<td>
<p>The distribution family for the GLM. Can be "qpois" for quasi-Poisson or "negbin" for negative binomial. Default is "qpois".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>padj</code></td>
<td>
<p>Logical value indicating whether to adjust p-values using the Benjamini-Hochberg (BH) procedure. Default is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pval_thre</code></td>
<td>
<p>The threshold for identifying differentially expressed genes based on adjusted p-values. Default is 0.05.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>l2fc</code></td>
<td>
<p>Logical value indicating whether to consider log2 fold change for identifying differentially expressed genes. Default is FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>l2fc_thre</code></td>
<td>
<p>The threshold for log2 fold change in identifying differentially expressed genes. Default is 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num_cores</code></td>
<td>
<p>The number of CPU cores to use for parallel computing. Default is 1.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list containing:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>DE_idx</code></td>
<td>
<p>A logical vector indicating differentially expressed genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pvals</code></td>
<td>
<p>A numeric vector of p-values for each gene.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log2fc</code></td>
<td>
<p>A numeric vector of log2 fold changes for each gene.</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R"># simulate gene expression data
data &lt;- matrix(rpois(1000, 10), nrow = 100, ncol = 10)
# simulate random treatment assignments
treatment &lt;- sample(0:1, 10, replace = TRUE)
# Run main function with parallel computing using 2 cores
result &lt;- dehogt_func(data, treatment, num_cores = 2)
</code></pre>


</div>