<div class="container">

<table style="width: 100%;"><tr>
<td>readsData</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>An S4 class to represent summarised counts and the output of Bayesian
inference.</h2>

<h3>Description</h3>

<p>An S4 class to represent summarised counts and the output of Bayesian
inference.
</p>


<h3>Slots</h3>


<dl>
<dt><code>counts</code></dt>
<dd>
<p>A positive integer matrix containing summarised counts for each
genomic event (genes, exons, transcripts, etc)  in the two conditions, A and
B.</p>
</dd>
<dt><code>replicates</code></dt>
<dd>
<p>A list containing the indices of the columns that belong to
each  of the two experimental conditions, A and B. It is defaulted to A =
1:(n/2), B = (n/2+1):n for an <code class="reqn">m</code> by <code class="reqn">n</code> matrix.</p>
</dd>
<dt><code>geneNames</code></dt>
<dd>
<p>A character vector containing the names of the genomic event.
It is appropriately defaulted to names of  the matrix.</p>
</dd>
<dt><code>initValues</code></dt>
<dd>
<p>A list containing initial values for each parameter.
Defaulted to <code>N_A</code> = rep(1, nrow(counts)), <code>N_B</code> = rep(1,
nrow(counts)), <code>p</code>= 0.0001, <code>f</code> = 0.01.</p>
</dd>
<dt><code>stepSizes</code></dt>
<dd>
<p>A list containing step sizes for sampling each parameter.
Defaulted to <code>stepsizeN_A</code> = rep(1, nrow(counts)), <code>stepsizeN_B</code> =
rep(1, nrow(counts)), <code>stepsize_p</code>= 1e3, <code>stepsize_f</code> = 5e7</p>
</dd>
<dt><code>output</code></dt>
<dd>
<p>A list containing the  samples for each parameter which are
generated by Bayesian inference. It can only be  filled
inside the results function.</p>
</dd>
</dl>
</div>