<div class="container">

<table style="width: 100%;"><tr>
<td>extractCoverageFromVcf</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Extract read counts from VCF</h2>

<h3>Description</h3>

<p>Extract read counts from VCF file of a single sample.
</p>


<h3>Usage</h3>

<pre><code class="language-R">extractCoverageFromVcf(vcfFileName, ADFieldIndex = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcfFileName</code></td>
<td>
<p>Path of the VCF file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ADFieldIndex</code></td>
<td>
<p>Index of the AD field of the sample field. For example,
if the format is "GT:AD:DP:GQ:PL", the AD index is 2 (by default).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data.frame contains four columns: chromosomes, positions, reference
allele count, alternative allele count.
</p>


<h3>Note</h3>

<p>The VCF file should only contain one sample. If more samples present in
the VCF, it only returns coverage for of the first sample.
</p>


<h3>Examples</h3>

<pre><code class="language-R">vcfFile = system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid")
PG0390 = extractCoverageFromVcf(vcfFile)

</code></pre>


</div>