<div class="container">

<table style="width: 100%;"><tr>
<td>Froh_inbreeding</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function to calculated Froh genome-wide or chromosome-wide</h2>

<h3>Description</h3>

<p>This function calculates the individual inbreeding coefficients based on runs of
homozygosity (ROH), either per-chromosome (chromosome-wide) or based on the
entire genome (genome-wide). See details of calculations below
</p>


<h3>Usage</h3>

<pre><code class="language-R">Froh_inbreeding(runs, mapFile, genome_wide = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>runs</code></td>
<td>
<p>R object (dataframe) with results on runs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mapFile</code></td>
<td>
<p>Plink map file (to retrieve SNP position)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome_wide</code></td>
<td>
<p>vector of TRUE/FALSE (genome-wide or chromosome-wide;
defaults to TRUE/genome-wide)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Froh is calculated as:
</p>
<p><code class="reqn"> F_{ROH} = \frac{\sum ROH_{length}}{Length_{genome}} </code>
</p>
<p>Depending on whether genome-wide or chromosome-wide calculations are required,
the terms in the numerator and denominator will refer to the entire genome
or will be restricted to specific chromosomes.
</p>


<h3>Value</h3>

<p>A data frame with the inbreeding coefficients of each individual sample
</p>


<h3>Examples</h3>

<pre><code class="language-R"># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# calculating runs of Homozygosity
## Not run: 
# skipping runs calculation
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxOppositeGenotype = 1, maxMiss = 1,  minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
runsDF &lt;- readExternalRuns(inputFile = runsFile, program = 'detectRUNS')

Froh_inbreeding(runs = runsDF, mapFile = mapFile)
Froh_inbreeding(runs = runsDF, mapFile = mapFile, genome_wide=FALSE)

</code></pre>


</div>