<div class="container">

<table style="width: 100%;"><tr>
<td>homoZygotTest</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function to check whether a window is (loosely) homozygous or not</h2>

<h3>Description</h3>

<p>This is a core function. Parameters on how to consider a window homozygous are here (maxHet, maxMiss)
</p>


<h3>Usage</h3>

<pre><code class="language-R">homoZygotTest(x, gaps, maxHet, maxMiss, maxGap, i, windowSize)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>vector of 0/1 genotypes (from genoConvert())</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gaps</code></td>
<td>
<p>vector of differences between consecutive positions (gaps) in bps</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxHet</code></td>
<td>
<p>max n. of heterozygous SNP in a homozygous window</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxMiss</code></td>
<td>
<p>max n. of missing in a window</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxGap</code></td>
<td>
<p>max distance between consecutive SNP in a window to be still considered a potential run</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>i</code></td>
<td>
<p>index along the genome (genome-vector for each individual)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>windowSize</code></td>
<td>
<p>size of window (n. of SNP)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a list: i) TRUE/FALSE (whether a window is heterozygous or NOT); ii) indexes of "opposite and missing" genotype
</p>


</div>