<div class="container">

<table style="width: 100%;"><tr>
<td>summaryRuns</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Summary statistics on detected runs</h2>

<h3>Description</h3>

<p>This function processes the results from <code>slidingRUNS.run</code> and
<code>consecutiveRUNS.run</code> and produces a number of interesting descriptives
statistics on results.
</p>


<h3>Usage</h3>

<pre><code class="language-R">summaryRuns(runs, mapFile, genotypeFile, Class = 2, snpInRuns = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>runs</code></td>
<td>
<p>R object (dataframe) with results on detected runs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mapFile</code></td>
<td>
<p>Plink map file (for SNP position)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotypeFile</code></td>
<td>
<p>Plink ped file (for SNP position)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Class</code></td>
<td>
<p>group of length (in Mbps) by class (default: 0-2, 2-4, 4-8, 8-16, &gt;16)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpInRuns</code></td>
<td>
<p>TRUE/FALSE (default): should the function <code>snpInsideRuns</code> be
called to compute the proportion of times each SNP falls inside a run in the
group/population?</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>summaryRuns</code> calculates: i) the number of runs per chromosome and group/population;
ii) the percent distribution of runs per chromosome and group; iii) the number of
runs per size-class and group; iv) the percent distribution of runs per size-class
and group; v) the mean length of runs per chromosome and group; vi) the mean
length of runs per size-class and group; vii) individual inbreeding coefficient
estimated from ROH; viii) individual inbreeding coefficient estimated from ROH
per chromosome; ix) individual inbreeding coefficient estimated from ROH per
size-class
</p>


<h3>Value</h3>

<p>A list of dataframes containing the most relevant descriptives
statistics on detected runs. The list conveniently contains 9 dataframes that can
be used for further processing and visualization, or can be written out to text files
</p>


<h3>Examples</h3>

<pre><code class="language-R"># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# calculating runs of Homozygosity
## Not run: 
# skipping runs calculation
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxOppositeGenotype = 1, maxMiss = 1,  minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
runsDF &lt;- readExternalRuns(inputFile = runsFile, program = 'detectRUNS')

summaryRuns(runs = runsDF, mapFile = mapFile, genotypeFile = genotypeFile, Class = 2,
snpInRuns = FALSE)

</code></pre>


</div>