<div class="container">

<table style="width: 100%;"><tr>
<td>plot_manhattanRuns</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot the proportion of times SNPs are inside runs - MANHATTAN PLOT</h2>

<h3>Description</h3>

<p>Function to plot the proportion of times/percentage each SNP in inside a run
(population-specific signals) against SNP position in all chromosomes together
Proportions on the y-axis, bps on the x-axis for all analysed chromosomes
This is similar to the familiar GWAS Manhattan plot
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_manhattanRuns(runs, genotypeFile, mapFile, savePlots = FALSE,
  outputName = NULL, plotTitle = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>runs</code></td>
<td>
<p>a data.frame with runs per individual (group, id, chrom, nSNP, start, end, length)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotypeFile</code></td>
<td>
<p>genotype (.ped) file path</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mapFile</code></td>
<td>
<p>map file (.map) file path</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>savePlots</code></td>
<td>
<p>should plots be saved out in files (default) or plotted in the graphical terminal?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outputName</code></td>
<td>
<p>title prefix (the base name of graph, if savePlots is TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plotTitle</code></td>
<td>
<p>title in plot (default)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Manhattan plots of proportion of times SNPs are inside runs,
per population (pdf files)
</p>


<h3>Examples</h3>

<pre><code class="language-R"># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# calculating runs of Homozygosity
## Not run: 
# skipping runs calculation
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxOppositeGenotype = 1, maxMiss = 1,  minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
runsDF &lt;- readExternalRuns(inputFile = runsFile, program = 'detectRUNS')

# plot runs per animal (interactive)
plot_manhattanRuns(runs = runsDF, genotypeFile = genotypeFile, mapFile = mapFile,
savePlots = FALSE, plotTitle = "ROHom")

</code></pre>


</div>