<div class="container">

<table style="width: 100%;"><tr>
<td>tableRuns</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function to retrieve most common runs in the population</h2>

<h3>Description</h3>

<p>This function takes in input either the run results or the output from
the function <code>snpInsideRuns</code> (proportion of times a SNP is inside a run)
in the population/group, and returns a subset of the runs most commonly
found in the group/population. The parameter <code>threshold</code> controls the definition
of most common (e.g. in at least 50%, 70% etc. of the sampled individuals)
</p>


<h3>Usage</h3>

<pre><code class="language-R">tableRuns(runs = NULL, SnpInRuns = NULL, genotypeFile, mapFile,
  threshold = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>runs</code></td>
<td>
<p>R object (dataframe) with results on detected runs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SnpInRuns</code></td>
<td>
<p>dataframe with the proportion of times each SNP falls inside a
run in the population (output from <code>snpInsideRuns</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotypeFile</code></td>
<td>
<p>Plink ped file (for SNP position)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mapFile</code></td>
<td>
<p>Plink map file (for SNP position)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>value from 0 to 1 (default 0.7) that controls the desired
proportion of individuals carrying that run (e.g. 70%)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A dataframe with the most common runs detected in the sampled individuals
(the group/population, start and end position of the run, chromosome and number of SNP
included in the run are reported in the output dataframe)
</p>


<h3>Examples</h3>

<pre><code class="language-R"># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# calculating runs of Homozygosity
## Not run: 
# skipping runs calculation
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxOppositeGenotype = 1, maxMiss = 1,  minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
runsDF = readExternalRuns(inputFile = runsFile, program = 'detectRUNS')

tableRuns(runs = runsDF, genotypeFile = genotypeFile, mapFile = mapFile, threshold = 0.5)

</code></pre>


</div>