<div class="container">

<table style="width: 100%;"><tr>
<td>slidingWindowCpp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function to slide a window over a vector (individual's genotypes)</h2>

<h3>Description</h3>

<p>This is a core function. The functions to detect RUNS are slid over the genome
</p>


<h3>Usage</h3>

<pre><code class="language-R">slidingWindowCpp(data, gaps, windowSize, step, maxGap, ROHet = TRUE,
  maxOppositeGenotype = 1L, maxMiss = 1L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>vector of 0/1/2 genotypes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gaps</code></td>
<td>
<p>vector of differences between consecutive positions (gaps) in bps</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>windowSize</code></td>
<td>
<p>size of window (n. of SNP)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>step</code></td>
<td>
<p>by which (how many SNP) is the window slid</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxGap</code></td>
<td>
<p>max distance between consecutive SNP in a window to be still considered a potential run</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ROHet</code></td>
<td>
<p>shall we detect ROHet or ROHom?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxOppositeGenotype</code></td>
<td>
<p>max n. of homozygous/heterozygous SNP</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxMiss</code></td>
<td>
<p>max. n. of missing SNP</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>vector of TRUE/FALSE (whether a window is homozygous or NOT)
</p>


</div>