<div class="container">

<table style="width: 100%;"><tr>
<td>plot_InbreedingChr</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot Froh-based inbreeding coefficients by group</h2>

<h3>Description</h3>

<p>The function plots the distribution of inbreeding/consanguinity coefficients
per chromosome and/or group. Three types of plots can be produces: barplots, boxplots,
violin plots. With <code>style="All"</code> all three plots are produced.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_InbreedingChr(runs, mapFile, groupSplit = TRUE,
  style = c("ChrBarPlot", "ChrBoxPlot", "FrohBoxPlot", "All"),
  outputName = NULL, plotTitle = NULL, savePlots = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>runs</code></td>
<td>
<p>R object (dataframe) with results on detected runs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mapFile</code></td>
<td>
<p>Plink map file (for SNP position)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>groupSplit</code></td>
<td>
<p>plots split by group (defaults to TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>style</code></td>
<td>
<p>type of plot: ChrBarPlot, ChrBoxPlot, FrohBoxPlot, All (all plots)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outputName</code></td>
<td>
<p>title prefix (the base name of graph, if savePlots is TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plotTitle</code></td>
<td>
<p>title in plot (default NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>savePlots</code></td>
<td>
<p>should plots be saved out to files or plotted in the graphical terminal (default)?</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>plots of the distribution of inbreeding by chromosome and group
</p>


<h3>Examples</h3>

<pre><code class="language-R"># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# calculating runs of Homozygosity
## Not run: 
# skipping runs calculation
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxOppositeGenotype = 1, maxMiss = 1,  minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
runsDF &lt;- readExternalRuns(inputFile = runsFile, program = 'detectRUNS')

plot_InbreedingChr(runs = runsDF, mapFile = mapFile, style='All')

</code></pre>


</div>