<div class="container">

<table style="width: 100%;"><tr>
<td>tsea.summary</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Tissue-specific enrichment analysis result summary
</h2>

<h3>Description</h3>

<p>Tissue-specific enrichment analysis result summary (list the top 3 most enriched tissues) from the given gene list or RNA-seq expression profiles.
</p>


<h3>Usage</h3>

<pre><code class="language-R">tsea.summary(tsea_result)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>tsea_result</code></td>
<td>
<p>the result of tissue-specific enrichment analysis, which produced by "tsea.analysis", "tsea.analysis.multiple" or "tsea.expression.decode".
</p>
</td>
</tr></table>
<h3>Details</h3>

<p>Tissue-specific enrichment analysis result summary
</p>


<h3>Value</h3>

  
<p>A data frame with summary result of top 3 most enriched tissues.
</p>
<p>Rows stand for sample names and columns stand for top 3 most enriched tissues (with p-value).
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Guangsheng Pei
</p>


<h3>References</h3>

<p>Pei G., Dai Y., Zhao Z., Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.
</p>


<h3>See Also</h3>

<p>https://github.com/bsml320/deTS
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(query_GTEx)
query_matrix = query_GTEx
data(correction_factor)
data(ENCODE_z_score)
query_mat_zscore_nor = tsea.expression.normalization(query_matrix, 
	correction_factor, normalization = "z-score")
tseaed_in_ENCODE = tsea.expression.decode(query_mat_zscore_nor, 
	ENCODE_z_score, 0.05, p.adjust.method = "BH")	
tseaed_in_ENCODE_summary = tsea.summary(tseaed_in_ENCODE)
</code></pre>


</div>