<div class="container">

<table style="width: 100%;"><tr>
<td>tsea.plot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Tissue-specific enrichment analysis result heatmap plot
</h2>

<h3>Description</h3>

<p>Heat map plot for tissue-specific enrichment analysis result.
</p>


<h3>Usage</h3>

<pre><code class="language-R">tsea.plot(tsea_result, threshold = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>tsea_result</code></td>
<td>
<p>the result of tissue-specific enrichment analysis, which produced by "tsea.analysis", "tsea.analysis.multiple" or "tsea.expression.decode".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>the p-value threshold to define if the gene list or RNA-seq data enriched in a given tissue, p-value greater than threshold will not be labeled in the plot. The default value is 0.05.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Heat map plot for tissue-specific enrichment analysis result 
</p>


<h3>Value</h3>

<p>Heatmap plot
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Guangsheng Pei
</p>


<h3>References</h3>

<p>Pei G., Dai Y., Zhao Z., Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.
</p>


<h3>See Also</h3>

<p>https://github.com/bsml320/deTS
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(GWAS_gene_multiple)
data(GTEx_t_score)
query_gene_list = GWAS_gene_multiple
tsea_t_multi = tsea.analysis.multiple(query_gene_list, 
	GTEx_t_score, 0.05, p.adjust.method = "BH")
tsea.plot(tsea_t_multi, 0.05)
</code></pre>


</div>