<div class="container">

<table style="width: 100%;"><tr>
<td>deTS-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Tissue-Specific Enrichment Analysis
Tissue-Specific Enrichment Analysis
</h2>

<h3>Description</h3>

<p>Tissue-specific enrichment analysis to assess lists of candidate genes and tissue-specific expression decode analysis for RNA-seq data to decode RNA expression matrices tissue heterogeneity.
</p>


<h3>Details</h3>

<p>Since disease and physiological condition are often associated with a specific tissue, understanding the tissue-specific genes (TSG) expression patterns will substantially reduce false discoveries in biomedical research. However, due to cell complexity in human system, heterogeneous tissues are frequently collected. Making it difficult to distinguish gene expression variability and mislead result interpretation. Here, we present deTS, an R package that conducts Tissue-Specific Enrichment Analysis (TSEA) using two built-in reference panels: the Genotype-Tissue Expression (GTEx) data and the ENCyclopedia Of DNA Elements (ENCODE) data. We implemented two major functions in TSEA to assess lists of candidate genes or expression matrices.
</p>
<p>The DESCRIPTION file:
</p>

<table>
<tr>
<td style="text-align: left;">
Package: </td>
<td style="text-align: left;"> deTS</td>
</tr>
<tr>
<td style="text-align: left;">
Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
Title: </td>
<td style="text-align: left;"> Tissue-Specific Enrichment Analysis</td>
</tr>
<tr>
<td style="text-align: left;">
Version: </td>
<td style="text-align: left;"> 1.0</td>
</tr>
<tr>
<td style="text-align: left;">
Date: </td>
<td style="text-align: left;"> 2019-02-06</td>
</tr>
<tr>
<td style="text-align: left;">
Author: </td>
<td style="text-align: left;"> Guangsheng Pei</td>
</tr>
<tr>
<td style="text-align: left;">
Maintainer: </td>
<td style="text-align: left;"> Guangsheng Pei &lt;peiguangsheng@gmail.com&gt;</td>
</tr>
<tr>
<td style="text-align: left;">
Imports: </td>
<td style="text-align: left;"> pheatmap, RColorBrewer</td>
</tr>
<tr>
<td style="text-align: left;">
Description: </td>
<td style="text-align: left;"> Tissue-specific enrichment analysis to assess lists of candidate genes or RNA-Seq expression profiles. Pei G., Dai Y., Zhao Z. Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.</td>
</tr>
<tr>
<td style="text-align: left;">
License: </td>
<td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p>Index of help topics:
</p>
<pre>
ENCODE_z_score          ENCODE z-score to define tissue-specific genes
GTEx_t_score            GTEx t-score to define tissue-specific genes
GWAS_gene               Gene symbol query data for single sample
GWAS_gene_multiple      Gene symbol query data for multiple samples
correction_factor       Gene average expression level and standard
                        deviation in GTEx data
deTS-package            Tissue-Specific Enrichment Analysis
                        Tissue-Specific Enrichment Analysis
query_ENCODE            ENCODE raw query data
query_GTEx              GTEx raw query data
tsea.analysis           Tissue-specific enrichment analysis for query
                        gene list
tsea.analysis.multiple
                        Tissue-specific enrichment analysis for multi
                        query gene lists
tsea.expression.decode
                        Tissue-specific enrichment analysis for RNA-Seq
                        expression profiles
tsea.expression.normalization
                        RNA-Seq expression profiles normalization
tsea.plot               Tissue-specific enrichment analysis result
                        heatmap plot
tsea.summary            Tissue-specific enrichment analysis result
                        summary
</pre>


<h3>Author(s)</h3>

<p>Guangsheng Pei
</p>
<p>Maintainer: Guangsheng Pei
</p>


<h3>References</h3>

<p>Pei G., Dai Y., Zhao Z., Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.
</p>


<h3>See Also</h3>

<p>https://github.com/bsml320/deTS
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(GTEx_t_score)
data(ENCODE_z_score)
library(pheatmap)

data(GWAS_gene)
query_gene_list = GWAS_gene
tsea_t = tsea.analysis(query_gene_list, GTEx_t_score, 0.05,
	p.adjust.method = "bonferroni")
tsea_t_summary = tsea.summary(tsea_t)

data(GWAS_gene_multiple)
query_gene_list = GWAS_gene_multiple[,1:2]
tsea_t_multi = tsea.analysis.multiple(query_gene_list, 
	GTEx_t_score, 0.05, p.adjust.method = "BH")

data(query_GTEx)
query_matrix = query_GTEx[,1:2]
data(correction_factor)
query_mat_zscore_nor = tsea.expression.normalization(query_matrix, 
	correction_factor, normalization = "z-score")
tseaed_in_ENCODE = tsea.expression.decode(query_mat_zscore_nor, 
	ENCODE_z_score, 0.05, p.adjust.method = "BH")
tseaed_in_ENCODE_summary = tsea.summary(tseaed_in_ENCODE)
</code></pre>


</div>