<div class="container">

<table style="width: 100%;"><tr>
<td>tsea.expression.decode</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Tissue-specific enrichment analysis for RNA-Seq expression profiles
</h2>

<h3>Description</h3>

<p>Tissue-specific enrichment analysis to decode whether a given RNA-seq sample (RPKM) with potential confounding effects based on expression profiles.
</p>


<h3>Usage</h3>

<pre><code class="language-R">tsea.expression.decode(query_mat_normalized_score, score, 
ratio = 0.05, p.adjust.method = "BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>query_mat_normalized_score</code></td>
<td>
<p>a normalized RNA-seq RPKM object, which produced by "tsea.expression.normalization".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>score</code></td>
<td>
<p>a gene tissue-specific score matrix, c("GTEx_t_score" or "ENCODE_z_score"), can be loaded by data(GTEx) or data(ENCODE), the default value is recommended "GTEx_t_score".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ratio</code></td>
<td>
<p>the threshold to define tissue-specific genes (with top t-score or z-score), the default value is 0.05.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.adjust.method</code></td>
<td>
<p>p.adjust.method, c("holm", "hochberg", "hommel", "bonferroni", "BH", "BY", "fdr", "none")
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Tissue-specific enrichment analysis for RNA-Seq expression profiles.
</p>


<h3>Value</h3>

<p>A data frame with p-value of tissue-specific enrichment result for RNA-Seq expression profiles.
</p>
<p>Rows stand for tissue names and columns stand for sample names.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Guangsheng Pei
</p>


<h3>References</h3>

<p>Pei G., Dai Y., Zhao Z., Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.
</p>


<h3>See Also</h3>

<p>https://github.com/bsml320/deTS
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(query_GTEx)
query_matrix = query_GTEx[,1:2]
data(correction_factor)
data(ENCODE_z_score)
query_mat_zscore_nor = tsea.expression.normalization(query_matrix, 
	correction_factor, normalization = "z-score")
tseaed_in_ENCODE = tsea.expression.decode(query_mat_zscore_nor, 
	ENCODE_z_score, 0.05, p.adjust.method = "BH")
</code></pre>


</div>