<div class="container">

<table style="width: 100%;"><tr>
<td>dGAselID</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>dGAselID</h2>

<h3>Description</h3>

<p>Initializes and starts the search with the genetic algorithm.
</p>


<h3>Usage</h3>

<pre><code class="language-R">dGAselID(x, response, method = knn.cvI(k = 3, l = 2), trainTest = "LOG",
  startGenes, populationSize, iterations, noChr = 22, elitism = NA,
  ID = "ID1", pMutationChance = 0, nSMutationChance = 0,
  fSMutationChance = 0, lSDeletionChance = 0, wChrDeletionChance = 0,
  transposonChance = 0, randomAssortment = TRUE, embryonicSelection = NA,
  EveryGeneInInitialPopulation = TRUE, nnetSize = NA, nnetDecay = NA,
  rdaAlpha = NA, rdaDelta = NA, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Dataset in ExpressionSet format.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>response</code></td>
<td>
<p>Response variable</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Supervised classifier for fitness evaluation. Most of the supervised classifiers in MLInterfaces are acceptable. The default is knn.cvI(k=3, l=2).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trainTest</code></td>
<td>
<p>Cross-validation method. The default is "LOG".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>startGenes</code></td>
<td>
<p>Genes in the genotypes at initialization.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>populationSize</code></td>
<td>
<p>Number of genotypes in initial population.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iterations</code></td>
<td>
<p>Number of iterations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>noChr</code></td>
<td>
<p>Number of chromosomes. The default value is 22.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>elitism</code></td>
<td>
<p>Elite population in percentages.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ID</code></td>
<td>
<p>Dominance. The default value is "ID1". Use "ID2" for Incomplete Dominance.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pMutationChance</code></td>
<td>
<p>Chance for a Point Mutation to occur. The default value is 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nSMutationChance</code></td>
<td>
<p>Chance for a Non-sense Mutation to occur. The default value is 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fSMutationChance</code></td>
<td>
<p>Chance for a Frameshift Mutation to occur. The default value is 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lSDeletionChance</code></td>
<td>
<p>Chance for a Large Segment Deletion to occur. The default value is 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>wChrDeletionChance</code></td>
<td>
<p>Chance for a Whole Chromosome Deletion to occur. The default value is 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>transposonChance</code></td>
<td>
<p>Chance for a Transposon Mutation to occur. The default value is 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>randomAssortment</code></td>
<td>
<p>Random Assortment of Chromosomes for recombinations. The default value is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>embryonicSelection</code></td>
<td>
<p>Remove chromosomes with fitness &lt; specified value. The default value is NA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>EveryGeneInInitialPopulation</code></td>
<td>
<p>Request for every gene to be present in the initial population. The default value is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nnetSize</code></td>
<td>
<p>for nnetI. The default value is NA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nnetDecay</code></td>
<td>
<p>for nnetI. The default value is NA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rdaAlpha</code></td>
<td>
<p>for rdaI. The default value is NA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rdaDelta</code></td>
<td>
<p>for rdaI. The default value is NA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional arguments.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The output is a list containing 5 named vectors, records of the evolution:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>DGenes</code></td>
<td>
<p>The occurrences in selected genotypes for every gene,</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dGenes</code></td>
<td>
<p>The occurrences in discarded genotypes for every gene,</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MaximumAccuracy</code></td>
<td>
<p>Maximum accuracy in every generation,</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MeanAccuracy</code></td>
<td>
<p>Average accuracy in every generation,</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MinAccuracy</code></td>
<td>
<p>Minimum accuracy in every generation,</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BestIndividuals</code></td>
<td>
<p>Best individual in every generation.</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
 library(genefilter)
 library(ALL)
 data(ALL)
 bALL = ALL[, substr(ALL$BT,1,1) == "B"]
 smallALL = bALL[, bALL$mol.biol %in% c("BCR/ABL", "NEG")]
 smallALL$mol.biol = factor(smallALL$mol.biol)
 smallALL$BT = factor(smallALL$BT)
 f1 &lt;- pOverA(0.25, log2(100))
 f2 &lt;- function(x) (IQR(x) &gt; 0.5)
 f3 &lt;- ttest(smallALL$mol.biol, p=0.1)
 ff &lt;- filterfun(f1, f2, f3)
 selectedsmallALL &lt;- genefilter(exprs(smallALL), ff)
 smallALL = smallALL[selectedsmallALL, ]
 rm(f1)
 rm(f2)
 rm(f3)
 rm(ff)
 rm(bALL)
 sum(selectedsmallALL)

 set.seed(149)
 res&lt;-dGAselID(smallALL, "mol.biol", trainTest=1:79, startGenes=12, populationSize=200,
               iterations=150, noChr=5, pMutationChance=0.0075, elitism=4)
 
## End(Not run)
</code></pre>


</div>