<div class="container">

<table style="width: 100%;"><tr>
<td>initDGEobj</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Initialize with base data (primaryAssayData, row annotations, col annotations)</h2>

<h3>Description</h3>

<p>Initialize with base data (primaryAssayData, row annotations, col annotations)
</p>


<h3>Usage</h3>

<pre><code class="language-R">initDGEobj(
  primaryAssayData,
  rowData,
  colData,
  level,
  customAttr,
  allowShortSampleIDs = FALSE,
  DGEobjDef = initDGEobjDef()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>primaryAssayData</code></td>
<td>
<p>A numeric matrix or dataframe with row and
colnames. Each column represents a sample.  Each row represents and assay.
This is typically the counts matrix in a DGE RNA-Seq experiment.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rowData</code></td>
<td>
<p>Gene, exon, isoform or protein level annotation. Rownames
must match the rownames in primaryAssayData</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colData</code></td>
<td>
<p>A dataframe describing the experiment design.
Rownames much match the colnames(primaryAssayData)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>level</code></td>
<td>
<p>One of "gene", "exon", "isoform" or "protein"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>customAttr</code></td>
<td>
<p>(optional) Named list of attributes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allowShortSampleIDs</code></td>
<td>
<p>Using sequential integer rownames (even if typed
as character) is discouraged and by default will abort the DGEobj creation.
If you have a legitimate need to have short sample names composed of
numeric characters, you can set this argument to TRUE (default = FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>DGEobjDef</code></td>
<td>
<p>An object definition. Defaults to the global
DGEobj definition (initDGEobjDef()) and you usually shouldn't change this unless
you're customizing the object for new data types.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A DGEobj
</p>


<h3>Examples</h3>

<pre><code class="language-R">   dgeObj   &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))
   MyCounts &lt;- dgeObj$counts
   geneinfo &lt;- dgeObj$geneData
   sampinfo &lt;- dgeObj$design

  myDgeObj &lt;- initDGEobj(primaryAssayData = MyCounts,
                         rowData          = geneinfo,
                         colData          = sampinfo,
                         level            = "gene",
                         customAttr       = list (Genome = "Rat.B6.0",
                                               GeneModel = "Ensembl.R89"))

</code></pre>


</div>