<div class="container">

<table style="width: 100%;"><tr>
<td>plotPolarized</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot Polarized Genotypes</h2>

<h3>Description</h3>

<p>Plots genotypes that can be optionally polarized.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotPolarized(
  genotypes,
  HI,
  cols = c("#FFFFFF", "#800080", "#FFE500", "#008080"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genotypes</code></td>
<td>
<p>A character matrix comprising of _012 encodings.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>HI</code></td>
<td>
<p>A numeric vector of individual hybrid indices with length equal to number
of rows in <code>genotypes</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cols</code></td>
<td>
<p>A vector of four colors, representing missing data, homozygotes for
genotype 0, heterozygotes and homozygotes for genotype 2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional selected arguments passed to image and
axis.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>To import and polarize genotypes, use the importPolarized function.
</p>
<p>When using diem, hybrid indices,
<code>HI</code>, can be found in the file 'HIwithOptimalPolarities.txt'. Alternatively,
calculate <code>HI</code> from the polarized genotypes as shown in the examples.
</p>
<p>By default, the function plots colored tick marks for individuals, changing the
color at the steepest change in sorted <code>HI</code>. The second and fourth colors in
<code>cols</code> are used for the tick marks.
</p>

<ul>
<li>
<p> To turn off this feature, use the argument <code>tick = FALSE</code>.
</p>
</li>
<li>
<p> To use custom tick mark colors, provide a vector of colors for all individuals
(equal to the number of rows in <code>genotypes</code>). The <strong>vector of colors must be
ordered</strong> according to <code>order(HI)</code>.
</p>
</li>
<li>
<p> To include individual <code>labels</code> (e.g., accession numbers), provide a character
vector with the <strong>names in the same order as they are</strong> in the <code>genotypes</code>.
</p>
</li>
</ul>
<h3>Value</h3>

<p>No return value, called for side effects. In the default plot, purple and green
represent sides of the barrier to gene flow encoded as <code>0</code> and <code>2</code>, respectively,
yellow shows heterozygotes and white represents missing or undetermined genotypes.
Individuals are ordered according to the sorted <code>HI</code>.
</p>


<h3>See Also</h3>

<p>plotMarkerAxis to add chromosome information to the x axis.
</p>


<h3>Examples</h3>

<pre><code class="language-R">gen &lt;- importPolarized(
  file = system.file("extdata", "data7x10.txt", package = "diemr"),
  changePolarity = c(TRUE, FALSE, TRUE, TRUE, FALSE, FALSE, TRUE, FALSE, FALSE, TRUE),
  ChosenInds = 1:7
)

h &lt;- apply(gen, 1, FUN = function(x) pHetErrOnStateCount(sStateCount(x)))[1, ]

plotPolarized(genotypes = gen, HI = h)

# Incorrect tick color order
plotPolarized(gen, h, col.ticks = c(rep("purple", 5), "green", "purple"), lwd = 3)

# Correct tick color order
plotPolarized(gen, h, col.ticks = c(rep("purple", 5), "green", "purple")[order(h)], lwd = 3)

# Correct individual label order 
plotPolarized(gen, h, labels = c(paste("purple", 1:5), "green 1", "purple 6"), ylab = "")
</code></pre>


</div>