<div class="container">

<table style="width: 100%;"><tr>
<td>importPolarized</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Imports genomic data polarized according to the specification</h2>

<h3>Description</h3>

<p>Reads genotypes from a file and changes marker polarity.
</p>


<h3>Usage</h3>

<pre><code class="language-R">importPolarized(
  files,
  changePolarity,
  ChosenInds,
  ChosenSites = "all",
  nCores = 1,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>files</code></td>
<td>
<p>A character vector with paths to files with genotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>changePolarity</code></td>
<td>
<p>A logical vector with length equal to the number of markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ChosenInds</code></td>
<td>
<p>A numeric vector of indices of individuals to be included in the analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ChosenSites</code></td>
<td>
<p>A logical vector indicating which sites are to be included in the
analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nCores</code></td>
<td>
<p>A numeric number of cores to be used for parallelisation. Must be
<code>nCores = 1</code> on Windows.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical whether to show messages on import progress.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Optional numeric vector of <code>compartmentSizes</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For details on the input data format, check the <code>file</code> with
CheckDiemFormat.
</p>
<p>The <code>changePolarity</code> argument influences how each marker is imported. Value
<code>FALSE</code> means that the marker will be imported as it is saved in the <code>file</code>. Value
<code>TRUE</code> means that the genotypes encoded as <code>0</code> will be imported as <code>2</code>, and genotypes
encoded in the <code>file</code> as <code>2</code> will be imported as <code>0</code>.
</p>


<h3>Value</h3>

<p>Returns a character matrix with rows containing individual genotypes and columns
containing markers.
</p>


<h3>See Also</h3>

<p>diem for determining appropriate marker polarity with
respect to a barrier to gene flow.
</p>


<h3>Examples</h3>

<pre><code class="language-R">dat &lt;- importPolarized(
  files = system.file("extdata", "data7x3.txt", package = "diemr"),
  changePolarity = c(FALSE, TRUE, TRUE),
  ChosenInds = 1:6,
  ChosenSites = "all"
)
dat
#    m1  m2  m3
# 1 "0" "1" "2"
# 2 "0" "0" "0"
# 3 "1" "1" "0"
# 4 "1" "2" "0"
# 5 "2" "2" "1"
# 6 "2" "2" "_"
</code></pre>


</div>