<div class="container">

<table style="width: 100%;"><tr>
<td>sStateCount</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Count states in a vector</h2>

<h3>Description</h3>

<p>Counts genomic states in one sample.
</p>


<h3>Usage</h3>

<pre><code class="language-R">sStateCount(s)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>s</code></td>
<td>
<p>A character vector with elements "_", "0", "1", "2" representing
missing data, homozygots for allele 1, heterozygots, and
homozygots for allele 2. The vector should represent a single individual.</p>
</td>
</tr></table>
<h3>Details</h3>

<p>Summarizes the number of markers that are fixed for an allele in the genome of
one individual. This is used to construct the I4 matrix in diem.
</p>


<h3>Value</h3>

<p>Numeric vector of length 4 with counts of "_", "0", "1", "2" respectively.
</p>


<h3>See Also</h3>

<p>emPolarise for changing marker polarity.
</p>


<h3>Examples</h3>

<pre><code class="language-R">genotype &lt;- c("0", "0", "_", "2", "1", "0", "1")
sStateCount(genotype)
# [1] 1 3 2 1

# calculate state counts for a polarised genotype
sStateCount(emPolarise(genotype, TRUE))
# [1] 1 1 2 3
</code></pre>


</div>