<div class="container">

<table style="width: 100%;"><tr>
<td>emPolarise</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Polarises a marker</h2>

<h3>Description</h3>

<p>Changes encodings of genomic markers according to user specification.
</p>


<h3>Usage</h3>

<pre><code class="language-R">emPolarise(origM, changePolarity = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>origM</code></td>
<td>
<p>A character vector of genotypes comprising of _012 encodings.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>changePolarity</code></td>
<td>
<p>A logical scalar, indicating whether to leave the marker as is
(<code>FALSE</code>) or whether to change its polarity (<code>TRUE</code>).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a character vector with polarised markers.
</p>


<h3>Note</h3>

<p>Note that diem and importPolarized accept also a <code>U</code>
encoding for an unknown or third allele, but <code>emPolarise</code> requires all <code>U</code> to
be replaced with <code style="white-space: pre;">⁠_⁠</code>.
</p>


<h3>See Also</h3>

<p><code>diem</code> for determining appropriate marker polarity with
respect to a barrier to geneflow.
</p>


<h3>Examples</h3>

<pre><code class="language-R">emPolarise(c("0", "0", "1", "2", "2"), TRUE)
# [1] "2" "2" "1" "0" "0"

emPolarise(c("0", "_", "2", "2", "1"), FALSE)
# [1] "0" "_" "2" "2" "1"
</code></pre>


</div>