<div class="container">

<table style="width: 100%;"><tr>
<td>pHetErrOnStateCount</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Hybrid index, heterozygosity, error rate</h2>

<h3>Description</h3>

<p>Using genotype allele counts, calculates
the hybrid index,
heterozygosity and error rate in a single individual.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pHetErrOnStateCount(sCount)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>sCount</code></td>
<td>
<p>A numeric vector of length 4 with allele counts for missing data,
homozygots for allele 1, heterozygots, and homozygots for allele 2.</p>
</td>
</tr></table>
<h3>Details</h3>

<p>Allele counts are genomic state counts multiplied by ploidy. As different
compartments might have different ploidies (e.g. autosomal markers, sex chromosomes,
mitochondrial markers), allele counts should be calculated per compartment and then
summarised to obtain the correct genomic allele counts.
When all individuals in each compartmenst have the same ploidy, state counts do
not need to be corrected.
</p>


<h3>Value</h3>

<p>Returns a named numeric vector with three values: p - hybrid index,
Het - heterozygosity,
Err - error rate.
</p>


<h3>Examples</h3>

<pre><code class="language-R">pHetErrOnStateCount(sCount = c(2, 4, 2, 6))
#         p       Het       Err
# 0.5833333 0.1666667 0.1428571
</code></pre>


</div>