<div class="container">

<table style="width: 100%;"><tr>
<td>diem</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Diagnostic Index Expectation Maximisation</h2>

<h3>Description</h3>

<p>Estimates how to assign alleles in a genome to maximise the distinction between two
unknown groups of individuals. Using expectation maximisation (EM) in likelihood
framework, <code>diem</code> provides marker
polarities for importing data, their likelihood-based diagnostic index and its support
for all markers, and hybrid indices for all individuals.
</p>


<h3>Usage</h3>

<pre><code class="language-R">diem(
  files,
  ploidy = FALSE,
  markerPolarity = FALSE,
  ChosenInds,
  ChosenSites = "all",
  epsilon = 0.99999,
  verbose = FALSE,
  nCores = parallel::detectCores() - 1,
  maxIterations = 50,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>files</code></td>
<td>
<p>A character vector with paths to files with genotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy</code></td>
<td>
<p>A logical or a list of length equal to length of <code>files</code>. Each
element of the list
contains a numeric vector with ploidy numbers for all individuals specified in
the <code>files</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>markerPolarity</code></td>
<td>
<p><code>FALSE</code> or a list of logical vectors.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ChosenInds</code></td>
<td>
<p>A numeric vector of indices of individuals to be included in the analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ChosenSites</code></td>
<td>
<p>A logical vector indicating which sites are to be included in the
analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>epsilon</code></td>
<td>
<p>A numeric, specifying how much the hypothetical diagnostic markers should
contribute to the likelihood calculations. Must be in <code>[0,1)</code>, keeping
tolerance setting of the <code>R</code> session in mind.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical or character with path to directory where run diagnostics will
be saved.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nCores</code></td>
<td>
<p>A numeric number of cores to be used for parallelisation. Must be
<code>nCores = 1</code> on Windows.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxIterations</code></td>
<td>
<p>A numeric.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional arguments.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Given two alleles of a marker, one allele can belong to one side of a barrier
to geneflow and the other to the other side. Which allele belongs where is a non-trivial
matter. A marker state in an individual can be encoded as 0 if the individual is
homozygous for the first allele, and 2 if the individual is homozygous for the second
allele. Marker polarity determines how the marker will be imported. Marker polarity
equal to <code>FALSE</code> means that the marker will be imported as-is. A marker with
polarity equal to <code>TRUE</code> will be imported with states 0 mapped as 2 and states 2
mapped as 0, in effect switching which allele belongs to which side of a barrier to
geneflow.
</p>
<p>When <code>markerPolarity = FALSE</code>, <code>diem</code> uses random null polarities to
initiate the EM algorithm. To fix the null polarities, <code>markerPolarity</code> must be
a list of length equal to the length of the <code>files</code> argument, where each element
in the list is a logical vector of length equal to the number of markers (rows) in
the specific file.
</p>
<p>Ploidy needs to be given for each compartment and for each individual. For example,
for a dataset of three diploid mammal males consisting of an autosomal
compartment, an X chromosome
compartment and a Y chromosome compartment, the ploidy list would be
<code>ploidy = list(rep(2, 3), rep(1, 3), rep(1, 3)</code>. If the dataset consisted of
one male and two females,
ploidy for the sex chromosomes should be vectors reflecting that females have two X
chromosomes, but males only one, and females have no Y chromosomes:
<code>ploidy = list(rep(2, 3), c(1, 2, 2), c(1, 0, 0))</code>.
</p>
<p>When <code>verbose = TRUE</code>, <code>diem</code> will output multiple files with information
on the iterations of the EM algorithm, including tracking marker polarities and the
respective likelihood-based diagnostics. See vignette <code>vignette("Understanding-genome-polarisation-output-files",
  package = "diemr")</code> for a detailed explanation of the individual output files.
</p>


<h3>Value</h3>

<p>A list including suggested marker polarities, diagnostic indices and support for all
markers, four genomic state counts matrix for all individuals, and polarity changes
for the EM iterations.
</p>


<h3>Note</h3>

<p>To ensure that the data input format of the genotype files, ploidies and individual
selection are readable for <code>diem</code>, first use CheckDiemFormat.
Fix all errors, and run <code>diem</code> only once the checks all passed.
</p>
<p>The working directory or a folder optionally specified in the <code>verbose</code>
argument must have write permissions. <code>diem</code> will store temporary files in the
location and output results files.
</p>
<p>The grain for parallelisation is the compartment <code>files</code>.
</p>


<h3>See Also</h3>

<p>CheckDiemFormat
</p>


<h3>Examples</h3>

<pre><code class="language-R"># set up input genotypes file names, ploidies and selection of individual samples
inputFile &lt;- system.file("extdata", "data7x3.txt", package = "diemr")
ploidies &lt;- list(c(2, 1, 2, 2, 2, 1, 2))
inds &lt;- 1:6

# check input data
CheckDiemFormat(files = inputFile, ploidy = ploidies, ChosenInds = inds)
#  File check passed: TRUE
#  Ploidy check passed: TRUE

# run diem
## Not run: 
# diem will write temporal files during EM iterations
# prior to running diem, set the working directory to a location with write permission
fit &lt;- diem(files = inputFile, ChosenInds = inds, ploidy = ploidies, nCores = 1)

# run diem with fixed null polarities
fit2 &lt;- diem(
  files = inputFile, ChosenInds = inds, ploidy = ploidies, nCores = 1,
  markerPolarity = list(c(TRUE, FALSE, TRUE))
)

## End(Not run)
</code></pre>


</div>