<div class="container">

<table style="width: 100%;"><tr>
<td>plotMarkerAxis</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Add a Marker Axis with Chromosome Names to a Plot of Polarized Genotypes</h2>

<h3>Description</h3>

<p>This function adds a marker axis with chromosome names to an existing plot of polarized
genotypes. It requires that the plot is already created using plotPolarized.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotMarkerAxis(
  includedSites,
  ChosenSites,
  tickDist = 1e+06,
  axisInfo = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>includedSites</code></td>
<td>
<p>A character path to a file with columns <code>CHROM</code> and <code>POS</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ChosenSites</code></td>
<td>
<p>A logical vector indicating which sites are to be included in the
analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tickDist</code></td>
<td>
<p>A numeric indicating the spacing of physical tick marks along a chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>axisInfo</code></td>
<td>
<p>A list with user-defined tick positions and labels for marker axis. See
Details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional arguments.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The <code>includedSites</code> file should ideally be generated by
vcf2diem to ensure congruence between the plotted genotypes and
the respective metadata.
</p>
<p>Tick mark distances within a chromosome are located at <code>tickDist</code> and formated to
multiples of millions.
</p>
<p>The optional <code>axisInfo</code> argument must have five named elements with the following
information:
</p>

<ul>
<li> <p><code>CHROMbreaks</code>: Numeric vector with positions defining ticks separating
chromosomes. The metric for all positions is in the number of markers.
</p>
</li>
<li> <p><code>CHROMnamesPos</code>: Numeric vector with positions to place the chromosome
labels.
</p>
</li>
<li> <p><code>CHROMnames</code>: Character vector with the names of the chromosomes. Must be the
same length as <code>CHROMnamesPos</code>.
</p>
</li>
<li> <p><code>ticksPos</code>: Numeric vector with positions of ticks within chromosomes.
</p>
</li>
<li> <p><code>ticksNames</code>: Character vector with the names to be displayed at <code>ticksPos</code>.
</p>
</li>
</ul>
<p>When <code>axisInfo = NULL</code>, the function extracts the necessary information from the
<code>includedSites</code> file.
</p>


<h3>Value</h3>

<p>Returns an invisible <code>axisInfo</code> list with the tick positions and labels
for the marker axis.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# Run this example in a working directory with write permissions
myo &lt;- system.file("extdata", "myotis.vcf", package = "diemr")
vcf2diem(myo, "myo")
inds &lt;- 1:14
fit &lt;- diem("myo-001.txt", ChosenInds = inds, ploidy = FALSE)
gen &lt;- importPolarized("myo-001.txt", fit$markerPolarity, inds)
h &lt;- apply(gen, 1, function(x) pHetErrOnStateCount(sStateCount(x)))[1, ]
plotPolarized(gen, h, xlab = "")
plotMarkerAxis("myo-includedSites.txt", rep(TRUE, 11), tickDist = 100)

## End(Not run)
</code></pre>


</div>