<div class="container">

<table style="width: 100%;"><tr>
<td>calculateEvalMetrics</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate evaluation metrics for bulk RNA-Seq samples from test data</h2>

<h3>Description</h3>

<p>Calculate evaluation metrics for bulk RNA-seq samples from test data to
understand model performance. By default, absolute error (<code>AbsErr</code>),
proportional absolute error (<code>ppAbsErr</code>), squared error (<code>SqrErr</code>)
and proportional squared error (<code>ppSqrErr</code>) are calculated for each test
sample. In addition, each of these metrics is aggregated using their mean
values according to three criteria: each cell type (<code>CellType</code>),
probability bins in ranges of 0.1 (<code>pBin</code>) and number of different cell
types present in the sample <code>nCellTypes</code>. Finally, the process is
repeated only considering bulk samples (filtering out single-cell profiles
from the evaluation). The evaluation metrics will be available in the
<code>test.deconv.metrics</code> slot of the
<code>DigitalDLSorterDNN</code> object (<code>trained.model</code> slot of
the <code>DigitalDLSorter</code> object).
</p>


<h3>Usage</h3>

<pre><code class="language-R">calculateEvalMetrics(object, metrics = c("MAE", "MSE"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p><code>DigitalDLSorter</code> object with a trained
model in the <code>trained.model</code> slot and the actual cell proportions of
pseudo-bulk samples in <code>prob.cell.matrix</code> slot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>metrics</code></td>
<td>
<p>Metrics used to evaluate the model performance. Mean absolute
error (<code>"MAE"</code>) and mean squared error (<code>"MSE"</code>) by default.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A <code>DigitalDLSorter</code> object with the
<code>trained.model</code> slot containing a
<code>DigitalDLSorterDNN</code> object with the
<code>test.deconv.metrics</code> slot. The last contains the metrics calculated.
</p>


<h3>See Also</h3>

<p><code>distErrorPlot</code> <code>corrExpPredPlot</code>
<code>blandAltmanLehPlot</code> <code>barErrorPlot</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
set.seed(123)
sce &lt;- SingleCellExperiment::SingleCellExperiment(
  assays = list(
    counts = matrix(
      rpois(30, lambda = 5), nrow = 15, ncol = 20,
      dimnames = list(paste0("Gene", seq(15)), paste0("RHC", seq(20)))
    )
  ),
  colData = data.frame(
    Cell_ID = paste0("RHC", seq(20)),
    Cell_Type = sample(x = paste0("CellType", seq(6)), size = 20,
                       replace = TRUE)
  ),
  rowData = data.frame(
    Gene_ID = paste0("Gene", seq(15))
  )
)
DDLS &lt;- createDDLSobject(
  sc.data = sce,
  sc.cell.ID.column = "Cell_ID",
  sc.gene.ID.column = "Gene_ID",
  sc.filt.genes.cluster = FALSE, 
  sc.log.FC = FALSE
)
probMatrixValid &lt;- data.frame(
  Cell_Type = paste0("CellType", seq(6)),
  from = c(1, 1, 1, 15, 15, 30),
  to = c(15, 15, 30, 50, 50, 70)
)
DDLS &lt;- generateBulkCellMatrix(
  object = DDLS,
  cell.ID.column = "Cell_ID",
  cell.type.column = "Cell_Type",
  prob.design = probMatrixValid,
  num.bulk.samples = 50,
  verbose = TRUE
)
# training of DDLS model
tensorflow::tf$compat$v1$disable_eager_execution()
DDLS &lt;- trainDDLSModel(
  object = DDLS,
  on.the.fly = TRUE,
  batch.size = 15,
  num.epochs = 5
)
# evaluation using test data
DDLS &lt;- calculateEvalMetrics(
  object = DDLS
)

## End(Not run)

</code></pre>


</div>