<div class="container">

<table style="width: 100%;"><tr>
<td>generateBulkCellMatrix</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate training and test cell composition matrices</h2>

<h3>Description</h3>

<p>Generate training and test cell composition matrices for the simulation of
pseudo-bulk RNA-Seq samples with known cell composition using single-cell
expression profiles. The resulting <code>ProbMatrixCellTypes</code>
object contains a matrix that determines the proportion of the different cell
types that will compose the simulated pseudo-bulk samples. In addition, this
object also contains other information relevant for the process. This
function does not simulate pseudo-bulk samples, this task is performed by the
<code>simBulkProfiles</code> or <code>trainDDLSModel</code>
functions (see Documentation).
</p>


<h3>Usage</h3>

<pre><code class="language-R">generateBulkCellMatrix(
  object,
  cell.ID.column,
  cell.type.column,
  prob.design,
  num.bulk.samples,
  n.cells = 100,
  train.freq.cells = 3/4,
  train.freq.bulk = 3/4,
  proportion.method = c(10, 5, 20, 15, 35, 15),
  prob.sparsity = 0.5,
  min.zero.prop = NULL,
  balanced.type.cells = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p><code>DigitalDLSorter</code> object with
<code>single.cell.real</code> slot and, optionally, with <code>single.cell.simul</code>
slot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cell.ID.column</code></td>
<td>
<p>Name or column number corresponding to the cell names
of expression matrix in cells metadata.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cell.type.column</code></td>
<td>
<p>Name or column number corresponding to the cell type
of each cell in cells metadata.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prob.design</code></td>
<td>
<p>Data frame with the expected frequency ranges for each
cell type present in the experiment. This information can be estimated from
literature or from the single-cell experiment itself. This data frame must
be constructed by three columns with specific headings (see examples):
</p>
 <ul>
<li>
<p> A cell type column with the same name of the cell type
column in cells metadata (<code>cell.type.column</code>). If the name of the
column is not the same, the function will return an error. All cell types
must appear in the cells metadata. </p>
</li>
<li>
<p> A second column called
<code>'from'</code> with the start frequency for each cell type. </p>
</li>
<li>
<p> A third
column called <code>'to'</code> with the ending frequency for each cell type.</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.bulk.samples</code></td>
<td>
<p>Number of bulk RNA-Seq sample proportions (and thus
simulated bulk RNA-Seq samples) to be generated taking into account
training and test data. We recommend seting this value according to the
number of single-cell profiles available in
<code>DigitalDLSorter</code> object avoiding an excesive
re-sampling, but generating a large number of samples for better training.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.cells</code></td>
<td>
<p>Number of cells that will be aggregated in order to simulate
one bulk RNA-Seq sample (100 by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>train.freq.cells</code></td>
<td>
<p>Proportion of cells used to simulate training
pseudo-bulk samples (2/3 by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>train.freq.bulk</code></td>
<td>
<p>Proportion of bulk RNA-Seq samples to the total number
(<code>num.bulk.samples</code>) used for the training set (2/3 by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>proportion.method</code></td>
<td>
<p>Vector of six integers that determines the
proportions of bulk samples generated by the different methods (see Details
and Torroja and Sanchez-Cabo, 2019. for more information). This vector
represents proportions, so its entries must add up 100. By default, a
majority of random samples will be generated without using predefined
ranges.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prob.sparsity</code></td>
<td>
<p>It only affects the proportions generated by the first
method (Dirichlet distribution). It determines the probability of having
missing cell types in each simulated spot, as opposed to a mixture of all
cell types. A higher value for this parameter will result in more sparse
simulated samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.zero.prop</code></td>
<td>
<p>This parameter controls the minimum number of cell types
that will be absent in each simulated spot. If <code>NULL</code> (by default),
this value will be half of the total number of different cell types, but
increasing it will result in more spots composed of fewer cell types. This
helps to create more sparse proportions and cover a wider range of
situations during model training.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>balanced.type.cells</code></td>
<td>
<p>Boolean indicating whether the training and test
cells will be split in a balanced way considering the cell types
(<code>FALSE</code> by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Show informative messages during the execution (<code>TRUE</code> by
default).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>First, the available single-cell profiles are split into training and test
subsets (2/3 for training and 1/3 for test by default (see
<code>train.freq.cells</code>)) to avoid falsifying the results during model
evaluation. Next, <code>num.bulk.samples</code> bulk samples proportions are built
and the single-cell profiles to be used to simulate each pseudo-bulk RNA-Seq
sample are set, being 100 cells per bulk sample by default (see
<code>n.cells</code> argument). The proportions of training and test pseudo-bulk
samples are set by <code>train.freq.bulk</code> (2/3 for training and 1/3 for
testing by default). Finally, in order to avoid biases due to the composition
of the pseudo-bulk RNA-Seq samples, cell type proportions (<code class="reqn">w_1,...,w_k</code>,
where <code class="reqn">k</code> is the number of cell types available in single-cell profiles)
are randomly generated by using six different approaches:
</p>
 <ol>
<li>
<p> Cell proportions are randomly sampled from a truncated
uniform distribution with predefined limits according to a priori knowledge
of the abundance of each cell type (see <code>prob.design</code> argument). This
information can be inferred from the single-cell experiment itself or from
the literature. </p>
</li>
<li>
<p> A second set is generated by randomly permuting cell
type labels from a distribution generated by the previous method. </p>
</li>
<li>
<p> Cell
proportions are randomly sampled as by method 1 without replacement. </p>
</li>
<li>
<p>Using the last method for generating proportions, cell types labels are
randomly sampled. </p>
</li>
<li>
<p> Cell proportions are randomly sampled from a
Dirichlet distribution. </p>
</li>
<li>
<p> Pseudo-bulk RNA-Seq samples composed of the
same cell type are generated in order to provide 'pure' pseudo-bulk samples.</p>
</li>
</ol>
<p>If you want to inspect the distribution of cell type proportions generated by
each method during the process, they can be visualized by the
<code>showProbPlot</code> function (see Documentation).
</p>


<h3>Value</h3>

<p>A <code>DigitalDLSorter</code> object with
<code>prob.cell.types</code> slot containing a <code>list</code> with two
<code>ProbMatrixCellTypes</code> objects (training and test). For
more information about the structure of this class, see
<code>?ProbMatrixCellTypes</code>.
</p>


<h3>References</h3>

<p>Torroja, C. and SÃ¡nchez-Cabo, F. (2019). digitalDLSorter: A Deep
Learning algorithm to quantify immune cell populations based on scRNA-Seq
data. Frontiers in Genetics 10, 978. doi: <a href="https://doi.org/10.3389/fgene.2019.00978">doi:10.3389/fgene.2019.00978</a>
</p>


<h3>See Also</h3>

<p><code>simBulkProfiles</code>
<code>ProbMatrixCellTypes</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(123) # reproducibility
# simulated data
sce &lt;- SingleCellExperiment::SingleCellExperiment(
  assays = list(
    counts = matrix(
      rpois(30, lambda = 5), nrow = 15, ncol = 10, 
      dimnames = list(paste0("Gene", seq(15)), paste0("RHC", seq(10)))
    )
  ),
  colData = data.frame(
    Cell_ID = paste0("RHC", seq(10)),
    Cell_Type = sample(x = paste0("CellType", seq(2)), size = 10, 
                       replace = TRUE)
  ),
  rowData = data.frame(
    Gene_ID = paste0("Gene", seq(15))
  )
)
DDLS &lt;- createDDLSobject(
  sc.data = sce,
  sc.cell.ID.column = "Cell_ID",
  sc.gene.ID.column = "Gene_ID",
  sc.filt.genes.cluster = FALSE, 
  sc.log.FC = FALSE
)
probMatrixValid &lt;- data.frame(
  Cell_Type = paste0("CellType", seq(2)),
  from = c(1, 30),
  to = c(15, 70)
)
DDLS &lt;- generateBulkCellMatrix(
  object = DDLS,
  cell.ID.column = "Cell_ID",
  cell.type.column = "Cell_Type",
  prob.design = probMatrixValid,
  num.bulk.samples = 10,
  verbose = TRUE
)

</code></pre>


</div>