<div class="container">

<table style="width: 100%;"><tr>
<td>DIME-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
DIME (Differential Identification using Mixtures Ensemble) 
</h2>

<h3>Description</h3>

<p>A robust differential identification method that considers an ensemble of finite 
mixture models combined with a local false discovery rate <em>(fdr)</em> for 
analyzing ChIP-seq data comparing two samples. <br>
This package can also be used to identify differential
in other high throughput data such as microarray, methylation etc.<br>
After normalization, an Exponential-Normal(k) or a Uniform-Normal(k) mixture is
fitted to the data. The (k)-normal component can represent either differential
regions or non-differential regions depending on their location and spread. The
exponential or uniform represent differentially sites. local <em>(fdr)</em> are
computed from the fitted model.
Unique features of the package: 
</p>
<ol>
<li>
<p>Accurate modeling of data that comes from 
any distribution by the use of multiple normal components (any distribution can be
accurately represented by mixture of normal).
</p>
</li>
<li>
<p>Using ensemble of mixture models
allowing data to be accurately &amp; efficiently represented. Then two-phase selection ensure the 
selection of the best overall model. 
</p>
</li>
<li>
<p>This method can be used as a general
program to fit a mixture of uniform-normal or uniform-k-normal or exponential-k-normal 
</p>
</li>
</ol>
<h3>Details</h3>


<table>
<tr>
<td style="text-align: left;">
Package: </td>
<td style="text-align: left;"> DIME</td>
</tr>
<tr>
<td style="text-align: left;">
Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
Version: </td>
<td style="text-align: left;"> 1.0</td>
</tr>
<tr>
<td style="text-align: left;">
Date: </td>
<td style="text-align: left;"> 2010-11-19</td>
</tr>
<tr>
<td style="text-align: left;">
License: </td>
<td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
<td style="text-align: left;">
LazyLoad: </td>
<td style="text-align: left;"> yes</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Cenny Taslim <a href="mailto:taslim.2@osu.edu">taslim.2@osu.edu</a>, with contributions 
from Abbas Khalili <a href="mailto:khalili@stat.ubc.ca">khalili@stat.ubc.ca</a>, 
Dustin Potter <a href="mailto:potterdp@gmail.com">potterdp@gmail.com</a>, and 
Shili Lin <a href="mailto:shili@stat.osu.edu">shili@stat.osu.edu</a><br>
Maintainer: Cenny Taslim <a href="mailto:taslim.2@osu.edu">taslim.2@osu.edu</a> or 
Shili Lin  <a href="mailto:shili@stat.osu.edu">shili@stat.osu.edu</a>
</p>


<h3>References</h3>


<ul>
<li>
<p>Khalili, A., Huang, T., and Lin, S. (2009). A robust unified approach to 
analyzing methylation and gene expression data. <em>Computational Statistics 
and Data Analysis</em>, 53(5), 1701-1710.
</p>
</li>
<li>
<p>Dean, N. and Raftery, A. E. (2005). Normal uniform mixture differential 
gene expression detection for cDNA microarrays. <em>BMC Bioinformatics</em>, 6, 173.
</p>
</li>
<li>
<p>Taslim, C., Wu, J., Yan, P., Singer, G., Parvin, J., Huang, T., Lin, S., 
and Huang, K. (2009). Comparative study on chip-seq data: normalization and 
binding pattern characterization. <em>Bioinformatics</em>, 25(18), 2334-2340.
</p>
</li>
</ul>
</div>