<div class="container">

<table style="width: 100%;"><tr>
<td>peelingTwoIterate</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>A Function to Apply the Peeling Algorithm for Two Copy Number Matrices</h2>

<h3>Description</h3>

<p>This function iteratively applies the peelingTwo function, thereby identifying multiple
</p>


<h3>Usage</h3>

<pre><code class="language-R">peelingTwoIterate(
  X,
  Y,
  posDT,
  gain = TRUE,
  nullDist = NULL,
  threshold = NULL,
  numIters = 5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>A matrix of normalized gene-level copy number data (rows = genes, columns = subjects).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>A matrix of normalized gene-level copy number data (rows = genes, columns = subjects).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>posDT</code></td>
<td>
<p>A data frame containing genomic position information for the genes in X.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gain</code></td>
<td>
<p>A logical value indicating whether gains (TRUE) or losses (FALSE) will be peeled.
</p>
<p>Default = TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nullDist</code></td>
<td>
<p>An empirical null distribution produced by the cyclic shift algorithm.
</p>
<p>Default = NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>A tuning parameter that controls the size of the peeled region.  Rows of X
</p>
<p>and Y with mean copy number differences less than threshold will not be peeled.  Default = NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>numIters</code></td>
<td>
<p>The number of times peelingTwo will be iterated.  Default = 5.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>differences across the genome between a two cohorts.  Gains and losses should be analyzed separately.
</p>
<p>The <code>peelingTwo</code> function applies the peeling procedure for two cohorts to "nullify"
entries in two copy number matrices X and Y that give rise to the most significant copy number
difference.  Because tumor genomes may contain multiple regions that harbor copy number differences,
it is important to apply the peeling procedure for two cohorts iteratively, thereby identifying
multiple markers and surrounding genomic regions.
</p>


<h3>Value</h3>

<p>A list containing two elements: X, Y, and interval.  X and Y are updated versions of the input
copy number matrices in which the peak difference at k has been removed, and interval is genomic region
containing k.  By construction, interval cannot extend beyond the chromosome arm containing k.
</p>


<h3>Examples</h3>

<pre><code class="language-R">luad=pD[["X"]]

 lusc=pD[["Y"]]

 posDT=pD[["posDT"]]

 gain = TRUE

 nullDist = NULL

 threshold = NULL

 numIters = 3

 out=peelingTwoIterate(X=luad,Y=lusc,posDT=posDT,gain=TRUE,nullDist=NULL,threshold=NULL,numIters=3)   

</code></pre>


</div>