<div class="container">

<table style="width: 100%;"><tr>
<td>peeling</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Apply the peeling procedure at a given marker</h2>

<h3>Description</h3>

<p>Apply the peeling procedure at a given marker
</p>


<h3>Usage</h3>

<pre><code class="language-R">peeling(x, marker.data, cytoband, k)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An n by m numeric matrix containing DNA copy number data from n subjects at m markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker.data</code></td>
<td>
<p>marker.data A two-column numeric matrix of marker position data for markers in the
autosomes.  Column 1 contains the chromosome number for each marker, and column 2 contains the position
(in base pairs) for each markers.  This is a submatrix of the marker position matrix used by
<code>quickLook</code> and <code>detailedLook</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cytoband</code></td>
<td>
<p>A character vector of length m that contains the chromosome arm (p or q) for each
marker.  This is produced by the <code>makeCytoband</code> function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>A positive integer between 1 and m that represents the most aberrant marker.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The peeling procedure is detailed in Algorithm 2 of Bioinformatics (2011) 27(5) 678 - 685, but here
we provide a brief overview.  By construction, marker <code>k</code> represents the most aberrant gain (loss).
The peeling procedure rescales all copy number values in <code>x</code> that contribute to making marker <code>k</code>
aberrant, so that after applying the peeling procedure marker <code>k</code> is "null."  By construction, the
rescaling procedure is restricted to entries in <code>x</code> that correspond to markers in the same chromosome
arm as <code>k</code>.  This allows users to assess the statistical significance of multiple gains (losses) throughout
the genome.
</p>


<h3>Value</h3>

<p>A list containing two components:  (1) the n by m matrix produced by applying the peeling algorithm
to the matrix <code>x</code> at marker <code>k</code>, and (2) the peak interval around marker <code>k</code>, as described
in Bioinformatics (2011) 27(5) 678 - 685.
</p>


</div>