<div class="container">

<table style="width: 100%;"><tr>
<td>exampleData</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
<img src="../help/figures/DiPalm.png" alt="">

Example Data: Data for use with the DiPALM vignette
</h2>

<h3>Description</h3>

<p>A list of data objects that are used to run through the DiPALM example vignette.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data("exampleData")</code></pre>


<h3>Format</h3>

<p>A List of 3 objects:
</p>

<ol>
<li>
<p>$rawCounts - A data frame [42383 rows X 48 columns] containing raw counts for mRNA expression data on <em>Brassica rapa</em> R500.
</p>

<ul>
<li>
<p>rows : Each row represents the expression of one gene across many different samples.
</p>
</li>
<li>
<p>columns: Each column represents a different sample. Both drought-treated and properly watered samples are present, with two replicates of each time-course. Tissue was sampled at 4-hour intervals and column names are encoded as follows:
S[replicate number] [D-Drought / W-Watered] [Time - sample]
Example: S2D10 is part of the second replcate of the drought data and is the 10th time-point (ZT13, 13th hour after dawn on day two).
</p>
</li>
</ul>
</li>
<li>
<p>$moduleEigengenes - A dataframe [12 rows and 90 columns] representing 90 different expression vectors that describe descrete expression patterns found in the whole dataset ($timeCourseList).
</p>

<ul>
<li>
<p>Rows: each row represents a single time-point.
</p>
</li>
<li>
<p>Columns: each column represents a distinct expression pattern seen repeatedly in the full data.
</p>
</li>
</ul>
</li>
<li>
<p>$geneAnnotations - A data frame [42383 rows and 5 columns] giving the locations within the <em>Brassica rapa</em> R500 genome where each gene is found. This is in SAF format which is used by the Linux Subread software package.
</p>

<ul>
<li>
<p>rows: Each row represents a single gene.
</p>
</li>
<li>
<p>$GeneID: (character string) Unique gene accession.  
</p>
</li>
<li>
<p>$Chr: (character string) Chromosome number.  
</p>
</li>
<li>
<p>$Start: (integer) The position where the gene starts (first basepair).  
</p>
</li>
<li>
<p>$End: (integer) The position where the gene ends (last basepair).  
</p>
</li>
<li>
<p>$Strand: (character) is the gene on the positive [+] or negative [-] strand.  
</p>
</li>
</ul>
</li>
</ol>
<h3>Details</h3>

<p>Raw count data was summarized by mapping RNA-seq counts to the R500 <em>Brassica rapa</em> genome using the hisat2 aligner. Mapped reads were counted using the featureCounts function from the Linux Subread software package.
</p>


<h3>Source</h3>

<p>Data is from: <a href="https://elifesciences.org/articles/29655">Greenham et al., eLife 2017;6:e29655</a>
</p>


</div>