<div class="container">

<table style="width: 100%;"><tr>
<td>BuildModMembership</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
<img src="../help/figures/DiPalm.png" alt="">

Calculate module membership (kME) for all genes in a time-course data set
</h2>

<h3>Description</h3>

<p>This function calculates module membership (kME) between each gene's expression vector throughout a time-course and the module eigengenes of the data set (from the WGCNA package, using <code>blockwiseModules</code>). Module membership is defined as the Pearson correlation between the expression of a single gene and the expression of a single eigengene. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">BuildModMembership(MeMat, TCsLst)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>MeMat</code></td>
<td>

<p>A data frame of eigengenes with each column being a different eigengene vector.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>TCsLst</code></td>
<td>

<p>A named list of time-course matrices. Each element of the list is a time-course expression matrix (genes as rows and time-points as columns).
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The matrix of eigengene vectors, <code>MeMat</code> (see <code>blockwiseModules</code> from the WGCNA package) is used as a convenient way to summarize all the unique expression patterns (across time) that are present in a time-course data set. 
Any large gene expression data set from a complex organism will contain tens of thousands of genes. However, extensive coordinated regulation is present in all biological systems. Because of this co-regulation, usually only dozens of distinct expression patterns exist and most genes can be categorized by one (or a few) of them.
By using a relatively small set of descriptive expression vectors (eigengenes), we can "encode" the pattern of expression for any single gene into a length N vector where N is the number of eigengenes and the values of the vector are the Pearson correlation of the gene to each eigengene (kME).
These kME values can then be compared using the <code>limma</code> package to determine differential patterning similar to how expression values are compared to determine differential expression.
This function is the first step to calculate all kMEs for each gene in a list of time-course matrices relative to the set of eigengenes for the whole data set.
</p>


<h3>Value</h3>

<p>This function returns a list of kME matrices. The list contains a separate element for each eigengene. Each element is a matrix with a row for each gene and a column for each member of the input <code>TCsLst</code>. 
</p>


<h3>Author(s)</h3>

<p>Ryan C. Sartor
</p>


<h3>See Also</h3>

<p><code>blockwiseModules</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("testData")
kMEsList&lt;-BuildModMembership(MeMat=testData$moduleEigengenes, TCsLst=testData$timeCourseList)
</code></pre>


</div>