<div class="container">

<table style="width: 100%;"><tr>
<td>make_blast_DB</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Builds local database for BLAST</h2>

<h3>Description</h3>

<p>Builds a BLAST database with local sequences using FASTA file.
</p>


<h3>Usage</h3>

<pre><code class="language-R">make_blast_DB(
  makeblastdb.way,
  fasta.way,
  db.way,
  db.type = "nucl",
  db.title,
  delete.fasta = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>makeblastdb.way</code></td>
<td>
<p>character; name and path to makeblastdb executable file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fasta.way</code></td>
<td>
<p>character; name and path to FASTA file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>db.way</code></td>
<td>
<p>character; name and path to local BLAST database</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>db.type</code></td>
<td>
<p>character; type of BLAST database</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>db.title</code></td>
<td>
<p>character; BLAST data base title</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>delete.fasta</code></td>
<td>
<p>logical; delete FASTA file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical; show messages</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is using BLAST applications. BLAST+ (UNIX or Windows) should be installed.
</p>


<h3>Value</h3>

<p>The function creates local BLAST data base. No additional data is returned.
</p>


<h3>Author(s)</h3>

<p>Elena N. Filatova
</p>


<h3>References</h3>

<p>Camacho C., Coulouris G., Avagyan V. et al. (2009). BLAST+: architecture and applications. BMC Bioinformatics 10, 421.
<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-10-421">https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-10-421</a>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# This function is using BLAST applications. BLAST+ should be installed.
# FASTA file with sequences for local data base should be downloaded first (see get_seq_for_DB ())
path &lt;- tempdir()
dir.create (path)
# load metadata for target sequences of Chlamydia pneumoniae
(meta.target)
# load sequences, it will take about 3 minutes
get_seq_for_DB (ids = meta.target$gi, db = "nucleotide", check.result = TRUE,
              return="fasta", fasta.file = paste0 (path, "/seq.fasta"), verbose = TRUE)
# create local data base, it will take 0.235217 seconds
make_blast_DB (makeblastdb.way = "D:/Blast/blast-2.11.0+/bin/makeblastdb.exe",
              fasta.way = paste0 (path, "/seq.fasta"), db.title = "Cl_pneumoniae",
              db.way = paste0 (path, "/DB"), db.type = "nucl", delete.fasta = FALSE)
# delete FASTA file (also can set delete.fasta = TRUE)
file.remove (paste0 (path, "/seq.fasta"))

## End(Not run)

</code></pre>


</div>