<div class="container">

<table style="width: 100%;"><tr>
<td>run_pcor</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Wrapper for partial correlations from corpcor</h2>

<h3>Description</h3>

<p>Conducts co-expression analysis using full partial correlations; these are
computed using the shrinkage approach for covariance estimation
(Schäfer and Strimmer 2005) from the
<code>corpcor</code> package (Schafer et al. 2017).
Can be used for the <code>network_inference</code> argument in <code>dnapath</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">run_pcor(x, weights = NULL, ranks = FALSE, verbose = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A n by p matrix of gene expression data (n samples and p genes).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p>An optional vector of weights. This is used by <code>dnapath()</code> to
apply the probabilistic group labels to each observation when estimating the
group-specific network.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ranks</code></td>
<td>
<p>If TRUE, the gene expression values will be converted to ranks
(across samples) prior to covariance estimation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Argument is passed into <code>pcor.shrink</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A p by p matrix of association scores.
</p>


<h3>References</h3>

<p>Schäfer J, Strimmer K (2005).
“A Shrinkage Approach to Large-Scale Covariance Matrix Estimation and Implications for Functional Genomics.”
<em>Statistical Applications in Genetics and Molecular Biology</em>, <b>4</b>(1), Article 32.
</p>
<p>Schafer J, Opgen-Rhein R, Zuber V, Ahdesmaki M, Silva APD, Strimmer. K (2017).
<em>corpcor: Efficient Estimation of Covariance and (Partial) Correlation</em>.
R package version 1.6.9, <a href="https://CRAN.R-project.org/package=corpcor">https://CRAN.R-project.org/package=corpcor</a>.
</p>


<h3>See Also</h3>

<p><code>run_aracne</code>,
<code>run_bc3net</code>, <code>run_c3net</code>,
<code>run_clr</code>, <code>run_corr</code>,
<code>run_dwlasso</code>, <code>run_genie3</code>,
<code>run_glasso</code>, <code>run_mrnet</code>, and <code>run_silencer</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(meso)
data(p53_pathways)

# To create a short example, we subset on two pathways from the p53 pathway list,
# and will only run 3 permutations for significance testing.
pathway_list &lt;- p53_pathways[c(8, 13)]
n_perm &lt;- 3

# Use this method to perform differential network analysis.
results &lt;- dnapath(x = meso$gene_expression,
                   pathway_list = pathway_list,
                   group_labels = meso$groups,
                   n_perm = n_perm,
                   network_inference = run_pcor)
summary(results)

# The group-specific association matrices can be extracted using get_networks().
nw_list &lt;- get_networks(results[[1]]) # Get networks for pathway 1.


# nw_list has length 2 and contains the inferred networks for the two groups.
# The gene names are the Entrezgene IDs from the original expression dataset.
# Renaming the genes in the dnapath results to rename those in the networks.
# NOTE: The temporary directory, tempdir(), is used in this example. In practice,
#       this argument can be removed or changed to an existing directory
results &lt;- rename_genes(results, to = "symbol", species = "human",
                        dir_save = tempdir())
nw_list &lt;- get_networks(results[[1]]) # The genes (columns) will have new names.

# (Optional) Plot the network using SeqNet package (based on igraph plotting).
# First rename entrezgene IDs into gene symbols.
SeqNet::plot_network(nw_list[[1]])

</code></pre>


</div>