<div class="container">

<table style="width: 100%;"><tr>
<td>likelihood</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Negative Log Likelihood Ratio</h2>

<h3>Description</h3>

<p>This function calculates log likelihood ratio value.</p>


<h3>Usage</h3>

<pre><code class="language-R">likelihood(thetast, fobs, merge2)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>thetast</code></td>
<td>
<p>a starting values for the parameter we need to estimate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fobs</code></td>
<td>
<p>the <code class="reqn">4^K</code> joint distribution array for K edge tree</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>merge2</code></td>
<td>
<p>a (K-1) x 2 matrix describing the tree topology</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function calculates the log likelihood ratio value for F(t). It needs
a vector of starting values for the parameters estimate, <code class="reqn">4^K</code> observed divergence array and merge matrix describing the tree topology.</p>


<h3>Value</h3>

<p>The value of the log likelihood ratio</p>


<h3>References</h3>

<p>Faisal Ababneh, Lars S Jermiin, Chunsheng Ma, John Robinson (2006). Matched-pairs tests of homogeneity with applications to homologous nucleotide sequences. Bioinformatics, 22(10), 1225-1231.
</p>


<h3>See Also</h3>

<p>gn, gn2</p>


<h3>Examples</h3>

<pre><code class="language-R">merge2&lt;-matrix(c(-1,-4,-3,2,-2,-5,1,3), 4, 2)
theta&lt;-c(rep(.25,3), rep(.25,3), rep(.25,3), c(.2,.35,.79,.01,.93,.47),3,.1,.5,.8)
F1&lt;-gn(theta, merge2)
lh&lt;-likelihood(theta, F1, merge2)
lh
</code></pre>


</div>