<div class="container">

<table style="width: 100%;"><tr>
<td>Distance</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Paralinear Distances</h2>

<h3>Description</h3>

<p>This function calculates the paralinear distance between K matched DNA sequences.</p>


<h3>Usage</h3>

<pre><code class="language-R">Distance(F4)</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>F4</code></td>
<td>
<p>a <code class="reqn">4^K</code> array containing the joint distribution array <code class="reqn">F(t)</code> or the observed array N</p>
</td>
</tr></table>
<h3>Details</h3>

<p>This function calculates the paralinear distances between K matched DNA sequences, depending on the joint distribution array for these K sequences or on the observed divergence array N.</p>


<h3>Value</h3>

<p>A K x K symmetric matrix distances between the K sequences</p>


<h3>References</h3>

<p>Faisal Ababneh, Lars S Jermiin, Chunsheng Ma, John Robinson (2006). Matched-pairs tests of homogeneity with applications to homologous nucleotide sequences. Bioinformatics, 22(10), 1225-1231.
</p>


<h3>See Also</h3>

<p>gn2, gn, Fmatrix, Ntml</p>


<h3>Examples</h3>

<pre><code class="language-R">merge2&lt;-matrix(c(-1,-4,-3,2,-2,-5,1,3), 4, 2)
theta&lt;-c(rep(.25,3), rep(.25,3), rep(.25,3), c(.2,.35,.79,.01,.93,.47), 3,.1,.5,.8)
F1&lt;-gn(theta,merge2)
dn&lt;-Distance(F1)
dn
</code></pre>


</div>