<div class="container">

<table style="width: 100%;"><tr>
<td>NMFpval</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>P value for discriminant Non-Negative Matrix Factorization</h2>

<h3>Description</h3>

<p>Estimate the significance of differentially expressed genes in parallel.
</p>


<h3>Usage</h3>

<pre><code class="language-R">NMFpval(
  nmf_res,
  np = 100,
  ncores = parallel::detectCores(),
  fdr = FALSE,
  top = 1000,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nmf_res</code></td>
<td>
<p>result from DNMF or dNMF</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>np</code></td>
<td>
<p>number of permutations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p>cores used. Default is all the availiable cores</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fdr</code></td>
<td>
<p>false discovery rate. Default is FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>top</code></td>
<td>
<p>only include top ranked genes. Default is 1000</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>verbose</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>P value is caculated based on aatricle, Wang, Hong-Qiang, Chun-Hou Zheng, and Xing-Ming Zhao. 
"jNMFMA: a joint non-negative matrix factorization meta-analysis of
transcriptomics data." Bioinformatics (2014): btu679.
</p>


<h3>Value</h3>

<p>a matrix with columns rnk, p (and fdr)
</p>


<h3>Author(s)</h3>

<p>Zhilong Jia
</p>


<h3>Examples</h3>

<pre><code class="language-R">dat &lt;- rbind(matrix(c(rep(3, 16), rep(8, 24)), ncol=5), 
matrix(c(rep(5, 16), rep(5, 24)), ncol=5), 
matrix(c(rep(18, 16), rep(7, 24)), ncol=5)) + 
matrix(runif(120,-1,1), ncol=5)
trainlabel &lt;- c(1,1,2,2,2)

nmf_res &lt;- ndNMF(dat, trainlabel, r=2, lambada = 0.1)
pMat &lt;- NMFpval(nmf_res, np=10, ncores=2, top=4)

</code></pre>


</div>