<div class="container">

<table style="width: 100%;"><tr>
<td>DoEstRare</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>DoEstRare</h2>

<h3>Description</h3>

<p>Rare variant association test comparing position density functions and mutation counts between cases and controls.
</p>


<h3>Usage</h3>

<pre><code class="language-R">DoEstRare(pheno, geno, position, genome.size,
          perm=NULL, alpha=NULL, c=NULL,
          autosomal=TRUE, gender=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p>a numeric vector of phenotypes. Affected individuals are coded 1 and unaffected individuals are coded 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>a numeric matrix of genotypes (row: individual, column: variant). Genotypes are coded 0,1 or 2 corresponding to the number of minor alleles. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>position</code></td>
<td>
<p>a numeric vector of variant positions. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome.size</code></td>
<td>
<p>a numeric value corresponding to the length of the analyzed region. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perm</code></td>
<td>
<p>number of permutations. If not NULL,a "standard permutation procedure" is performed to compute the significance. See Details. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>error level. If not NULL,an "adaptive permutation procedure" is performed to compute the significance. See Details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>c</code></td>
<td>
<p>precision of the p-value. If not NULL,an "adaptive permutation procedure" is performed to compute the significance. See Details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>autosomal</code></td>
<td>
<p>boolean. If TRUE, autosomal chromosome; FALSE, X chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gender</code></td>
<td>
<p>numeric vector. 1=male; 2=female.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Two types of permutations procedures can be defined in the function: the standard permutation procedure and the adaptive permutation procedure.
</p>
<p>In the standard permutation procedure, the user specifies, in the argument "perm", the number of permutations to be done. The p-value will be <code class="reqn">(R+1)(B+1)</code>. With <code class="reqn">R</code> the number of permutation statistics superior to the observed statistic and <code class="reqn">B</code> the number of permutations.
</p>
<p>In the adaptive permutation procedure, the user specifies, in the argument "alpha", the significance to achieve after multiple testing correction. In the argument "c", the estimation precision of the p-value. In function of these two paremeters, the maximal number of permutations and the maximal number of success to achieve will be computed. If the maximal number of success is reached, the p-value will be <code class="reqn">R/B</code>. If not, the maximal number of permutations will be used to compute the p-value <code class="reqn">(R+1)(B+1)</code>.
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>p.value</code></td>
<td>
<p>the p-value obtained by the phenotype permutation procedure. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stat</code></td>
<td>
<p>the test statistic. </p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Elodie Persyn, <a href="mailto:elodie.persyn@univ-nantes.fr">elodie.persyn@univ-nantes.fr</a></p>


<h3>References</h3>

<p>Persyn E, Karakachoff M, Le Scouarnec S, Le Cl??zio C, Campion D, French Exome Consortium, et al. DoEstRare: A statistical test to identify local enrichments in rare genomic variants associated with disease. Wang K, editor. PLOS ONE. 2017 Jul 24;12(7):e0179364.
</p>
<p>Che R, Jack JR, Motsinger-Reif AA, Brown CC. An adaptive permutation approach for genome-wide association study: evaluation and recommendations for use. BioData Min. 2014;7:9.
</p>


<h3>Examples</h3>

<pre><code class="language-R">pheno=rep(c(0,1), 500)
geno=matrix(sample(c(0,1),prob=c(0.7,0.3) ,1000*30, replace=TRUE), ncol=30)
position=sample(1:500,30)
genome.size=500
perm=200

#Autosomal gene
#standard phenotype permutation procedure
DoEstRare(pheno, geno, position, genome.size, perm)
#adaptive phenotype permutation procedure
DoEstRare(pheno, geno, position, genome.size, alpha=0.05, c=0.2)

#X gene
gender=rep(c(1,2), each=500)
#standard phenotype permutation procedure
DoEstRare(pheno, geno, position, genome.size, perm, autosomal=FALSE, gender=gender)
#adaptive phenotype permutation procedure
DoEstRare(pheno, geno, position, genome.size, alpha=0.05, c=0.2, autosomal=FALSE, gender=gender)


</code></pre>


</div>