<div class="container">

<table style="width: 100%;"><tr>
<td>getSubclones</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>#' Deprecated! Use resolveLightChains</h2>

<h3>Description</h3>

<p><code>getSubClones</code> plots a tree or group of trees
</p>


<h3>Usage</h3>

<pre><code class="language-R">getSubclones(
  heavy,
  light,
  nproc = 1,
  minseq = 1,
  id = "sequence_id",
  seq = "sequence_alignment",
  clone = "clone_id",
  cell = "cell_id",
  v_call = "v_call",
  j_call = "j_call",
  junc_len = "junction_length",
  nolight = "missing"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>heavy</code></td>
<td>
<p>a tibble containing heavy chain sequences with clone_id</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>light</code></td>
<td>
<p>a tibble containing light chain sequences</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nproc</code></td>
<td>
<p>number of cores for parallelization</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minseq</code></td>
<td>
<p>minimum number of sequences per clone</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p>name of the column containing sequence identifiers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p>name of the column containing observed DNA sequences. All 
sequences in this column must be multiple aligned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clone</code></td>
<td>
<p>name of the column containing the identifier for the clone. All 
entries in this column should be identical.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cell</code></td>
<td>
<p>name of the column containing identifier for cells.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_call</code></td>
<td>
<p>name of the column containing V-segment allele assignments. All 
entries in this column should be identical to the gene level.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>j_call</code></td>
<td>
<p>name of the column containing J-segment allele assignments. All 
entries in this column should be identical to the gene level.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>junc_len</code></td>
<td>
<p>name of the column containing the length of the junction as a 
numeric value. All entries in this column should be identical 
for any given clone.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nolight</code></td>
<td>
<p>string to use to indicate a missing light chain</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a tibble containing
</p>


</div>