<div class="container">

<table style="width: 100%;"><tr>
<td>createGermlines</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>createGermlines Determine consensus clone sequence and create germline for clone</h2>

<h3>Description</h3>

<p>createGermlines Determine consensus clone sequence and create germline for clone
</p>


<h3>Usage</h3>

<pre><code class="language-R">createGermlines(
  data,
  references,
  locus = "locus",
  nproc = 1,
  seq = "sequence_alignment",
  v_call = "v_call",
  d_call = "d_call",
  j_call = "j_call",
  amino_acid = FALSE,
  id = "sequence_id",
  clone = "clone_id",
  v_germ_start = "v_germline_start",
  v_germ_end = "v_germline_end",
  v_germ_length = "v_germline_length",
  d_germ_start = "d_germline_start",
  d_germ_end = "d_germline_end",
  d_germ_length = "d_germline_length",
  j_germ_start = "j_germline_start",
  j_germ_end = "j_germline_end",
  j_germ_length = "j_germline_length",
  np1_length = "np1_length",
  np2_length = "np2_length",
  na.rm = TRUE,
  fields = NULL,
  verbose = 0,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>AIRR-table containing sequences from one clone</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>references</code></td>
<td>
<p>Full list of reference segments, see readIMGT</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locus</code></td>
<td>
<p>Name of the locus column in the input data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nproc</code></td>
<td>
<p>Number of cores to use</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p>Column name for sequence alignment</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_call</code></td>
<td>
<p>Column name for V gene segment gene call</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d_call</code></td>
<td>
<p>Column name for D gene segment gene call</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>j_call</code></td>
<td>
<p>Column name for J gene segment gene call</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>amino_acid</code></td>
<td>
<p>Perform reconstruction on amino acid sequence (experimental)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p>Column name for sequence ID</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clone</code></td>
<td>
<p>Column name for clone ID</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_germ_start</code></td>
<td>
<p>Column name of index of V segment start within germline</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_germ_end</code></td>
<td>
<p>Column name of index of V segment end within germline</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_germ_length</code></td>
<td>
<p>Column name of index of V segment length within germline</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d_germ_start</code></td>
<td>
<p>Column name of index of D segment start within germline</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d_germ_end</code></td>
<td>
<p>Column name of index of D segment end within germline</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d_germ_length</code></td>
<td>
<p>Column name of index of D segment length within germline</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>j_germ_start</code></td>
<td>
<p>Column name of index of J segment start within germline</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>j_germ_end</code></td>
<td>
<p>Column name of index of J segment end within germline</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>j_germ_length</code></td>
<td>
<p>Column name of index of J segment length within germline</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>np1_length</code></td>
<td>
<p>Column name in receptor specifying np1 segment length</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>np2_length</code></td>
<td>
<p>Column name in receptor specifying np2 segment length</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.rm</code></td>
<td>
<p>Remove clones with failed germline reconstruction?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fields</code></td>
<td>
<p>Character vector of additional columns to use for grouping. 
Sequences with disjoint values in the specified fields 
will be considered as separate clones.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>amount of rubbish to print</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional arguments passed to buildGermline</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Return object adds/edits following columns:
</p>

<ul>
<li>  <p><code>seq</code>:  Sequences potentially padded  same length as germline
</p>
</li>
<li>  <p><code>germline_alignment</code>: Full length germline
</p>
</li>
<li>  <p><code>germline_alignment_d_mask</code>: Full length, D region masked
</p>
</li>
<li>  <p><code>vonly</code>:   V gene segment of germline if vonly=TRUE
</p>
</li>
<li>  <p><code>regions</code>: String of VDJ segment in position if use_regions=TRUE
</p>
</li>
</ul>
<h3>Value</h3>

<p>Tibble with reconstructed germlines
</p>


<h3>See Also</h3>

<p>createGermlines buildGermline, stitchVDJ
</p>


<h3>Examples</h3>

<pre><code class="language-R">vdj_dir &lt;- system.file("extdata", "germlines", "imgt", "human", "vdj", package="dowser")
imgt &lt;- readIMGT(vdj_dir)
db &lt;- createGermlines(ExampleAirr[1,], imgt)
</code></pre>


</div>