<div class="container">

<table style="width: 100%;"><tr>
<td>maskSequences</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
<code>maskSequences</code> Mask codons split by insertions in V gene</h2>

<h3>Description</h3>

<p><code>maskSequences</code> Mask codons split by insertions in V gene
</p>


<h3>Usage</h3>

<pre><code class="language-R">maskSequences(
  data,
  sequence_id = "sequence_id",
  sequence = "sequence",
  sequence_alignment = "sequence_alignment",
  v_sequence_start = "v_sequence_start",
  v_sequence_end = "v_sequence_end",
  v_germline_start = "v_germline_start",
  v_germline_end = "v_germline_end",
  junction_length = "junction_length",
  keep_alignment = FALSE,
  keep_insertions = FALSE,
  mask_codons = TRUE,
  mask_cdr3 = TRUE,
  nproc = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>BCR data table</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sequence_id</code></td>
<td>
<p>sequence id column</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sequence</code></td>
<td>
<p>input sequence column (query)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sequence_alignment</code></td>
<td>
<p>aligned (IMGT-gapped) sequence column (subject)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_sequence_start</code></td>
<td>
<p>V gene start position in sequence</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_sequence_end</code></td>
<td>
<p>V gene end position in sequence</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_germline_start</code></td>
<td>
<p>V gene start position in sequence_alignment</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_germline_end</code></td>
<td>
<p>V gene end position in sequence_alignment</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>junction_length</code></td>
<td>
<p>name of junction_length column</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep_alignment</code></td>
<td>
<p>store alignment of query and subject sequences?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep_insertions</code></td>
<td>
<p>return removed insertion sequences?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mask_codons</code></td>
<td>
<p>mask split codons?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mask_cdr3</code></td>
<td>
<p>mask CDR3 sequences?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nproc</code></td>
<td>
<p>number of cores to use</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Performs global alignment of sequence and sequence_alignment, 
masking codons in sequence_alignment that are split by insertions (see examples)
masking_note notes codon positions in subject_alignment sequence that 
were masked, if found.
subject_alignment contains subject sequence aligned to query sequence (only 
if keep_alignment=TRUE)
query_alignment contains query sequence aligned to subject sequence (only if 
keep_alignment=TRUE)
sequence_masked will be NA if frameshift or alignment error detected. This 
will be noted
insertions column will be returned if keep_insertions=TRUE, contains a
comma-separated list of each &lt;position in query alignment&gt;-&lt;sequence&gt;. See example.
in masking_note.
</p>


<h3>Value</h3>

<p>A tibble with masked sequence in sequence_masked column, 
as well as other columns.
</p>


<h3>See Also</h3>

<p>maskCodons, Biostrings::pairwiseAlignment.
</p>


</div>