<div class="container">

<table style="width: 100%;"><tr>
<td>ExampleMixedDb</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Example Change-O database</h2>

<h3>Description</h3>

<p>A small example database subset from Turner, J. S. et al. Human germinal centres 
engage memory and naive B cells after influenza vaccination. Nature 586, 127â€“132 (2020).
</p>


<h3>Usage</h3>

<pre><code class="language-R">ExampleMixedDb
</code></pre>


<h3>Format</h3>

<p>A data.frame with the following Change-O style columns:
</p>

<ul>
<li>  <p><code>sequence_id</code>:           Sequence identifier
</p>
</li>
<li> <p><code>sequence</code>:               B cell sequence
</p>
</li>
<li> <p><code>productive</code>:             A logical indicating if the sequence is productive.
</p>
</li>
<li>  <p><code>v_call</code>:                V region allele assignments.
</p>
</li>
<li> <p><code>d_call</code>:                 D region allele assignments. 
</p>
</li>
<li>  <p><code>j_call</code>:                J region allele assignments.
</p>
</li>
<li>  <p><code>sequence_alignment</code>:    Sequence alignment.
</p>
</li>
<li> <p><code>germline_alignment</code>:     Germline alignment without gaps. 
</p>
</li>
<li> <p><code>junction</code>:               Junction
</p>
</li>
<li> <p><code>juncation_aa</code>:           Junction aa
</p>
</li>
<li> <p><code>vj_inframe</code>:             A logical to see if the vj genes are in frame
</p>
</li>
<li> <p><code>stop_codon</code>:             A indicator if there is a stop codon within the alignment
</p>
</li>
<li> <p><code>locus</code>:                  Locus identifier. 
</p>
</li>
<li> <p><code>v_sequence_start</code>:       Where the V gene starts
</p>
</li>
<li> <p><code>v_sequence_end</code>:         Where the V gene ends
</p>
</li>
<li> <p><code>v_germline_start</code>:       Where the V germline starts
</p>
</li>
<li> <p><code>v_germline_end</code>:         Where the V germline ends
</p>
</li>
<li> <p><code>np1_length</code>:             Length of np1
</p>
</li>
<li> <p><code>d_sequence_start</code>:       Where the D gene starts
</p>
</li>
<li> <p><code>d_sequence_end</code>:         Where the D gene ends
</p>
</li>
<li> <p><code>d_germline_start</code>:       Where the D germline starts
</p>
</li>
<li> <p><code>d_germline_end</code>:         Where the D germline ends
</p>
</li>
<li> <p><code>np2_length</code>:             Length of np2
</p>
</li>
<li> <p><code>j_sequence_start</code>:       Where the J gene starts
</p>
</li>
<li> <p><code>j_sequence_end</code>:         Where the J gene ends
</p>
</li>
<li> <p><code>j_germline_start</code>:       Where the J germline starts
</p>
</li>
<li> <p><code>j_germline_end</code>:         Where the J germline ends
</p>
</li>
<li>  <p><code>junction_length</code>:       Length of the junction region in nucleotides.
</p>
</li>
<li> <p><code>v_score</code>:                V score
</p>
</li>
<li> <p><code>v_identity</code>:             Identity score of V
</p>
</li>
<li> <p><code>v_support</code>:              V support
</p>
</li>
<li> <p><code>d_score</code>:                D score
</p>
</li>
<li> <p><code>d_identity</code>:             D identity 
</p>
</li>
<li> <p><code>d_support</code>:              D support
</p>
</li>
<li> <p><code>j_score</code>:                J score
</p>
</li>
<li> <p><code>j_support</code>:              J support
</p>
</li>
<li> <p><code>j_identity</code>:             J identity 
</p>
</li>
<li> <p><code>cell_id</code>:                Cell identifier 
</p>
</li>
<li> <p><code>consensus_count</code>:        Consensus count 
</p>
</li>
<li> <p><code>indels</code>:                 Logical if indels are present 
</p>
</li>
<li> <p><code>sequence_vdj</code>:           VDJ sequence
</p>
</li>
<li> <p><code>v_germ_start_vdj</code>:       Where the V germline starts on the VDJ
</p>
</li>
<li> <p><code>v_germ_end_vdj</code>:         Where the V germline ends on the VDJ
</p>
</li>
<li> <p><code>subject</code>:                Subject identifier 
</p>
</li>
<li> <p><code>timepoint</code>:              Day the sample was taken 
</p>
</li>
<li> <p><code>cell_type</code>:              Type of cell 
</p>
</li>
<li> <p><code>replicate</code>:              Replicate number 
</p>
</li>
<li>  <p><code>clone_id</code>:              Change-O assignment clonal group identifier.
</p>
</li>
<li> <p><code>seq_type</code>:               Identifier of data type (10x)
</p>
</li>
<li> <p><code>vj_gene</code>:                VJ gene
</p>
</li>
<li> <p><code>vj_alt_gene</code>:            Alternative VJ gene
</p>
</li>
<li> <p><code>v_germline_length</code>:      Length of the V germline segment
</p>
</li>
<li> <p><code>d_germline_length</code>:      Length of the D germline segment 
</p>
</li>
<li> <p><code>j_germline_lenght</code>:      Length of the J germline segment 
</p>
</li>
<li> <p><code>germline_alignment_d_mask</code>:  Germline alignment with gaps
</p>
</li>
</ul>
</div>