<div class="container">

<table style="width: 100%;"><tr>
<td>getBootstraps</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Creates a bootstrap distribution for clone sequence alignments, and returns  
estimated trees for each bootstrap replicate as a nested list as a new input 
tibble column.</h2>

<h3>Description</h3>

<p><code>getBootstraps</code> Phylogenetic bootstrap function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getBootstraps(
  clones,
  bootstraps,
  nproc = 1,
  bootstrap_nodes = TRUE,
  dir = NULL,
  id = NULL,
  build = "pratchet",
  exec = NULL,
  quiet = 0,
  rm_temp = TRUE,
  rep = NULL,
  seq = NULL,
  boot_part = "locus",
  by_codon = TRUE,
  starting_tree = FALSE,
  switches = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>clones</code></td>
<td>
<p>tibble <code>airrClone</code> objects, the output of 
formatClones</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bootstraps</code></td>
<td>
<p>number of bootstrap replicates to perform</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nproc</code></td>
<td>
<p>number of cores to parallelize computations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bootstrap_nodes</code></td>
<td>
<p>a logical if the the nodes for each tree in the trees
column (required) should report their bootstrap value</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dir</code></td>
<td>
<p>directory where temporary files will be placed (required
if <code>igphyml</code> or <code>dnapars</code> specified)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p>unique identifer for this analysis (required if 
<code>igphyml</code> or <code>dnapars</code> specified)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>build</code></td>
<td>
<p>program to use for tree building (phangorn, dnapars, igphyml)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exec</code></td>
<td>
<p>location of desired phylogenetic executable</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>amount of rubbish to print to console</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rm_temp</code></td>
<td>
<p>remove temporary files (default=TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rep</code></td>
<td>
<p>current bootstrap replicate (experimental)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p>column name containing sequence information</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>boot_part</code></td>
<td>
<p>is  "locus" bootstrap columns for each locus separately</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>by_codon</code></td>
<td>
<p>a logical if the user wants to bootstrap by codon or by 
nucleotide. Default (codon based bootstrapping) is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>starting_tree</code></td>
<td>
<p>An indicator to use the existing trees column as the starting trees for RAxML</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>switches</code></td>
<td>
<p>a logical indicator to allow findSwitches to do permutations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional arguments to be passed to tree building program</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The input clones tibble with an additional column for the bootstrap replicate trees.
</p>


</div>