<div class="container">

<table style="width: 100%;"><tr>
<td>gendata_RNAExp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate simulated data</h2>

<h3>Description</h3>

<p>Generate simulated spatial transcriptomics data or scRNAseq data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  gendata_RNAExp(height=30, width=30, platform="ST", p =100, q=10, K=7, 
                            G=4,sigma2=1, tau=8, seed=1, view=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>height, width</code></td>
<td>
<p>Height and width of lattice grids for generating spatial coordinates. n=height * width cells for scRNAseq data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>platform</code></td>
<td>
<p>set the platform for the simulated data, only support 'ST' and 'scRNAseq'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>number of genes to generate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>q</code></td>
<td>
<p>number of true latent features to generate gene expression</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>number of clusters (cell types).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>random seed for generate data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>G</code></td>
<td>
<p>the number of neighbors. The latter must be one of G = 4 or G = 8, which respectively correspond to a first order and a second order dependency structure. By default, G = 4.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma2</code></td>
<td>
<p>Variance of error term in probabilitic PCA model.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tau</code></td>
<td>
<p>a positive factor of mixture mean values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>view</code></td>
<td>
<p>Logical value indicating whether the draw should be printed. Do not display the optional borders.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>return a "Seurat" object. If <code>platform="ST"</code>, then the metadata of this Seurat object will include two columns with names "row" and "col" which are the spatial coordinates; If <code>platform="scRNAseq"</code>, then the metadata of this Seurat object will not have them.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p>None
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## we generate the spatial transcriptomics data with lattice neighborhood, i.e. ST platform.
seu &lt;- gendata_RNAExp(height=20, width=20,p=200, K=4)
seu

## generate scRNAseq data
seu &lt;- gendata_RNAExp(height=20, width=20, platform="scRNAseq", p=100, K=4)
seu
</code></pre>


</div>