<div class="container">

<table style="width: 100%;"><tr>
<td>FindSVGs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Find spatially variable genes</h2>

<h3>Description</h3>

<p>Identifies features that have spatially variation along spots using SPARK-X.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  FindSVGs(seu, nfeatures=2000, covariates=NULL,
             preHVGs=5000,num_core=1, verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seu</code></td>
<td>
<p>an object of class "Seurat".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nfeatures</code></td>
<td>
<p>a positive integer, means how many spatially variable genes to be chosen. If there are less than 2000 features in seu, then all features are identified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>covariates</code></td>
<td>
<p>a covariate matrix named control variable matrix whose number of rows is equal to the number of columns of seu.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>preHVGs</code></td>
<td>
<p>a positive integer, the number of highly variable genes selected for speeding up computation of SPARK-X  in selecting spatially variable features. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num_core</code></td>
<td>
<p>an optional positive integer, specify the cores used for identifying the SVGs in parallel.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>an optional logical value, whether output the related information.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>return a revised Seurat object by adding three columns named "is.SVGs", "order.SVGs" and "adjusted.pval.SVGs" in the meta.features  of default Assay.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>References</h3>

<p>Zhu, J., Sun, S., Zhou, X.: Spark-x: non-parametric modeling enables scalable and robust detection of spatialexpression patterns for large spatial transcriptomic studies. Genome Biology 22(1), 1-25 (2021)
</p>


<h3>See Also</h3>

<p>topSVGs
</p>


<h3>Examples</h3>

<pre><code class="language-R">  seu&lt;-gendata_RNAExp(height=20, width=20,p=200, K=4)
  seu&lt;-FindSVGs(seu, nfeatures=100)
  topSVGs(seu)

</code></pre>


</div>