<div class="container">

<table style="width: 100%;"><tr>
<td>selectModel</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Select the number of clusters</h2>

<h3>Description</h3>

<p>Select the number of clusters by specified criteria.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  
  selectModel(obj, criteria = 'MBIC', pen.const=1)
  ## S3 method for class 'drscObject'
selectModel(obj, criteria = 'MBIC', pen.const=1)
  ## S3 method for class 'Seurat'
selectModel(obj, criteria = 'MBIC', pen.const=1)
  
  
</code></pre>


<h3>Arguments</h3>

<p>S
</p>
<table>
<tr style="vertical-align: top;">
<td><code>obj</code></td>
<td>
<p>an object with class <code>Seurat</code> by DR.SC or class <code>drscObject</code> by DR.SC_fit.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>criteria</code></td>
<td>
<p>a string, specify the criteria used for selecting the number of clusters, supporting "MBIC", "BIC" and "AIC".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pen.const</code></td>
<td>
<p>an optional positive value, the adjusted constant used in the MBIC criteria. It usually takes value between 0.1 to 1.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>For S3 method of <code>Seurat</code>, it return a  revised "Seurat" object with updated <code>Idents(seu)</code>, <code>spatial.drsc.cluster</code> in the metadata and DimReduc object named <code>dr-sc</code> in the slot <code>reductions</code>. For S3 method of <code>drscObject</code>, it returns a list with the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>bestK</code></td>
<td>
<p>the selected number of clusters.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cluster</code></td>
<td>
<p>inferred class labels</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hZ</code></td>
<td>
<p>extracted latent features.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>icMat</code></td>
<td>
<p>a numeric matrix including the criteria value for each number of clusters K.</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p><a href="https://academic.oup.com/nar/article/50/12/e72/6555431">Wei Liu, Xu Liao, Yi Yang, Huazhen Lin, Joe Yeong, Xiang Zhou, Xingjie Shi &amp; Jin Liu (2022). Joint dimension reduction and clustering analysis of single-cell RNA-seq and spatial transcriptomics data, Nucleic Acids Research, gkac219.</a>
</p>


<h3>See Also</h3>

<p>DR.SC, DR.SC_fit.
</p>


<h3>Examples</h3>

<pre><code class="language-R">seu &lt;- gendata_RNAExp(height=10, width=10,p=50, K=4)
library(Seurat)
seu &lt;- NormalizeData(seu, verbose=FALSE)
# or choose 40 spatailly variable features using FindSVGs in DR.SC
seu &lt;- FindSVGs(seu, nfeatures = 40, verbose=FALSE)
# users define the adjacency matrix
Adj_sp &lt;- getAdj(seu, platform = 'ST')
dat &lt;- GetAssayData(seu, assay = "RNA", slot='data')
X &lt;- Matrix::t(dat)
# maxIter = 2 is only used for illustration, and user can use default.
drscList &lt;- DR.SC_fit(X,Adj_sp=Adj_sp ,K=4, maxIter=2, verbose=TRUE)
drsc1 &lt;- selectModel(drscList)
str(drsc1)
</code></pre>


</div>