<div class="container">

<table style="width: 100%;"><tr>
<td>getAdj</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate the adjacency matrix given the spatial coordinates</h2>

<h3>Description</h3>

<p>Calculate the adjacency matrix for the spatial transcriptomics data measured on 10X Visium, ST or other platforms as a Seurat object.  
</p>


<h3>Usage</h3>

<pre><code class="language-R">
  ## S3 method for class 'Seurat'
getAdj(obj, platform = c('Visium', "ST", "Other_SRT"), ...) 
  
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>obj</code></td>
<td>
<p>an object with class "Seurat", there are spatial coordinates information in the metadata of obj, named "row" and "col", where first column is x-axis coordinate, the second column is y-axis coordinate. getAdj_manual and getAdj_auto  supports multi-dimensional spatial coordinates with a matrix as input.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>platform</code></td>
<td>
<p>a string, specify the platform of the provided data, default as "Visium". There are more platforms to be chosen, including ("Visuim", "ST", "Other_SRT"), which means there are spatial coordinates information in the metadata of obj, named "row" and "col". The platform helps to calculate the adjacency matrix by defining the neighborhoods.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Other arguments to pass into getAdj_auto function.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For lattice grids in ST platform (ST), the interior spot has four neighbors (left, right, up and down),the boundary spot has three neighbors, and the spot in the corner has two neighbors. For hexagon grids, such as spatial coordinate in 10X Visium platform (Visium), the interior spot has six neighbors. For the irregular coordinates in other platforms (Other_SRT), Euclidean distance is adopted to decide whether a spot is an neighbor of another spot. For example, if the Euclidean distance between spot A and B is less than a radius, then A is taken as the neighbourhood of B. See functions getAdj_auto and getAdj_manual for more details.
</p>


<h3>Value</h3>

<p>Return a <code>dgCMatrix</code> object recording the information of neighborhoods about each spot.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p><a href="https://academic.oup.com/nar/article/50/12/e72/6555431">Wei Liu, Xu Liao, Yi Yang, Huazhen Lin, Joe Yeong, Xiang Zhou, Xingjie Shi &amp; Jin Liu (2022). Joint dimension reduction and clustering analysis of single-cell RNA-seq and spatial transcriptomics data, Nucleic Acids Research, gkac219.</a>
</p>


<h3>See Also</h3>

<p>getAdj_auto, getAdj_manual.
</p>


<h3>Examples</h3>

<pre><code class="language-R">  ## S3 method for class "Seurat"
  seu &lt;- gendata_RNAExp(height=20, width=20,p=200, K=4)
  Adj_sp &lt;- getAdj(seu, platform = 'ST')
</code></pre>


</div>