<div class="container">

<table style="width: 100%;"><tr>
<td>DrImpute</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Imputing dropout events in single-cell RNA-sequencing data.</h2>

<h3>Description</h3>

<p>Imputing dropout events in single-cell RNA-sequencing data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">DrImpute(X, ks = 10:15, dists = c("spearman", "pearson"), method = "mean",
  cls = NULL, seed = 1, zerop = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>Gene expression matrix (gene by cell).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ks</code></td>
<td>
<p>Number of cell clustering groups. Default set to ks = 10:15.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dists</code></td>
<td>
<p>Distribution matrices to use. Default is set to c("spearman", "pearson"). "eucleadian" can be added as well.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Use "mean" for mean imputation, "med" for median imputation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cls</code></td>
<td>
<p>User can manually provide clustering information. Using different base clusterings. each row represent different clusterings. each column represent each cell.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>User can provide a seed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>zerop</code></td>
<td>
<p>zero percentage of resulting imputation is at least zerop.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Imputed Gene expression matrix (gene by cell).
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak
</p>


<h3>References</h3>

<p>Il-Youp Kwak, Wuming Gong, Kaoko Koyano-Nakagawa and Daniel J. Garry (2017+)
DrImpute: Imputing dropout eveents in single cell RNA sequencing data
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(exdata)
exdata &lt;- preprocessSC(exdata)
exdata &lt;- exdata[1:3000, 1:80]
logdat &lt;- log(exdata+1)
cls &lt;- getCls(logdat)
logdat_imp &lt;- DrImpute(logdat, cls = cls)

</code></pre>


</div>