<div class="container">

<table style="width: 100%;"><tr>
<td>getCls</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>get base clustering results using SC3 based clustering methods.</h2>

<h3>Description</h3>

<p>Similarity matrix constructed using "pearson", "spearman" or "euclidean". K-means clustering is performed on first few number of principal components of similarity matrix.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getCls(X, ks = 10:15, dists = c("spearman", "pearson"),
  dim.reduc.prop = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>Log transformed gene expression matrix (Gene by Cell).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ks</code></td>
<td>
<p>Number of cell clustering groups. Default set to ks = 10:15.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dists</code></td>
<td>
<p>Distribution matrices to use. Default is set to c("spearman", "pearson"). "euclidean" can be added as well.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dim.reduc.prop</code></td>
<td>
<p>Proportion of principal components to use for K-means clustering.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A matrix object, Each row represent different clustering results.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak
</p>


<h3>References</h3>

<p>Il-Youp Kwak, Wuming Gong, Kaoko Koyano-Nakagawa and Daniel J. Garry (2017+)
DrImpute: Imputing dropout eveents in single cell RNA sequencing data
</p>


<h3>See Also</h3>

<p><code>DrImpute</code>  <code>preprocessSC</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(exdata)
exdata &lt;- preprocessSC(exdata)
exdata &lt;- exdata[1:3000, 1:80]
logdat &lt;- log(exdata+1)
cls &lt;- getCls(logdat)

</code></pre>


</div>