<div class="container">

<table style="width: 100%;"><tr>
<td>preprocessSC</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>A function for preprocessing gene expression matrix.</h2>

<h3>Description</h3>

<p>Preprocess gene expression data
</p>


<h3>Usage</h3>

<pre><code class="language-R">preprocessSC(X, min.expressed.gene = 0, min.expressed.cell = 2,
  max.expressed.ratio = 1, normalize.by.size.effect = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>Gene expression matrix (Gene by Cell).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.expressed.gene</code></td>
<td>
<p>Cell level filtering criteria. For a given cell, if the number of expressed genes are less than min.expressed.gene, we filter it out.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.expressed.cell</code></td>
<td>
<p>Gene level filtering criteria. For a given gene, if the number of expressed cells are less than min.expressed.cell, we filter it out.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.expressed.ratio</code></td>
<td>
<p>Gene level filtering criteria. For a given gene, if the ratio of expressed cells are larger than max.expressed.ratio, we filter it out.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normalize.by.size.effect</code></td>
<td>
<p>Normaize using size factor.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Filtered gene expression matrix
</p>


<h3>Author(s)</h3>

<p>Wuming Gong
</p>


<h3>References</h3>

<p>Il-Youp Kwak, Wuming Gong, Kaoko Koyano-Nakagawa and Daniel J. Garry (2017+)
DrImpute: Imputing dropout eveents in single cell RNA sequencing data
</p>


<h3>See Also</h3>

<p><code>DrImpute</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(exdata)
exdata &lt;- preprocessSC(exdata)

</code></pre>


</div>