<div class="container">

<table style="width: 100%;"><tr>
<td>create_features_df</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create Data Frame of Features for Driver Gene Prioritization</h2>

<h3>Description</h3>

<p>Create Data Frame of Features for Driver Gene Prioritization
</p>


<h3>Usage</h3>

<pre><code class="language-R">create_features_df(
  annovar_csv_path,
  scna_df,
  phenolyzer_annotated_gene_list_path,
  batch_analysis = FALSE,
  prep_phenolyzer_input = FALSE,
  build = "GRCh37",
  log2_ratio_threshold = 0.25,
  gene_overlap_threshold = 25,
  MCR_overlap_threshold = 25,
  hotspot_threshold = 5L,
  log2_hom_loss_threshold = -1,
  verbose = TRUE,
  na.string = "."
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>annovar_csv_path</code></td>
<td>
<p>path to 'ANNOVAR' csv output file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scna_df</code></td>
<td>
<p>the SCNA segments data frame. Must contain: </p>

<dl>
<dt>chr</dt>
<dd>
<p>chromosome the segment is located in</p>
</dd>
<dt>start</dt>
<dd>
<p>start position of the segment</p>
</dd>
<dt>end</dt>
<dd>
<p>end position of the segment</p>
</dd>
<dt>log2ratio</dt>
<dd>log<sub>2</sub><p> ratio of the segment</p>
</dd>
</dl>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phenolyzer_annotated_gene_list_path</code></td>
<td>
<p>path to 'phenolyzer'
"annotated_gene_list" file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batch_analysis</code></td>
<td>
<p>boolean to indicate whether to perform batch analysis
(<code>TRUE</code>, default) or personalized analysis (<code>FALSE</code>). If <code>TRUE</code>,
a column named 'tumor_id' should be present in both the ANNOVAR csv and the SCNA
table.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prep_phenolyzer_input</code></td>
<td>
<p>boolean to indicate whether or not to create
a vector of genes for use as the input of 'phenolyzer' (default = <code>FALSE</code>).
If <code>TRUE</code>, the features data frame is not created and instead the vector
of gene symbols (union of all genes for which scores are available) is
returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>build</code></td>
<td>
<p>genome build for the SCNA segments data frame (default = "GRCh37")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log2_ratio_threshold</code></td>
<td>
<p>the log<sub>2</sub>
ratio threshold for keeping high-confidence SCNA events (default = 0.25)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene_overlap_threshold</code></td>
<td>
<p>the percentage threshold for the overlap between
a segment and a transcript (default = 25). This means that if only a segment
overlaps a transcript more than this threshold, the transcript is assigned
the segment's SCNA event.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MCR_overlap_threshold</code></td>
<td>
<p>the percentage threshold for the overlap between
a gene and an MCR region (default = 25). This means that if only a gene
overlaps an MCR region more than this threshold, the gene is assigned the
SCNA density of the MCR</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hotspot_threshold</code></td>
<td>
<p>to determine hotspot genes, the (integer) threshold
for the minimum number of cases with certain mutation in COSMIC (default = 5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log2_hom_loss_threshold</code></td>
<td>
<p>to determine double-hit events, the
log<sub>2</sub> threshold for identifying
homozygous loss events (default = -1).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>boolean controlling verbosity (default = <code>TRUE</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.string</code></td>
<td>
<p>string that was used to indicate when a score is not available
during annotation with ANNOVAR (default = ".")</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>If <code>prep_phenolyzer_input=FALSE</code> (default), a data frame of
features for prioritizing cancer driver genes (<code>gene_symbol</code> as
the first column and 26 other columns containing features). If
<code>prep_phenolyzer_input=TRUE</code>, the functions returns a vector gene symbols
(union of all gene symbols for which scores are available) to be used as the
input for performing 'phenolyzer' analysis.
</p>
<p>The features data frame contains the following columns:
</p>

<dl>
<dt>gene_symbol</dt>
<dd>
<p>HGNC gene symbol</p>
</dd>
<dt>metaprediction_score</dt>
<dd>
<p>the maximum metapredictor (coding) impact score for the gene</p>
</dd>
<dt>noncoding_score</dt>
<dd>
<p>the maximum non-coding PHRED-scaled CADD score for the gene</p>
</dd>
<dt>scna_score</dt>
<dd>
<p>SCNA proxy score. SCNA density (SCNA/Mb) of the minimal common region (MCR) in which the gene is located</p>
</dd>
<dt>hotspot_double_hit</dt>
<dd>
<p>boolean indicating whether the gene is a hotspot gene (indication of oncogenes) or subject to double-hit (indication of tumor-suppressor genes)</p>
</dd>
<dt>phenolyzer_score</dt>
<dd>
<p>'phenolyzer' score for the gene</p>
</dd>
<dt>hsa03320</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04010</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04020</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04024</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04060</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04066</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04110</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04115</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04150</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04151</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04210</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04310</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04330</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04340</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04350</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04370</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04510</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04512</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04520</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04630</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04915</dt>
<dd>
<p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
</dl>
<h3>See Also</h3>

<p><code>prioritize_driver_genes</code> for prioritizing cancer driver genes
</p>


<h3>Examples</h3>

<pre><code class="language-R">
path2annovar_csv &lt;- system.file("extdata/example.hg19_multianno.csv",
                                package = "driveR")
path2phenolyzer_out &lt;- system.file("extdata/example.annotated_gene_list",
                                   package = "driveR")
features_df &lt;- create_features_df(annovar_csv_path = path2annovar_csv,
                                  scna_df = example_scna_table,
                                  phenolyzer_annotated_gene_list_path = path2phenolyzer_out)


</code></pre>


</div>