<div class="container">

<table style="width: 100%;"><tr>
<td>create_gene_level_scna_df</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create Gene-level SCNA Data Frame</h2>

<h3>Description</h3>

<p>Create Gene-level SCNA Data Frame
</p>


<h3>Usage</h3>

<pre><code class="language-R">create_gene_level_scna_df(
  scna_df,
  build = "GRCh37",
  gene_overlap_threshold = 25
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>scna_df</code></td>
<td>
<p>the SCNA segments data frame. Must contain: </p>

<dl>
<dt>chr</dt>
<dd>
<p>chromosome the segment is located in</p>
</dd>
<dt>start</dt>
<dd>
<p>start position of the segment</p>
</dd>
<dt>end</dt>
<dd>
<p>end position of the segment</p>
</dd>
<dt>log2ratio</dt>
<dd>log<sub>2</sub><p> ratio of the segment</p>
</dd>
</dl>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>build</code></td>
<td>
<p>genome build for the SCNA segments data frame (default = "GRCh37")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene_overlap_threshold</code></td>
<td>
<p>the percentage threshold for the overlap between
a segment and a transcript (default = 25). This means that if only a segment
overlaps a transcript more than this threshold, the transcript is assigned
the segment's SCNA event.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>data frame of gene-level SCNA events, i.e. table of genes overlapped
by SCNA segments.
</p>


</div>