<div class="container">

<table style="width: 100%;"><tr>
<td>microarraydata</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Import, check and normalization of single-channel microarray data</h2>

<h3>Description</h3>

<p>Single-channel microarray data in log2 are imported from a .txt file 
(internally imported using the function <code>read.table</code>), 
checked or from a R object of class <code>data.frame</code> 
(see the description 
of argument <code>file</code> for the required format
of data)and normalized (between arrays
normalization).
<code>omicdata</code> is a deprecated version of <code>microarraydata</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">microarraydata(file, backgrounddose, check = TRUE, 
  norm.method = c("cyclicloess", "quantile", "scale", "none"))
  
omicdata(file, backgrounddose, check = TRUE, 
  norm.method = c("cyclicloess", "quantile", "scale", "none"))

## S3 method for class 'microarraydata'
print(x, ...)
## S3 method for class 'microarraydata'
plot(x, range4boxplot = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>The name of the .txt file (e.g. <code>"mydata.txt"</code>)
containing one row per item, with the first column corresponding to the identifier of each item, 
and the other columns giving the responses
of the item for each replicate at each dose or concentration. 
In the first line, after a name for the identifier column, we must have the tested
doses or concentrations in a numeric format for the corresponding replicate 
(for example, if there are triplicates for each treatment, 
the first line could be "item", 0, 0, 0, 0.1, 0.1, 0.1, etc.). 
This file is imported within the function using the function
<code>read.table</code> with its default field separator (sep argument) and 
its default decimal separator (dec argument at ".").
Alternatively an R object of class <code>data.frame</code> can be directly given in input, 
corresponding to the output
of <code>read.table(file, header = FALSE)</code> on a file described as above. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>backgrounddose</code></td>
<td>
<p>This argument must be used when there is no dose at zero in the data, to prevent the calculation of the BMD by extrapolation. All doses below  or equal to the value given in backgrounddose will be fixed at 0, so as to be considered at the background level of exposition.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check</code></td>
<td>
<p>If TRUE the format of the input file is checked.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>norm.method</code></td>
<td>
<p>If <code>"none"</code> no normalization is performed, else a 
normalization is performed using the function <code>normalizeBetweenArrays</code> of the 
<code>limma</code> package using the specified method.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object of class <code>"microarraydata"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>range4boxplot</code></td>
<td>
<p>An argument passed to boxplot(), fixed by default at 0 
to prevent the producing of very large plot files due to many outliers. 
Can be put at 1.5 to obtain more classical boxplots.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> further arguments passed to print or plot functions.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function imports the data, checks their format (see the description 
of argument <code>file</code> for the required format
of data) and gives in the <code>print</code> information
that should help the user to check that the coding of data is correct : the tested doses (or concentrations) 
the number of replicates for each dose, the number of items, the identifiers 
of the first 20 items. If the argument <code>norm.method</code> is not <code>"none"</code>,
data are normalized using the function <code>normalizeBetweenArrays</code> of the 
<code>limma</code> package using the specified method : <code>"cyclicloess"</code> (default choice), 
<code>"quantile"</code> or <code>"scale"</code>.
</p>


<h3>Value</h3>

 
<p><code>microarraydata</code> returns an object of class "microarraydata", a list with 9 components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p> the numeric matrix of normalized responses of each item in each replicate
(one line per item, one column per replicate)  </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dose</code></td>
<td>
<p> the numeric vector of the tested doses or concentrations corresponding
to each column of data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>item</code></td>
<td>
<p> the character vector of the identifiers of the items, corresponding to
each line of data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>design</code></td>
<td>
<p> a table with the experimental design (tested doses and number of 
replicates for each dose) for control by the user</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data.mean</code></td>
<td>
<p> the numeric matrix of mean responses of each item per dose (mean 
of the corresponding replicates) (one line per item, one column per unique value 
of the dose</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data.sd</code></td>
<td>
<p> the numeric matrix of standard deviations of the response of each 
item per dose (sd of the corresponding replicates, NA if no replicate) 
(one line per item, one column per unique value of the dose)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>norm.method</code></td>
<td>
<p>The normalization method specified in input</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data.beforenorm</code></td>
<td>
<p> the numeric matrix of responses of each item in each replicate
(one line per item, one column per replicate) before normalization  </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>containsNA</code></td>
<td>
<p>always at FALSE as microarray data are not allowed to contain 
NA values</p>
</td>
</tr>
</table>
<p>The print of a <code>microarraydata</code> object gives the tested doses (or concentrations) 
and number of replicates for each dose, the number of items, the identifiers 
of the first 20 items (for check of good coding of data) and the normalization method.
The plot of a <code>microarraydata</code> object shows the data distribution for each dose or concentration and replicate before and after normalization.
</p>


<h3>Author(s)</h3>

 
<p>Marie-Laure Delignette-Muller
</p>


<h3>References</h3>

 
<p>Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W, and Smyth, GK (2015), <em>limma powers differential expression analyses for RNA-sequencing and microarray studies</em>. Nucleic Acids Research 43, e47.
</p>


<h3>See Also</h3>

<p>See <code>read.table</code> the function used to import data, 
<code>normalizeBetweenArrays</code> for details about the normalization and 
<code>RNAseqdata</code>, <code>continuousomicdata</code> and
<code>continuousanchoringdata</code> for other types of data.</p>


<h3>Examples</h3>

<pre><code class="language-R"># (1) import, check and normalization of microarray data 
# (an example on a subsample of a greater data set published in Larras et al. 2018
# Transcriptomic effect of triclosan in the chlorophyte Scenedesmus vacuolatus)
datafilename &lt;- system.file("extdata", "transcripto_very_small_sample.txt", 
  package="DRomics")
o &lt;- microarraydata(datafilename, check = TRUE, norm.method = "cyclicloess")
print(o)
plot(o)
PCAdataplot(o)
PCAdataplot(o, label = TRUE)


# If you want to use your own data set just replace datafilename, 
# the first argument of microarraydata(), 
# by the name of your data file (e.g. "mydata.txt")
# 
# You should take care that the field separator of this data file is one
# of the default field separators recognised by the read.table() function
# when it is used with its default field separator (sep argument)
# Tabs are recommended.



# (2) normalization with other methods
(o.2 &lt;- microarraydata(datafilename, check = TRUE, norm.method = "quantile"))
plot(o.2)
(o.3 &lt;- microarraydata(datafilename, check = TRUE, norm.method = "scale"))
plot(o.3)


</code></pre>


</div>