<div class="container">

<table style="width: 100%;"><tr>
<td>gsa_label</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Sample labels in gene set analysis
</h2>

<h3>Description</h3>

<p>P53 NCI-60 data set provided by Subramanian <em>et al</em>., (2005). A list with phenotypes vector and a vector of sample label values.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(gsa_label)</code></pre>


<h3>Format</h3>

<p>A list on the following 2 variables.
</p>

<dl>
<dt><code>pheotype</code></dt>
<dd>
<p>a character vector contains two genotypes.</p>
</dd>
<dt><code>value</code></dt>
<dd>
<p>a numeric vector contains sample label 0 and 1, where 0 and 1 stands for “MUT” and “WT”, respectively.</p>
</dd>
</dl>
<h3>Source</h3>

<p>http://www.broadinstitute.org/gsea
</p>


<h3>References</h3>

<p>Subramanian, A., Tamayo, P., Mootha, V. K., <em>et al</em>. Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles. <em>Proceedings of the National Academy of Sciences of the United States of America</em>, 2005, 102(43): 15545-15550.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(gsa_label)
</code></pre>


</div>