<div class="container">

<table style="width: 100%;"><tr>
<td>dyngen</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>dyngen: A multi-modal simulator for spearheading single-cell omics analyses</h2>

<h3>Description</h3>

<p>A toolkit for generating synthetic single cell data.
</p>


<h3>Step 1, initialise dyngen model</h3>


<ul>
<li> <p><code>initialise_model()</code>: Define and store settings for all following steps. See each of the sections below for more information.
</p>
</li>
<li>
<p> Use a predefined backbone:
</p>

<ul>
<li> <p><code>list_backbones()</code>
</p>
</li>
<li> <p><code>backbone_bifurcating()</code>
</p>
</li>
<li> <p><code>backbone_bifurcating_converging()</code>
</p>
</li>
<li> <p><code>backbone_bifurcating_cycle()</code>
</p>
</li>
<li> <p><code>backbone_bifurcating_loop()</code>
</p>
</li>
<li> <p><code>backbone_branching()</code>
</p>
</li>
<li> <p><code>backbone_binary_tree()</code>
</p>
</li>
<li> <p><code>backbone_consecutive_bifurcating()</code>
</p>
</li>
<li> <p><code>backbone_trifurcating()</code>
</p>
</li>
<li> <p><code>backbone_converging()</code>
</p>
</li>
<li> <p><code>backbone_cycle()</code>
</p>
</li>
<li> <p><code>backbone_cycle_simple()</code>
</p>
</li>
<li> <p><code>backbone_linear()</code>
</p>
</li>
<li> <p><code>backbone_linear_simple()</code>
</p>
</li>
<li> <p><code>backbone_disconnected()</code>
</p>
</li>
</ul>
</li>
<li>
<p> Create a custom backbone:
</p>

<ul>
<li> <p><code>backbone()</code>
</p>
</li>
<li> <p><code>bblego()</code>
</p>
</li>
<li> <p><code>bblego_linear()</code>
</p>
</li>
<li> <p><code>bblego_branching()</code>
</p>
</li>
<li> <p><code>bblego_start()</code>
</p>
</li>
<li> <p><code>bblego_end()</code>
</p>
</li>
</ul>
</li>
<li>
<p> Visualise the backbone:
</p>

<ul>
<li> <p><code>plot_backbone_modulenet()</code>
</p>
</li>
<li> <p><code>plot_backbone_statenet()</code>
</p>
</li>
</ul>
</li>
</ul>
<h3>Step 2, generate TF network</h3>


<ul>
<li> <p><code>generate_tf_network()</code>: Generate a transcription factor network from the backbone
</p>
</li>
<li> <p><code>tf_network_default()</code>: Parameters for configuring this step
</p>
</li>
</ul>
<h3>Step 3, add more genes to the gene network</h3>


<ul>
<li> <p><code>generate_feature_network()</code>: Generate a target network
</p>
</li>
<li> <p><code>feature_network_default()</code>: Parameters for configuring this step
</p>
</li>
<li> <p><code>plot_feature_network()</code>: Visualise the gene network
</p>
</li>
</ul>
<h3>Step 4, generate gene kinetics</h3>


<ul>
<li> <p><code>generate_kinetics()</code>: Generate the gene kinetics
</p>
</li>
<li> <p><code>kinetics_default()</code>, <code>kinetics_random_distributions()</code>: Parameters for configuring this step
</p>
</li>
</ul>
<h3>Step 5, simulate the gold standard</h3>


<ul>
<li> <p><code>generate_gold_standard()</code>: Simulate the gold standard backbone, used for mapping to cell states afterwards
</p>
</li>
<li> <p><code>gold_standard_default()</code>: Parameters for configuring this step
</p>
</li>
<li> <p><code>plot_gold_mappings()</code>: Visualise the mapping of the simulations to the gold standard
</p>
</li>
<li> <p><code>plot_gold_simulations()</code>: Visualise the gold standard simulations using the dimred
</p>
</li>
<li> <p><code>plot_gold_expression()</code>: Visualise the expression of the gold standard over simulation time
</p>
</li>
</ul>
<h3>Step 6, simulate the cells</h3>


<ul>
<li> <p><code>generate_cells()</code>: Simulate the cells based on its GRN
</p>
</li>
<li> <p><code>simulation_default()</code>: Parameters for configuring this step
</p>
</li>
<li> <p><code>simulation_type_wild_type()</code>, <code>simulation_type_knockdown()</code>: Used for configuring the type of simulation
</p>
</li>
<li> <p><code>kinetics_noise_none()</code>, <code>kinetics_noise_simple()</code>: Different kinetics randomisers to apply to each simulation
</p>
</li>
<li> <p><code>plot_simulations()</code>: Visualise the simulations using the dimred
</p>
</li>
<li> <p><code>plot_simulation_expression()</code>: Visualise the expression of the simulations over simulation time
</p>
</li>
</ul>
<h3>Step 7, simulate cell and transcripting sampling</h3>


<ul>
<li> <p><code>generate_experiment()</code>: Sample cells and transcripts from experiment
</p>
</li>
<li> <p><code>list_experiment_samplers()</code>, <code>experiment_snapshot()</code>, <code>experiment_synchronised()</code>: Parameters for configuring this step
</p>
</li>
<li> <p><code>simtime_from_backbone()</code>: Determine the simulation time from the backbone
</p>
</li>
<li> <p><code>plot_experiment_dimred()</code>: Plot a dimensionality reduction of the final dataset
</p>
</li>
</ul>
<h3>Step 8, convert to dataset</h3>


<ul>
<li> <p><code>as_dyno()</code>, <code>wrap_dataset()</code>: Convert a dyngen model to a dyno dataset
</p>
</li>
<li> <p><code>as_anndata()</code>: Convert a dyngen model to an anndata dataset
</p>
</li>
<li> <p><code>as_sce()</code>: Convert a dyngen model to a SingleCellExperiment dataset
</p>
</li>
<li> <p><code>as_seurat()</code>: Convert a dyngen model to a Seurat dataset
</p>
</li>
</ul>
<h3>One-shot function</h3>


<ul>
<li> <p><code>generate_dataset()</code>: Run through steps 2 to 8 with a single function
</p>
</li>
<li> <p><code>plot_summary()</code>: Plot a summary of all dyngen simulation steps
</p>
</li>
</ul>
<h3>Data objects</h3>


<ul>
<li> <p>example_model: A (very) small toy dyngen model, used for documentation and testing purposes
</p>
</li>
<li> <p>realcounts: A set of real single-cell expression datasets, to be used as reference datasets
</p>
</li>
<li> <p>realnets: A set of real gene regulatory networks, to be sampled in step 3
</p>
</li>
</ul>
<h3>Varia functions</h3>


<ul>
<li> <p>dyngen: This help page
</p>
</li>
<li> <p><code>get_timings()</code>: Extract execution timings for each of the dyngen steps
</p>
</li>
<li> <p><code>combine_models()</code>: Combine multiple dyngen models
</p>
</li>
<li> <p><code>rnorm_bounded()</code>: A bounded version of <code>rnorm()</code>
</p>
</li>
<li> <p><code>runif_subrange()</code>: A subrange version of <code>runif()</code>
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">model &lt;- initialise_model(
  backbone = backbone_bifurcating()
)


model &lt;- model %&gt;%
  generate_tf_network() %&gt;%
  generate_feature_network() %&gt;%
  generate_kinetics() %&gt;%
  generate_gold_standard() %&gt;%
  generate_cells() %&gt;%
  generate_experiment()
  
dataset &lt;- wrap_dataset(model, format = "dyno")
# format can also be set to "sce", "seurat", "anndata" or "list"

# library(dynplot)
# plot_dimred(dataset)


</code></pre>


</div>